From 39126809c2854a88175500eb3f70fed139e6ee14 Mon Sep 17 00:00:00 2001
From: Led <led@altlinux.ru>
Date: Thu, 23 May 2013 20:18:35 +0300
Subject: [PATCH 02/28] fix Makefile

---
 MPlayer/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MPlayer/Makefile b/MPlayer/Makefile
index d2e406723a..1f125f5ff5 100644
--- a/MPlayer/Makefile
+++ b/MPlayer/Makefile
@@ -978,7 +978,7 @@ endif
 
 ###### tests / tools #######
 
-MP_MSG_OBJS = mp_msg.o mp_fifo.o osdep/$(GETCH) osdep/$(TIMER)
+MP_MSG_OBJS = mp_msg.o mp_fifo.o osdep/$(GETCH) osdep/$(TIMER) -ltinfo -lm
 
 libvo/aspecttest$(EXESUF): libvo/aspect.o libvo/geometry.o $(MP_MSG_OBJS)
 libvo/aspecttest$(EXESUF): LIBS = $(MP_MSG_LIBS)
-- 
2.25.3

