From 5cec969243af9a3ea3d512444614a67ca6e55506 Mon Sep 17 00:00:00 2001
From: Led <led@altlinux.ru>
Date: Thu, 23 May 2013 20:18:35 +0300
Subject: [PATCH] fix Makefile

---
 MPlayer/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MPlayer/Makefile b/MPlayer/Makefile
index 56fb7780ed..24ffd22a21 100644
--- a/MPlayer/Makefile
+++ b/MPlayer/Makefile
@@ -998,7 +998,7 @@ endif
 
 ###### tests / tools #######
 
-MP_MSG_OBJS = mp_msg.o mp_fifo.o osdep/$(GETCH) osdep/$(TIMER)
+MP_MSG_OBJS = mp_msg.o mp_fifo.o osdep/$(GETCH) osdep/$(TIMER) -ltinfo -lm
 
 libvo/aspecttest$(EXESUF): libvo/aspect.o libvo/geometry.o $(MP_MSG_OBJS)
 libvo/aspecttest$(EXESUF): LIBS = $(MP_MSG_LIBS)
-- 
2.29.2

