From 14698b376f5a9520f06bfa9c85d1cb61830a6355 Mon Sep 17 00:00:00 2001
From: Led <led@altlinux.org>
Date: Sat, 25 May 2013 05:47:09 +0300
Subject: [PATCH] fixed subreader

---
 MPlayer/sub/subreader.c | 18 ++++++------------
 1 file changed, 6 insertions(+), 12 deletions(-)

diff --git a/MPlayer/sub/subreader.c b/MPlayer/sub/subreader.c
index c862be7bc0..b6799af1cb 100644
--- a/MPlayer/sub/subreader.c
+++ b/MPlayer/sub/subreader.c
@@ -1884,22 +1884,16 @@ char * strreplace( char * in,char * what,char * whereof )
 static void strcpy_trim(char *d, const char *s)
 {
     // skip leading whitespace
-    while (*s && isspace(*s)) {
-	s++;
-    }
-    for (;;) {
+    for (; *s && isascii(*s) && !isalnum(*s); s++);
+    for (;; d++) {
 	// copy word
-	while (*s && !isspace(*s)) {
-	    *d = tolower(*s);
-	    s++; d++;
-	}
+	for (; *s && (!isascii(*s) || isalnum(*s)); s++, d++)
+	    *d = *s;
 	if (*s == 0) break;
 	// trim excess whitespace
-	while (*s && isspace(*s)) {
-	    s++;
-	}
+	for (; *s && isascii(*s) && !isalnum(*s); s++);
 	if (*s == 0) break;
-	*d++ = ' ';
+	*d = ' ';
     }
     *d = 0;
 }
-- 
2.29.2

