From 85b1bf33028092f5d5145fe86d3b6991aae3c7dc Mon Sep 17 00:00:00 2001
From: Led <led@altlinux.ru>
Date: Thu, 30 May 2013 19:56:28 +0300
Subject: [PATCH 10/28] update ru translation

---
 MPlayer/help/help_mp-ru.h | 195 ++++++++++++++++++++++----------------
 1 file changed, 114 insertions(+), 81 deletions(-)

diff --git a/MPlayer/help/help_mp-ru.h b/MPlayer/help/help_mp-ru.h
index bcd480a898..888c368723 100644
--- a/MPlayer/help/help_mp-ru.h
+++ b/MPlayer/help/help_mp-ru.h
@@ -75,8 +75,8 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_CantOpenDumpfile "Не могу открыть файл дампа!!!\n"
 #define MSGTR_CoreDumped "Создан дамп ядра ;)\n"
 #define MSGTR_DumpBytesWrittenPercent "дамп: %"PRIu64" байт(а) записано (~%.1f%%)\r"
-#define MSGTR_DumpBytesWritten "дамп: %"PRIu64" байт записано\r"
-#define MSGTR_DumpBytesWrittenTo "дамп: %"PRIu64" байт записано в '%s'.\n"
+#define MSGTR_DumpBytesWritten "дамп: %"PRIu64" байт(а) записано\r"
+#define MSGTR_DumpBytesWrittenTo "дамп: %"PRIu64" байт(а) записано в '%s'.\n"
 #define MSGTR_FPSnotspecified "В заголовке кадры/сек не указаны (или недопустимые)! Используйте опцию -fps!\n"
 #define MSGTR_TryForceAudioFmtStr "Попытка форсировать семейство аудиокодеков %s...\n"
 #define MSGTR_CantFindAudioCodec "Не могу найти кодек для аудиоформата 0x%X!\n"
@@ -163,16 +163,46 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_LinuxRTCReadError "Ошибка чтения Linux RTC: %s\n"
 #define MSGTR_SoftsleepUnderflow "Предупреждение! Недопустимо низкое значение программной задержки!\n"
 
-#define MSGTR_EdlOutOfMem "Не могу выделить достаточный объём памяти для хранения данных EDL.\n"
-#define MSGTR_EdlOutOfMemFile "Не могу выделить достаточный объём памяти для хранения данных EDL имени файла [%s].\n"
+#define MSGTR_DvdnavNullEvent "Событие DVDNAV NULL?!\n"
+#define MSGTR_DvdnavHighlightEventBroken "Событие DVDNAV: Событие выделения сломано\n"
+#define MSGTR_DvdnavEvent "Событие DVDNAV: %s\n"
+#define MSGTR_DvdnavHighlightHide "Событие DVDNAV: Выделение скрыто\n"
+#define MSGTR_DvdnavStillFrame "######################################## Событие DVDNAV: Стоп-кадр: %d сек\n"
+#define MSGTR_DvdnavNavStop "Событие DVDNAV: Остановка Nav \n"
+#define MSGTR_DvdnavNavNOP "Событие DVDNAV: Nav NOP\n"
+#define MSGTR_DvdnavNavSpuStreamChangeVerbose "Событие DVDNAV: Изменение SPU-потока Nav: физически: %d/%d/%d логически: %d\n"
+#define MSGTR_DvdnavNavSpuStreamChange "Событие DVDNAV: Изменение SPU-потока Nav: физически: %d логически: %d\n"
+#define MSGTR_DvdnavNavAudioStreamChange "Событие DVDNAV: Изменение аудиопотока Nav: физически: %d логически: %d\n"
+#define MSGTR_DvdnavNavVTSChange "Событие DVDNAV: Изменение Nav VTS\n"
+#define MSGTR_DvdnavNavCellChange "Событие DVDNAV: Изменение ячейки Nav\n"
+#define MSGTR_DvdnavNavSpuClutChange "Событие DVDNAV: Изменение Nav SPU CLUT\n"
+#define MSGTR_DvdnavNavSeekDone "Событие DVDNAV: Завершено позиционирование Nav\n"
+#define MSGTR_MenuCall "Вызов меню\n"
+#define MSGTR_MasterQuit "Опция -udp-slave: выход по причине выхода мастера\n"
+#define MSGTR_InvalidIP "Опция -udp-ip: неверный IP-адрес\n"
+#define MSGTR_Forking "Ответвление...\n"
+#define MSGTR_Forked "Ответвлён...\n"
+#define MSGTR_CouldntStartGdb "Невозможно запустить gdb\n"
+#define MSGTR_CouldntFork "Ответвление невозможно\n"
+#define MSGTR_FilenameTooLong "Имя файла слишком длинное, невозможно загрузить файл или специфичные для каталога конфигурационные файлы\n"
+#define MSGTR_AudioDeviceStuck "Звуковое устройство залипло!\n"
+#define MSGTR_AudioOutputTruncated "Вывод звука усечен в конце.\n"
+#define MSGTR_ASSCannotAddVideoFilter "ASS: невозможно добавить видео-фильтр\n"
+#define MSGTR_PtsAfterFiltersMissing "pts после фильтров ОТСУТСВУЕТ\n"
+#define MSGTR_CommandLine "Командная строка:"
+#define MSGTR_MenuInitFailed "Сбой инициализации меню.\n"
+
+// --- edit decision lists
+#define MSGTR_EdlOutOfMem "Не могу выделить достаточно памяти для хранения данных EDL.\n"
+#define MSGTR_EdlOutOfMemFile "Не могу выделить достаточо памяти для хранения данных EDL имени файла [%s].\n"
 #define MSGTR_EdlRecordsNo "Чтение %d EDL действий.\n"
 #define MSGTR_EdlQueueEmpty "Нет действий EDL, которые следует исполнить (очередь пуста).\n"
 #define MSGTR_EdlCantOpenForWrite "Не могу открыть файл EDL [%s] для записи.\n"
 #define MSGTR_EdlNOsh_video "Нельзя использовать EDL без видео, отключаю.\n"
 #define MSGTR_EdlNOValidLine "Неверная строка EDL: %s\n"
 #define MSGTR_EdlBadlyFormattedLine "Плохо форматированная строка EDL [%d]. Пропускаю.\n"
-#define MSGTR_EdlBadLineOverlap "Последняя позиция останова была [%f]; следующая стартовая "\
-"позиция [%f]. Записи должны быть в хронологическом порядке, не могу перекрыть.\nПропускаю.\n"
+#define MSGTR_EdlBadLineOverlap "Последняя позиция останова была [%f]; следующая стартовая позиция [%f].\n"\
+"Записи должны быть в хронологическом порядке, не могу перекрыть.\nПропускаю.\n"
 #define MSGTR_EdlBadLineBadStop "Время останова должно быть после времени старта.\n"
 #define MSGTR_EdloutBadStop "EDL пропуск отменён, последний start > stop\n"
 #define MSGTR_EdloutStartSkip "Начало EDL пропуска, нажмите 'i' ещё раз для завершения блока.\n"
@@ -212,7 +242,7 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_Brightness "Яркость"
 #define MSGTR_Contrast "Контрастность"
 #define MSGTR_Saturation "Насыщенность"
-#define MSGTR_Hue "Цвет"
+#define MSGTR_Hue "Оттенок"
 #define MSGTR_Balance "Баланс"
 
 // property state
@@ -723,7 +753,7 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_PlaylistSelect "Выбрать список воспроизведения..."
 #define MSGTR_SelectTitleChapter "Выбрать ролик/раздел..."
 #define MSGTR_MENU_DebugConsole "Консоль отладки"
-#define MSGTR_MENU_OnlineHelp "Онлай справка"
+#define MSGTR_MENU_OnlineHelp "Онлайн справка"
 #define MSGTR_MENU_PlayDirectory "Воспроизвести каталог..."
 #define MSGTR_MENU_SeekBack "Перейти назад"
 #define MSGTR_MENU_SeekForw "Перейти вперёд"
@@ -801,6 +831,7 @@ static const char help_text[] = MSGTR_Help;
 "           1 : тусклый\n           2 : жирный\n           3 : жирный шрифт\n" \
 "           4 : инверсный\n           5 : специальный\n\n\n"
 
+
 // libvo/vo_dxr3.c
 #define MSGTR_LIBVO_DXR3_UnableToLoadNewSPUPalette "[VO_DXR3] Не могу загрузить новую палитру SPU!\n"
 #define MSGTR_LIBVO_DXR3_UnableToSetPlaymode "[VO_DXR3] Не могу задать режим воспроизведения!\n"
@@ -998,17 +1029,17 @@ static const char help_text[] = MSGTR_Help;
 
 // libvo/vo_xv.c
 #define MSGTR_LIBVO_XV_SharedMemoryNotSupported "[VO_XV] Разделяемая память не поддерживается\nВозврат к обычному Xv.\n"
-#define MSGTR_LIBVO_XV_XvNotSupportedByX11 "[VO_XV] Извините, Xv не поддерживается этой версией/драйвером X11\n[VO_XV] ******** Попробуйте -vo x11  или  -vo sdl  *********\n"
+#define MSGTR_LIBVO_XV_XvNotSupportedByX11 "[VO_XV] К сожалению, Xv не поддерживается этой версией/драйвером X11\n[VO_XV] ******** Попробуйте -vo x11  или  -vo sdl  *********\n"
 #define MSGTR_LIBVO_XV_XvQueryAdaptorsFailed  "[VO_XV] Ошибка XvQueryAdaptors.\n"
 #define MSGTR_LIBVO_XV_InvalidPortParameter "[VO_XV] Неверный параметр port, переопределение параметром port 0.\n"
 #define MSGTR_LIBVO_XV_CouldNotGrabPort "[VO_XV] Невозможно захватить порт %i.\n"
-#define MSGTR_LIBVO_XV_CouldNotFindFreePort "[VO_XV] Невозможно найти свободный Xvideo порт - возможно другое приложение\n"\
-"[VO_XV] уже его использует. Закройте все программы видео, и попытайтесь снова. Если это\n"\
-"[VO_XV] не помогает, смотрите другие (не-xv) драйверы в выводе 'mplayer -vo help'.\n"
-#define MSGTR_LIBVO_XV_NoXvideoSupport "[VO_XV] Похоже, ваша карта не имеет поддержки Xvideo.\n"\
-"[VO_XV] Запустите 'xvinfo', чтобы проверить поддержку Xv и прочтите \n"\
+#define MSGTR_LIBVO_XV_CouldNotFindFreePort "[VO_XV] Невозможно найти свободный Xvideo порт - возможно другое приложение уже\n"\
+"[VO_XV] его использует. Закройте все видео-приложения и попытайтесь снова. Если это\n"\
+"[VO_XV] не помогает, смотрите 'mplayer -vo help'для другич (не-xv) драйверов видко-вывода.\n"
+#define MSGTR_LIBVO_XV_NoXvideoSupport "[VO_XV] Похоже, что ваша видео-карта не имеет поддержки Xvideo.\n"\
+"[VO_XV] Запустите 'xvinfo', чтобы проверить поддержку Xv и читайте \n"\
 "[VO_XV] DOCS/HTML/ru/video.html#xv!\n"\
-"[VO_XV] Другие (не-xv) драйверы ищите в выводе 'mplayer -vo help'.\n"\
+"[VO_XV] Смотрите 'mplayer -vo help' для других (не-xv) драйверов видео-вывода.\n"\
 "[VO_XV] Попробуйте -vo x11.\n"
 #define MSGTR_VO_XV_ImagedimTooHigh "Размеры изображения источника слишком велики: %ux%u (максимум %ux%u)\n"
 
@@ -1020,7 +1051,7 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_VO_YUV4MPEG_UnknownSubDev "Неизвестное подустройство: %s"
 #define MSGTR_VO_YUV4MPEG_InterlacedTFFMode "Использую режим вывода с чересстрочной развёрткой, верхнее поле первое."
 #define MSGTR_VO_YUV4MPEG_InterlacedBFFMode "Использую режим вывода с чересстрочной развёрткой, нижнее поле первое."
-#define MSGTR_VO_YUV4MPEG_ProgressiveMode "Использую (по умолчанию) режим построчной развёртки кадров."
+#define MSGTR_VO_YUV4MPEG_ProgressiveMode "Использую (по умолчанию) прогрессивній режим кадров."
 
 // vosub_vidix.c
 #define MSGTR_LIBVO_SUB_VIDIX_CantStartPlayback "[VO_SUB_VIDIX] Не могу начать воспроизведение: %s\n"
@@ -1028,10 +1059,10 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_LIBVO_SUB_VIDIX_InterleavedUvForYuv410pNotSupported "[VO_SUB_VIDIX] Слоёный UV для YUV410P не поддерживается.\n"
 #define MSGTR_LIBVO_SUB_VIDIX_DummyVidixdrawsliceWasCalled "[VO_SUB_VIDIX] Был вызван фиктивный vidix_draw_slice().\n"
 #define MSGTR_LIBVO_SUB_VIDIX_UnsupportedFourccForThisVidixDriver "[VO_SUB_VIDIX] Неподдерживаемый FourCC для этого драйвера VIDIX: %x (%s).\n"
-#define MSGTR_LIBVO_SUB_VIDIX_VideoServerHasUnsupportedResolution "[VO_SUB_VIDIX] У сервера видео разрешение (%dx%d) не поддерживается,\nподдерживаемое: %dx%d-%dx%d.\n"
-#define MSGTR_LIBVO_SUB_VIDIX_VideoServerHasUnsupportedColorDepth "[VO_SUB_VIDIX] Сервер видео не поддерживает глубину цвета vidix (%d).\n"
-#define MSGTR_LIBVO_SUB_VIDIX_DriverCantUpscaleImage "[VO_SUB_VIDIX] Драйвер VIDIX не может увеличить изображение (%d%d -> %d%d).\n"
-#define MSGTR_LIBVO_SUB_VIDIX_DriverCantDownscaleImage "[VO_SUB_VIDIX] Драйвер VIDIX не может уменьшить изображение (%d%d -> %d%d).\n"
+#define MSGTR_LIBVO_SUB_VIDIX_VideoServerHasUnsupportedResolution "[VO_SUB_VIDIX] Видео-сервер не поддерживает разрешение (%dx%d), поддерживаемое: %dx%d-%dx%d.\n"
+#define MSGTR_LIBVO_SUB_VIDIX_VideoServerHasUnsupportedColorDepth "[VO_SUB_VIDIX] Видео-сервер не поддерживает глубину цвета vidix (%d).\n"
+#define MSGTR_LIBVO_SUB_VIDIX_DriverCantUpscaleImage "[VO_SUB_VIDIX] Драйвер VIDIX не может увеличить масштаб изображения (%d%d -> %d%d).\n"
+#define MSGTR_LIBVO_SUB_VIDIX_DriverCantDownscaleImage "[VO_SUB_VIDIX] Драйвер VIDIX не может уменьшить масштаб изображения (%d%d -> %d%d).\n"
 #define MSGTR_LIBVO_SUB_VIDIX_CantConfigurePlayback "[VO_SUB_VIDIX] Не могу настроить воспроизведение: %s.\n"
 #define MSGTR_LIBVO_SUB_VIDIX_CouldntFindWorkingVidixDriver "[VO_SUB_VIDIX] Не могу найти работающий драйвер VIDIX.\n"
 #define MSGTR_LIBVO_SUB_VIDIX_CouldntGetCapability "[VO_SUB_VIDIX] Не могу получить возможность: %s.\n"
@@ -1045,27 +1076,27 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_NoBalance "[Микшер] Управление балансом недоступно.\n"
 
 // old vo drivers that have been replaced
-#define MSGTR_VO_PGM_HasBeenReplaced "Драйвер видеовывода pgm был заменён -vo pnm:pgmyuv.\n"
-#define MSGTR_VO_MD5_HasBeenReplaced "Драйвер видеовывода md5 был заменён -vo md5sum.\n"
+#define MSGTR_VO_PGM_HasBeenReplaced "Драйвер видеовывода pgm был заменён на -vo pnm:pgmyuv.\n"
+#define MSGTR_VO_MD5_HasBeenReplaced "Драйвер видеовывода md5 был заменён на -vo md5sum.\n"
 
 
 // ======================= audio output drivers ========================
 
 // audio_out.c
-#define MSGTR_AO_ALSA9_1x_Removed "аудиовывод: модули alsa9 и alsa1x были удалены, используйте -ao alsa взамен.\n"
-#define MSGTR_AO_NoSuchDriver "Отсутствует аудио драйвер '%.*s'\n"
-#define MSGTR_AO_FailedInit "Не могу инициализировать аудио драйвер '%s'\n"
+#define MSGTR_AO_ALSA9_1x_Removed "audio_out: модули alsa9 и alsa1x были удалены, используйте -ao alsa взамен.\n"
+#define MSGTR_AO_NoSuchDriver "Отсутствует аудио-драйвер '%.*s'\n"
+#define MSGTR_AO_FailedInit "Сбой инициализации аудио-драйвер '%s'\n"
 
 // ao_oss.c
-#define MSGTR_AO_OSS_CantOpenMixer "[AO OSS] инициализация аудио: Не могу открыть устройство микшера %s: %s\n"
-#define MSGTR_AO_OSS_ChanNotFound "[AO OSS] инициализация аудио: У микшера аудиокарты отсутствует канал '%s',\nиспользую канал по умолчанию.\n"
-#define MSGTR_AO_OSS_CantOpenDev "[AO OSS] инициализация аудио: Не могу открыть аудиоустройство %s: %s\n"
-#define MSGTR_AO_OSS_CantMakeFd "[AO OSS] инициализация аудио: Не могу заблокировать файловый описатель: %s\n"
+#define MSGTR_AO_OSS_CantOpenMixer "[AO OSS] audio_setup:: Не могу открыть устройство микшера %s: %s\n"
+#define MSGTR_AO_OSS_ChanNotFound "[AO OSS] audio_setup:: У микшера аудиокарты отсутствует канал '%s',\nиспользую канал по умолчанию.\n"
+#define MSGTR_AO_OSS_CantOpenDev "[AO OSS] audio_setup:: Не могу открыть аудиоустройство %s: %s\n"
+#define MSGTR_AO_OSS_CantMakeFd "[AO OSS] audio_setup:: Не могу заблокировать файловый описатель: %s\n"
 #define MSGTR_AO_OSS_CantSet "[AO OSS] Не могу установить аудиоустройство %s в %s вывод, пробую %s...\n"
-#define MSGTR_AO_OSS_CantSetChans "[AO OSS] инициализация аудио: Не могу установить аудиоустройство\nв %d-канальный режим.\n"
-#define MSGTR_AO_OSS_CantUseGetospace "[AO OSS] инициализация аудио: драйвер не поддерживает SNDCTL_DSP_GETOSPACE :-(\n"
+#define MSGTR_AO_OSS_CantSetChans "[AO OSS] audio_setup:: Не могу установить аудиоустройство\nв %d-канальный режим.\n"
+#define MSGTR_AO_OSS_CantUseGetospace "[AO OSS] audio_setup:: драйвер не поддерживает SNDCTL_DSP_GETOSPACE :-(\n"
 #define MSGTR_AO_OSS_CantUseSelect "[AO OSS]\n   ***  Ваш аудиодрайвер НЕ поддерживает select()  ***\n Перекомпилируйте MPlayer с #undef HAVE_AUDIO_SELECT в config.h !\n\n"
-#define MSGTR_AO_OSS_CantReopen "[AO OSS] Фатальная ошибка:\n*** НЕ МОГУ ПОВТОРНО ОТКРЫТЬ / СБРОСИТЬ АУДИОУСТРОЙСТВО (%s) ***\n"
+#define MSGTR_AO_OSS_CantReopen "[AO OSS] Фатальная ошибка:\n*** НЕ МОГУ ПЕРЕОТКРЫТЬ / СБРОСИТЬ АУДИОУСТРОЙСТВО *** %s\n"
 #define MSGTR_AO_OSS_UnknownUnsupportedFormat "[AO OSS] Неизвестный/Неподдерживаемый формат OSS: %x.\n"
 
 // ao_arts.c
@@ -1076,53 +1107,55 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_AO_ARTS_BufferSize "[AO ARTS] размер буфера: %d\n"
 
 // ao_dxr2.c
-#define MSGTR_AO_DXR2_SetVolFailed "[AO DXR2] Не могу установить громкость в %d.\n"
-#define MSGTR_AO_DXR2_UnsupSamplerate "[AO DXR2] dxr2: %d Гц не поддерживается, попробуйте изменить\nчастоту дискретизации.\n"
+#define MSGTR_AO_DXR2_SetVolFailed "[AO DXR2] Не удалось установить громкость в %d.\n"
+#define MSGTR_AO_DXR2_UnsupSamplerate "[AO DXR2] dxr2: %d Гц не поддерживается, попробуйте изменить частоту дискретизации.\n"
 
 // ao_esd.c
-#define MSGTR_AO_ESD_CantOpenSound "[AO ESD] Выполнить esd_open_sound не удалось: %s\n"
+#define MSGTR_AO_ESD_CantOpenSound "[AO ESD] Сбой esd_open_sound: %s\n"
 #define MSGTR_AO_ESD_LatencyInfo "[AO ESD] задержка: [сервер: %0.2fs, сеть: %0.2fs] (подстройка %0.2fs)\n"
-#define MSGTR_AO_ESD_CantOpenPBStream "[AO ESD] не могу открыть поток воспроизведения esd: %s\n"
+#define MSGTR_AO_ESD_CantOpenPBStream "[AO ESD] не удалось открыть поток воспроизведения ESD: %s\n"
 
 // ao_mpegpes.c
-#define MSGTR_AO_MPEGPES_CantSetMixer "[AO MPEGPES] DVB аудио: не могу установить микшер: %s\n"
-#define MSGTR_AO_MPEGPES_UnsupSamplerate "[AO MPEGPES] %d Гц не поддерживается, попробуйте изменить\nчастоту дискретизации.\n"
+#define MSGTR_AO_MPEGPES_CantSetMixer "[AO MPEGPES] Не удалось установить микшер DVB аудио: %s\n"
+#define MSGTR_AO_MPEGPES_UnsupSamplerate "[AO MPEGPES] %d Гц не поддерживается, попробуйте изменить частоту дискретизации.\n"
 
 // ao_pcm.c
 #define MSGTR_AO_PCM_FileInfo "[AO PCM] Файл: %s (%s)\nPCM: Частота дискретизации: %i Гц Каналы: %s Формат %s\n"
-#define MSGTR_AO_PCM_HintInfo "[AO PCM] Информация: наиболее быстрый дамп достигается с -vc null -vo null\n[AO PCM]: Информация: -ao pcm:fast. Для записи WAVE файлов используйте\n[AO PCM]: Информация: -ao pcm:waveheader (по умолчанию).\n"
-#define MSGTR_AO_PCM_CantOpenOutputFile "[AO PCM] Не могу открыть %s для записи!\n"
+#define MSGTR_AO_PCM_HintInfo "[AO PCM] Информация: наиболее быстрый дамп достигается с -benchmark -vc null -vo null -ao pcm:fast.\n[AO PCM]: Для записи WAVE файлов используйте -ao pcm:waveheader (по умолчанию).\n"
+#define MSGTR_AO_PCM_CantOpenOutputFile "[AO PCM] Сбой открытия %s для записи!\n"
 
 // ao_sdl.c
 #define MSGTR_AO_SDL_INFO "[AO SDL] Частота дискретизации: %i Гц Каналы: %s Формат %s\n"
-#define MSGTR_AO_SDL_DriverInfo "[AO SDL] использую %s аудиодрайвер.\n"
-#define MSGTR_AO_SDL_UnsupportedAudioFmt "[AO SDL] Неподдерживаемый аудиоформат: 0x%x.\n"
-#define MSGTR_AO_SDL_CantInit "[AO SDL] Не могу инициализировать SDL аудио: %s\n"
-#define MSGTR_AO_SDL_CantOpenAudio "[AO SDL] Не могу открыть аудио: %s\n"
+#define MSGTR_AO_SDL_DriverInfo "[AO SDL] используется аудио-драйвер %s.\n"
+#define MSGTR_AO_SDL_UnsupportedAudioFmt "[AO SDL] Неподдерживаемый аудио-формат: 0x%x.\n"
+#define MSGTR_AO_SDL_CantInit "[AO SDL] Сбой инициализации SDL Audio: %s\n"
+#define MSGTR_AO_SDL_CantOpenAudio "[AO SDL] Невозможно открыть аудио: %s\n"
 
 // ao_sgi.c
 #define MSGTR_AO_SGI_INFO "[AO SGI] управление.\n"
-#define MSGTR_AO_SGI_InitInfo "[AO SGI] инициализация: Частота дискретизации: %i Гц Каналы: %s Формат %s\n"
-#define MSGTR_AO_SGI_InvalidDevice "[AO SGI] воспроизведение: неверное устройство.\n"
-#define MSGTR_AO_SGI_CantSetParms_Samplerate "[AO SGI] инициализация: ошибка установки параметров: %s\nНе могу установить требуемую частоту дискретизации.\n"
-#define MSGTR_AO_SGI_CantSetAlRate "[AO SGI] инициализация: AL_RATE не доступен на заданном ресурсе.\n"
-#define MSGTR_AO_SGI_CantGetParms "[AO SGI] инициализация: ошибка получения параметров: %s\n"
-#define MSGTR_AO_SGI_SampleRateInfo "[AO SGI] инициализация: частота дискретизации теперь %f (требуемая частота %f)\n"
-#define MSGTR_AO_SGI_InitConfigError "[AO SGI] инициализация: %s\n"
-#define MSGTR_AO_SGI_InitOpenAudioFailed "[AO SGI] инициализация: Не могу открыть канал аудио: %s\n"
-#define MSGTR_AO_SGI_Uninit "[AO SGI] деинициализация: ...\n"
-#define MSGTR_AO_SGI_Reset "[AO SGI] сброс: ...\n"
-#define MSGTR_AO_SGI_PauseInfo "[AO SGI] пауза аудио: ...\n"
-#define MSGTR_AO_SGI_ResumeInfo "[AO SGI] возобновление аудио: ...\n"
+#define MSGTR_AO_SGI_InitInfo "[AO SGI] init: Частота дискретизации: %i Гц Каналы: %s Формат %s\n"
+#define MSGTR_AO_SGI_InvalidDevice "[AO SGI] play: неверное устройство.\n"
+#define MSGTR_AO_SGI_CantSetParms_Samplerate "[AO SGI] init: сбой setparams: %s\nНе могу установить желаемую частоту дискретизации.\n"
+#define MSGTR_AO_SGI_CantSetAlRate "[AO SGI] init: AL_RATE не доступен на заданном ресурсе.\n"
+#define MSGTR_AO_SGI_CantGetParms "[AO SGI] init: сбой getparams: %s\n"
+#define MSGTR_AO_SGI_SampleRateInfo "[AO SGI] init: частота дискретизации теперь %f (желаемая частота %f)\n"
+#define MSGTR_AO_SGI_InitConfigError "[AO SGI] init: %s\n"
+#define MSGTR_AO_SGI_InitOpenAudioFailed "[AO SGI] init: Невозможно открыть аудио-канал: %s\n"
+#define MSGTR_AO_SGI_Uninit "[AO SGI] uninit: ...\n"
+#define MSGTR_AO_SGI_Reset "[AO SGI] reset: ...\n"
+#define MSGTR_AO_SGI_PauseInfo "[AO SGI] audio_pause: ...\n"
+#define MSGTR_AO_SGI_ResumeInfo "[AO SGI] audio_resume: ...\n"
 
 // ao_sun.c
 #define MSGTR_AO_SUN_RtscSetinfoFailed "[AO SUN] rtsc: Выполнить SETINFO не удалось.\n"
 #define MSGTR_AO_SUN_RtscWriteFailed "[AO SUN] rtsc: запись не удалась."
 #define MSGTR_AO_SUN_CantOpenAudioDev "[AO SUN] Не могу открыть аудиоустройство %s, %s -> нет звука.\n"
 #define MSGTR_AO_SUN_UnsupSampleRate "[AO SUN] инициализация аудио: ваша карта не поддерживает канал %d,\n%s, частоту дискретизации %d Гц.\n"
+#define MSGTR_AO_SUN_CantUseSelect "[AO SUN]\n   ***  Ваш аудиодрайвер НЕ поддерживает select()  ***\nПерекомпилируйте MPlayer с #undef HAVE_AUDIO_SELECT в config.h !\n\n"
+#define MSGTR_AO_SUN_CantReopenReset "[AO SUN] Фатальная ошибка:\n*** НЕ МОГУ ПЕРЕОТКРЫТЬ / СБРОСИТЬ АУДИО-УСТРОЙСТВО (%s) ***\n"
 
 // ao_alsa.c
-#define MSGTR_AO_ALSA_InvalidMixerIndexDefaultingToZero "[AO_ALSA] Неверный индекс микшера. Возврат к 0.\n"
+#define MSGTR_AO_ALSA_InvalidMixerIndexDefaultingToZero "[AO_ALSA] Неверный индекс микшера. Возвращаю к 0.\n"
 #define MSGTR_AO_ALSA_MixerOpenError "[AO_ALSA] Ошибка открытия микшера: %s\n"
 #define MSGTR_AO_ALSA_MixerAttachError "[AO_ALSA] Ошибка присоединения %s к микшеру: %s\n"
 #define MSGTR_AO_ALSA_MixerRegisterError "[AO_ALSA] Ошибка регистрации микшера: %s\n"
@@ -1138,38 +1171,38 @@ static const char help_text[] = MSGTR_Help;
 "[AO_ALSA]     Открывает устройство в неблокирующем режиме.\n"\
 "[AO_ALSA]   device=<имя-устройства>\n"\
 "[AO_ALSA]     Устанавливает устройство (замените , на . и : на =)\n"
-#define MSGTR_AO_ALSA_ChannelsNotSupported "[AO_ALSA] Количество каналов %d не поддерживается.\n"
-#define MSGTR_AO_ALSA_OpenInNonblockModeFailed "[AO_ALSA] Ошибка открытия в неблокирующем режиме, попытка открыть в режиме блокирования.\n"
-#define MSGTR_AO_ALSA_PlaybackOpenError "[AO_ALSA] Ошибка начала воспроизведения: %s\n"
+#define MSGTR_AO_ALSA_ChannelsNotSupported "[AO_ALSA] %d каналов не поддерживается.\n"
+#define MSGTR_AO_ALSA_OpenInNonblockModeFailed "[AO_ALSA] Сбой открытия в неблокирующем режиме, попытка открыть в блокирующем режиме.\n"
+#define MSGTR_AO_ALSA_PlaybackOpenError "[AO_ALSA] Ошибка открітия воспроизведения: %s\n"
 #define MSGTR_AO_ALSA_ErrorSetBlockMode "[AL_ALSA] Ошибка установки блокирующего режима: %s.\n"
 #define MSGTR_AO_ALSA_UnableToGetInitialParameters "[AO_ALSA] Невозможно получить начальные параметры: %s\n"
 #define MSGTR_AO_ALSA_UnableToSetAccessType "[AO_ALSA] Невозможно установить тип доступа: %s\n"
 #define MSGTR_AO_ALSA_FormatNotSupportedByHardware "[AO_ALSA] Формат %s не поддерживается оборудованием, пробуем значение по-умолчанию.\n"
 #define MSGTR_AO_ALSA_UnableToSetFormat "[AO_ALSA] Невозможно установить формат: %s\n"
-#define MSGTR_AO_ALSA_UnableToSetChannels "[AO_ALSA] Невозможно установить количество каналов: %s\n"
+#define MSGTR_AO_ALSA_UnableToSetChannels "[AO_ALSA] Невозможно установить каналы: %s\n"
 #define MSGTR_AO_ALSA_UnableToDisableResampling "[AO_ALSA] Невозможно отключить ресэмплинг: %s\n"
 #define MSGTR_AO_ALSA_UnableToSetSamplerate2 "[AO_ALSA] Невозможно установить частоту дискретизации-2: %s\n"
 #define MSGTR_AO_ALSA_UnableToSetBufferTimeNear "[AO_ALSA] Невозможно установить примерное время буфера: %s\n"
-#define MSGTR_AO_ALSA_UnableToGetPeriodSize "[AO ALSA] Невозможно получить время периода: %s\n"
+#define MSGTR_AO_ALSA_UnableToGetPeriodSize "[AO ALSA] Невозможно получить размер периода: %s\n"
 #define MSGTR_AO_ALSA_UnableToSetPeriods "[AO_ALSA] Невозможно установить периоды: %s\n"
 #define MSGTR_AO_ALSA_UnableToSetHwParameters "[AO_ALSA] Невозможно установить hw-параметры: %s\n"
-#define MSGTR_AO_ALSA_UnableToGetBufferSize "[AO_ALSA] Невозможно получить размер буфера: %s\n"
+#define MSGTR_AO_ALSA_UnableToGetBufferSize "[AO_ALSA] Невозможно получить buffersize: %s\n"
 #define MSGTR_AO_ALSA_UnableToGetSwParameters "[AO_ALSA] Невозможно получить sw-параметры: %s\n"
 #define MSGTR_AO_ALSA_UnableToGetBoundary "[AO_ALSA] Невозможно получить границу: %s\n"
 #define MSGTR_AO_ALSA_UnableToSetStartThreshold "[AO_ALSA] Невозможно установить порог запуска: %s\n"
 #define MSGTR_AO_ALSA_UnableToSetStopThreshold "[AO_ALSA] Невозможно установить порог останова: %s\n"
 #define MSGTR_AO_ALSA_UnableToSetSilenceSize "[AO_ALSA] Невозможно установить величину тишины: %s\n"
-#define MSGTR_AO_ALSA_PcmCloseError "[AO_ALSA] ошибка pcm close: %s\n"
+#define MSGTR_AO_ALSA_PcmCloseError "[AO_ALSA] ошибка закрытия pcm: %s\n"
 #define MSGTR_AO_ALSA_NoHandlerDefined "[AO_ALSA] Не определен обработчик!\n"
-#define MSGTR_AO_ALSA_PcmPrepareError "[AO_ALSA] ошибка pcm prepare: %s\n"
-#define MSGTR_AO_ALSA_PcmPauseError "[AO_ALSA] ошибка pcm pause: %s\n"
-#define MSGTR_AO_ALSA_PcmDropError "[AO_ALSA] ошибка pcm drop: %s\n"
-#define MSGTR_AO_ALSA_PcmResumeError "[AO_ALSA] ошибка pcm resume: %s\n"
+#define MSGTR_AO_ALSA_PcmPrepareError "[AO_ALSA] ошибка подготовки pcm: %s\n"
+#define MSGTR_AO_ALSA_PcmPauseError "[AO_ALSA] ошибка паузы pcm: %s\n"
+#define MSGTR_AO_ALSA_PcmDropError "[AO_ALSA] ошибка сброса pcm: %s\n"
+#define MSGTR_AO_ALSA_PcmResumeError "[AO_ALSA] ошибка восстановления pcm: %s\n"
 #define MSGTR_AO_ALSA_DeviceConfigurationError "[AO_ALSA] Ошибка настройки устройства."
-#define MSGTR_AO_ALSA_PcmInSuspendModeTryingResume "[AO_ALSA] Pcm в ждущем режиме, попытка продолжить.\n"
-#define MSGTR_AO_ALSA_WriteError "[AO_ALSA] ошибка записи: %s\n"
-#define MSGTR_AO_ALSA_TryingToResetSoundcard "[AO_ALSA] попытка сбросить звуковую карту.\n"
-#define MSGTR_AO_ALSA_CannotGetPcmStatus "[AO_ALSA] Невозможно получить pcm статус: %s\n"
+#define MSGTR_AO_ALSA_PcmInSuspendModeTryingResume "[AO_ALSA] Pcm в ждущем режиме, попытка восстановления.\n"
+#define MSGTR_AO_ALSA_WriteError "[AO_ALSA] Ошибка записи: %s\n"
+#define MSGTR_AO_ALSA_TryingToResetSoundcard "[AO_ALSA] Попытка сбросить звуковую карту.\n"
+#define MSGTR_AO_ALSA_CannotGetPcmStatus "[AO_ALSA] Невозможно получить состояние pcm: %s\n"
 
 // ao_plugin.c
 
@@ -1185,8 +1218,8 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_AF_LADSPA_ErrNoOutputs "У этого LADSPA плагина отсутствуют выходы аудио."
 #define MSGTR_AF_LADSPA_ErrInOutDiff "Число аудиовходов и аудиовыходов у LADSPA плагина отличается."
 #define MSGTR_AF_LADSPA_ErrFailedToLoad "не могу загрузить"
-#define MSGTR_AF_LADSPA_ErrNoDescriptor "Не могу найти функцию ladspa_descriptor() в указанном файле библиотеки."
-#define MSGTR_AF_LADSPA_ErrLabelNotFound "Не могу найти метку в библиотеке плагина."
+#define MSGTR_AF_LADSPA_ErrNoDescriptor "Невозможно найти функцию ladspa_descriptor() в указанном файле библиотеки."
+#define MSGTR_AF_LADSPA_ErrLabelNotFound "Невзможно найти метку в библиотеке плагина."
 #define MSGTR_AF_LADSPA_ErrNoSuboptions "Не указаны субопции."
 #define MSGTR_AF_LADSPA_ErrNoLibFile "Не указан файл библиотеки."
 #define MSGTR_AF_LADSPA_ErrNoLabel "Не указана метка фильтра."
@@ -1202,13 +1235,13 @@ static const char help_text[] = MSGTR_Help;
 
 // joystick.c
 #define MSGTR_INPUT_JOYSTICK_CantOpen "Не могу отрыть устройство джойстика %s: %s\n"
-#define MSGTR_INPUT_JOYSTICK_ErrReading "Ошибка чтения устройства джойстика: %s\n"
-#define MSGTR_INPUT_JOYSTICK_LoosingBytes "Джойстик: Потеряно %d байт(а/ов) данных\n"
-#define MSGTR_INPUT_JOYSTICK_WarnLostSync "Джойстик: предупреждение о событии инициализации,\nпотеряна синхронизация с драйвером.\n"
-#define MSGTR_INPUT_JOYSTICK_WarnUnknownEvent "Джойстик: предупреждение о неизвестном типе события %d\n"
+#define MSGTR_INPUT_JOYSTICK_ErrReading "Ошибка при чтении устройства джойстика: %s\n"
+#define MSGTR_INPUT_JOYSTICK_LoosingBytes "Joystick: Потеряно %d байт(а/ов) данных\n"
+#define MSGTR_INPUT_JOYSTICK_WarnLostSync "Joystick: предупреждение об init-событии, потеряна синхронизация с драйвером.\n"
+#define MSGTR_INPUT_JOYSTICK_WarnUnknownEvent "Joystick: предупреждение о неизвестном типе события %d\n"
 
 // appleir.c
-#define MSGTR_INPUT_APPLE_IR_CantOpen "Невозможно открыть устройство Apple IR: %s\n"
+#define MSGTR_INPUT_APPLE_IR_CantOpen "Не могу открыть устройство Apple IR: %s\n"
 
 // input.c
 #define MSGTR_INPUT_INPUT_ErrCantRegister2ManyCmdFds "Слишком много описателей файлов команд,\nне могу зарегистрировать файловый описатель %d.\n"
@@ -1236,8 +1269,8 @@ static const char help_text[] = MSGTR_Help;
 #define MSGTR_INPUT_INPUT_ErrCantInitAppleRemote "Не могу инициализировать Пульт ДУ Apple Remote.\n"
 
 // lirc.c
-#define MSGTR_LIRCopenfailed "Неудачное открытие поддержки LIRC.\nВы не сможете использовать Ваш пульт управления.\n"
-#define MSGTR_LIRCcfgerr "Неудачная попытка чтения файла конфигурации LIRC '%s'!\n"
+#define MSGTR_LIRCopenfailed "Сбой открытия поддержки LIRC. Вы не сможете воспозоваться вашим дистанционным управлением.\n"
+#define MSGTR_LIRCcfgerr "Сбой чтения файла конфигурации LIRC '%s'!\n"
 
 
 // ========================== LIBMPDEMUX ===================================
-- 
2.25.3

