From a0d1b5e876059105a9f9bbf928bb1b89cf5a77a5 Mon Sep 17 00:00:00 2001
From: "Vladimir D. Seleznev" <vseleznv@altlinux.org>
Date: Sat, 1 Apr 2017 18:22:49 +0300
Subject: [PATCH 15/28] fix: add missing ld flag

---
 MPlayer/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MPlayer/Makefile b/MPlayer/Makefile
index 1f125f5ff5..92f3061de3 100644
--- a/MPlayer/Makefile
+++ b/MPlayer/Makefile
@@ -1015,7 +1015,7 @@ toolsclean:
 	-rm -f $(call ADD_ALL_EXESUFS,$(TOOLS) $(TOOLS-no))
 	-rm -f TOOLS/realcodecs/*.so.6.0
 
-TOOLS/bmovl-test$(EXESUF): LIBS = -lSDL_image
+TOOLS/bmovl-test$(EXESUF): LIBS = -lSDL -lSDL_image
 TOOLS/vfw2menc$(EXESUF):   LIBS = -lwinmm -lole32
 TOOLS/subrip$(EXESUF):     LIBS = $(MP_MSG_LIBS) -lm
 TOOLS/subrip$(EXESUF): path.o sub/vobsub.o sub/spudec.o sub/unrar_exec.o \
-- 
2.25.3

