From 64190f449296b81b4c92da1f119256ca3aa2ad08 Mon Sep 17 00:00:00 2001
From: Anton Farygin <rider@altlinux.com>
Date: Wed, 20 Jun 2018 19:29:39 +0300
Subject: [PATCH 17/21] compilation fix with glibc 2.27

---
 MPlayer/cpudetect.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MPlayer/cpudetect.c b/MPlayer/cpudetect.c
index d7a8f1a3e2..982a1de780 100644
--- a/MPlayer/cpudetect.c
+++ b/MPlayer/cpudetect.c
@@ -15,6 +15,7 @@
  * with MPlayer; if not, write to the Free Software Foundation, Inc.,
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+#define _GNU_SOURCE
 
 #include "mpx86asm.h"
 #include "config.h"
-- 
2.24.0

