diff -urN ffmpeg-svn-r15375.orig/libavcodec/libx264.c ffmpeg-svn-r15375/libavcodec/libx264.c
--- ffmpeg-svn-r15375.orig/libavcodec/libx264.c	2008-09-22 00:08:24 +0300
+++ ffmpeg-svn-r15375/libavcodec/libx264.c	2008-12-14 04:46:37 +0200
@@ -227,8 +227,10 @@
     x4->params.analyse.i_me_range = avctx->me_range;
     x4->params.analyse.i_subpel_refine = avctx->me_subpel_quality;
 
+#if X264_BUILD < 65
     x4->params.analyse.b_bidir_me = avctx->bidir_refine > 0;
     x4->params.analyse.b_bframe_rdo = avctx->flags2 & CODEC_FLAG2_BRDO;
+#endif
     x4->params.analyse.b_mixed_references =
         avctx->flags2 & CODEC_FLAG2_MIXED_REFS;
     x4->params.analyse.b_chroma_me = avctx->me_cmp & FF_CMP_CHROMA;
