diff -urN MPlayer-1.0rc1.orig/configure MPlayer-1.0rc1/configure
--- MPlayer-1.0rc1.orig/configure	2006-10-23 01:32:31 +0300
+++ MPlayer-1.0rc1/configure	2006-10-23 15:03:52 +0300
@@ -3272,7 +3272,7 @@
 
 
 echocheck "termios"
-if test "$_termios" = auto ; then
+if test "$_termios" != no ; then
   cat > $TMPC <<EOF
 #include <sys/termios.h>
 int main(void) { return 0; }
@@ -4193,6 +4193,8 @@
   _def_aa='#define HAVE_AA 1'
   if cygwin ; then
     _ld_aa=`aalib-config --libs | cut -d " " -f 2,5,6`
+  else
+    _ld_aa=`aalib-config --libs | cut -d " " -f 1,2,4-6`
   fi
   _vosrc="$_vosrc vo_aa.c"
   _vomodules="aa $_vomodules"
@@ -4220,7 +4222,7 @@
 if test "$_caca" = yes ; then
   _def_caca='#define HAVE_CACA 1'
   _inc_extra="$_inc_extra `caca-config --cflags`"
-  _ld_caca=`caca-config --libs`
+  _ld_caca=`caca-config --libs | cut -d " " -f 1-4`
   _vosrc="$_vosrc vo_caca.c"
   _vomodules="caca $_vomodules"
 else
@@ -4910,7 +4912,7 @@
 
 
 echocheck "JACK"
-if test "$_jack" = auto ; then
+if test "$_jack" != no ; then
   _jack=yes
 
 cat > $TMPC << EOF
@@ -4937,7 +4939,7 @@
 echores "$_jack"
 
 echocheck "OpenAL"
-if test "$_openal" = auto ; then
+if test "$_openal" != no ; then
   _openal=no
 cat > $TMPC << EOF
 #include <AL/al.h>
@@ -5398,7 +5400,7 @@
     _res_comment="freetype support needed"
 fi
 echocheck "fontconfig"
-if test "$_fontconfig" = auto ; then
+if test "$_fontconfig" != no ; then
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <fontconfig/fontconfig.h>
@@ -5845,7 +5847,7 @@
 
 
 echocheck "FAAC (AAC encoder) support"
-if test "$_faac" = auto ; then
+if test "$_faac" != no ; then
   cat > $TMPC <<EOF
 #include <inttypes.h> 
 #include <faac.h>
@@ -5945,7 +5947,7 @@
 
 if x86 && not qnx; then
 
-if test "$_win32" = auto ; then
+if test "$_win32" != no ; then
     if test -z "$_win32libdir" ; then
       for I in "$_libdir/codecs" "$_libdir/win32" /usr/local/lib/codecs /usr/lib/codecs /usr/local/lib/win32 /usr/lib/win32 ; do
         if test -d "$I" ; then
@@ -6546,7 +6548,7 @@
 fi
 
 echocheck "x264"
-if test "$_x264" = auto ; then
+if test "$_x264" != no ; then
   cat > $TMPC << EOF
 #include <inttypes.h>
 #include <x264.h>
@@ -6581,7 +6583,7 @@
 
 
 echocheck "nut"
-if test "$_nut" = auto ; then
+if test "$_nut" != no ; then
   cat > $TMPC << EOF
 #include <stdio.h>
 #include <stdlib.h>
@@ -7327,6 +7329,9 @@
 if test "$_vidix_internal" = yes || test "$_vidix_external" = yes ; then
   _vidix=yes
   _def_vidix='#define CONFIG_VIDIX 1'
+  if test "$_vidix_external" = yes ; then
+    _def_vidix_ext='#define VIDIX_EXT 1'
+  fi
 else
   _vidix=no
   _def_vidix='#undef CONFIG_VIDIX'
@@ -7336,7 +7341,7 @@
   _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/mplayer/vidix/\" "
 elif test "$_vidix_external" = yes ; then
   _ld_vidix_external="-lvidix"
-  _def_vidix_pfx='#define MP_VIDIX_PFX "" '
+  _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/vidix/\" "
 fi
 
 if test "$_vidix" = yes; then
@@ -8289,6 +8294,7 @@
 /* enables / disables VIDIX usage */
 $_def_vidix
 $_def_vidix_pfx
+$_def_vidix_ext
 
 /* enables / disables new input joystick support */
 $_def_joystick
