diff -urN mplayer-svn-r19389.orig/configure mplayer-svn-r19389/configure
--- mplayer-svn-r19389.orig/configure	2006-08-13 22:37:15 +0300
+++ mplayer-svn-r19389/configure	2006-08-14 05:29:23 +0300
@@ -5201,6 +5201,8 @@
 fi
 echores "$_dvdnav"
 
+_ld_swscale="-lswscale"
+
 echocheck "cdparanoia"
 if test "$_cdparanoia" = auto ; then
     cat > $TMPC <<EOF
@@ -7373,6 +7375,7 @@
 DVDKIT2 = $_mpdvdkit
 DVDNAV = $_dvdnav
 DVDNAV_LIB = $_ld_dvdnav
+SWSCALE_LIB = $_ld_swscale
 WIN32DLL = $_win32
 W32_DEP = $_dep_win32
 W32_LIB = $_ld_win32
diff -urN mplayer-svn-r19389.orig/Gui/win32/skinload.c mplayer-svn-r19389/Gui/win32/skinload.c
--- mplayer-svn-r19389.orig/Gui/win32/skinload.c	2006-08-13 21:42:16 +0300
+++ mplayer-svn-r19389/Gui/win32/skinload.c	2006-08-14 05:14:08 +0300
@@ -26,8 +26,8 @@
 
 #include <mp_msg.h>
 #include <cpudetect.h>
-#include <libswscale/rgb2rgb.h>
-#include <libswscale/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
+#include <ffmpeg/swscale.h>
 
 #include "gui.h"
 
diff -urN mplayer-svn-r19389.orig/Gui/wm/ws.c mplayer-svn-r19389/Gui/wm/ws.c
--- mplayer-svn-r19389.orig/Gui/wm/ws.c	2006-08-13 21:42:21 +0300
+++ mplayer-svn-r19389/Gui/wm/ws.c	2006-08-14 05:13:51 +0300
@@ -24,8 +24,8 @@
 #include "ws.h"
 #include "wsxdnd.h"
 #include "../cpudetect.h"
-#include "../libswscale/swscale.h"
-#include "../libswscale/rgb2rgb.h"
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
 #include "../libmpcodecs/vf_scale.h"
 #include "../mp_msg.h"
 #include "../help_mp.h"
diff -urN mplayer-svn-r19389.orig/libmpcodecs/vf_halfpack.c mplayer-svn-r19389/libmpcodecs/vf_halfpack.c
--- mplayer-svn-r19389.orig/libmpcodecs/vf_halfpack.c	2006-08-13 21:41:44 +0300
+++ mplayer-svn-r19389/libmpcodecs/vf_halfpack.c	2006-08-14 05:17:17 +0300
@@ -12,7 +12,7 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/rgb2rgb.h"
+#include <ffmpeg/rgb2rgb.h>
 
 struct vf_priv_s {
 	int field;
diff -urN mplayer-svn-r19389.orig/libmpcodecs/vf_palette.c mplayer-svn-r19389/libmpcodecs/vf_palette.c
--- mplayer-svn-r19389.orig/libmpcodecs/vf_palette.c	2006-08-13 21:41:45 +0300
+++ mplayer-svn-r19389/libmpcodecs/vf_palette.c	2006-08-14 05:17:32 +0300
@@ -11,7 +11,7 @@
 #include "mp_image.h"
 #include "vf.h"
 
-#include "libswscale/rgb2rgb.h"
+#include <ffmpeg/rgb2rgb.h>
 
 //===========================================================================//
 
diff -urN mplayer-svn-r19389.orig/libmpcodecs/vf_rgb2bgr.c mplayer-svn-r19389/libmpcodecs/vf_rgb2bgr.c
--- mplayer-svn-r19389.orig/libmpcodecs/vf_rgb2bgr.c	2006-08-13 21:41:45 +0300
+++ mplayer-svn-r19389/libmpcodecs/vf_rgb2bgr.c	2006-08-14 05:17:03 +0300
@@ -11,7 +11,7 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/rgb2rgb.h"
+#include <ffmpeg/rgb2rgb.h>
 
 //===========================================================================//
 
diff -urN mplayer-svn-r19389.orig/libmpcodecs/vf_sab.c mplayer-svn-r19389/libmpcodecs/vf_sab.c
--- mplayer-svn-r19389.orig/libmpcodecs/vf_sab.c	2006-08-13 21:41:44 +0300
+++ mplayer-svn-r19389/libmpcodecs/vf_sab.c	2006-08-14 05:16:48 +0300
@@ -33,7 +33,7 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "vf_scale.h"
 
 
diff -urN mplayer-svn-r19389.orig/libmpcodecs/vf_scale.c mplayer-svn-r19389/libmpcodecs/vf_scale.c
--- mplayer-svn-r19389.orig/libmpcodecs/vf_scale.c	2006-08-13 21:41:45 +0300
+++ mplayer-svn-r19389/libmpcodecs/vf_scale.c	2006-08-14 05:17:24 +0300
@@ -12,7 +12,7 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "vf_scale.h"
 
 #include "m_option.h"
diff -urN mplayer-svn-r19389.orig/libmpcodecs/vf_screenshot.c mplayer-svn-r19389/libmpcodecs/vf_screenshot.c
--- mplayer-svn-r19389.orig/libmpcodecs/vf_screenshot.c	2006-08-13 21:41:44 +0300
+++ mplayer-svn-r19389/libmpcodecs/vf_screenshot.c	2006-08-14 05:16:55 +0300
@@ -22,7 +22,7 @@
 #include "vf_scale.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 
 struct vf_priv_s {
     int frameno;
diff -urN mplayer-svn-r19389.orig/libmpcodecs/vf_smartblur.c mplayer-svn-r19389/libmpcodecs/vf_smartblur.c
--- mplayer-svn-r19389.orig/libmpcodecs/vf_smartblur.c	2006-08-13 21:41:44 +0300
+++ mplayer-svn-r19389/libmpcodecs/vf_smartblur.c	2006-08-14 05:16:41 +0300
@@ -33,7 +33,7 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r19389.orig/libmpcodecs/vf_yuy2.c mplayer-svn-r19389/libmpcodecs/vf_yuy2.c
--- mplayer-svn-r19389.orig/libmpcodecs/vf_yuy2.c	2006-08-13 21:41:45 +0300
+++ mplayer-svn-r19389/libmpcodecs/vf_yuy2.c	2006-08-14 05:17:10 +0300
@@ -12,7 +12,7 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/rgb2rgb.h"
+#include <ffmpeg/rgb2rgb.h>
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r19389.orig/libvo/mga_common.c mplayer-svn-r19389/libvo/mga_common.c
--- mplayer-svn-r19389.orig/libvo/mga_common.c	2006-08-13 21:59:25 +0300
+++ mplayer-svn-r19389/libvo/mga_common.c	2006-08-14 05:16:16 +0300
@@ -1,7 +1,7 @@
 
 #include "fastmemcpy.h"
 #include "cpudetect.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 #include "mp_msg.h"
 #include "help_mp.h"
diff -urN mplayer-svn-r19389.orig/libvo/vo_aa.c mplayer-svn-r19389/libvo/vo_aa.c
--- mplayer-svn-r19389.orig/libvo/vo_aa.c	2006-08-13 21:59:25 +0300
+++ mplayer-svn-r19389/libvo/vo_aa.c	2006-08-14 05:16:32 +0300
@@ -27,7 +27,7 @@
 #include "video_out.h"
 #include "video_out_internal.h"
 #include "aspect.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 #include "font_load.h"
 #include "sub.h"
diff -urN mplayer-svn-r19389.orig/libvo/vo_vesa.c mplayer-svn-r19389/libvo/vo_vesa.c
--- mplayer-svn-r19389.orig/libvo/vo_vesa.c	2006-08-13 21:59:25 +0300
+++ mplayer-svn-r19389/libvo/vo_vesa.c	2006-08-14 05:16:01 +0300
@@ -47,7 +47,7 @@
 #endif
 #include "mp_msg.h"
 
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 
 
diff -urN mplayer-svn-r19389.orig/libvo/vo_x11.c mplayer-svn-r19389/libvo/vo_x11.c
--- mplayer-svn-r19389.orig/libvo/vo_x11.c	2006-08-13 21:59:25 +0300
+++ mplayer-svn-r19389/libvo/vo_x11.c	2006-08-14 05:16:24 +0300
@@ -38,7 +38,7 @@
 #include "fastmemcpy.h"
 #include "sub.h"
 
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 #define MODE_RGB  0x1
 #define MODE_BGR  0x2
diff -urN mplayer-svn-r19389.orig/libvo/vo_yuv4mpeg.c mplayer-svn-r19389/libvo/vo_yuv4mpeg.c
--- mplayer-svn-r19389.orig/libvo/vo_yuv4mpeg.c	2006-08-13 21:59:25 +0300
+++ mplayer-svn-r19389/libvo/vo_yuv4mpeg.c	2006-08-14 05:16:09 +0300
@@ -39,7 +39,7 @@
 #include "sub.h"
 
 #include "fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 
 static vo_info_t info = 
diff -urN mplayer-svn-r19389.orig/Makefile mplayer-svn-r19389/Makefile
--- mplayer-svn-r19389.orig/Makefile	2006-08-13 22:37:15 +0300
+++ mplayer-svn-r19389/Makefile	2006-08-14 05:08:52 +0300
@@ -106,7 +106,7 @@
               libaf/libaf.a \
               libmpdemux/libmpdemux.a \
               stream/stream.a \
-              libswscale/libswscale.a \
+              $(SWSCALE_LIB) \
               osdep/libosdep.a \
               $(DVDREAD_LIB) \
               $(DVDNAV_LIB) \
@@ -146,7 +146,6 @@
         libavcodec \
         libpostproc \
         libavformat \
-        libswscale \
         libao2 \
         osdep \
         input \
@@ -201,7 +200,6 @@
               libmpcodecs/libmpcodecs.a \
               libao2/libao2.a \
               osdep/libosdep.a \
-              libswscale/libswscale.a \
               input/libinput.a \
               libvo/libvo.a \
               libaf/libaf.a \
@@ -301,9 +299,6 @@
 libavformat/libavformat.a:
 	$(MAKE) -C libavformat LIBPREF=lib LIBSUF=.a
 
-libswscale/libswscale.a:
-	$(MAKE) -C libswscale LIBPREF=lib LIBSUF=.a
-
 libmpeg2/libmpeg2.a:
 	$(MAKE) -C libmpeg2
 
@@ -570,7 +565,6 @@
 libavutil/libavutil.a: .norecurse $(wildcard libavutil/*.[ch])
 libavcodec/libavcodec.a: .norecurse $(wildcard libavcodec/*.[ch] libavcodec/*/*.[chS])
 libavformat/libavformat.a: .norecurse $(wildcard libavformat/*.[ch])
-libswscale/libswscale.a: .norecurse $(wildcard libswscale/*.[ch])
 
 libmpeg2/libmpeg2.a: .norecurse $(wildcard libmpeg2/*.[ch])
 liba52/liba52.a: .norecurse $(wildcard liba52/*.[ch])
diff -urN mplayer-svn-r19389.orig/spudec.c mplayer-svn-r19389/spudec.c
--- mplayer-svn-r19389.orig/spudec.c	2006-08-13 22:37:15 +0300
+++ mplayer-svn-r19389/spudec.c	2006-08-14 05:15:51 +0300
@@ -23,7 +23,7 @@
 #include <math.h>
 #include "libvo/video_out.h"
 #include "spudec.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 
 #define MIN(a, b)	((a)<(b)?(a):(b))
 
diff -urN mplayer-svn-r19389.orig/TOOLS/Makefile mplayer-svn-r19389/TOOLS/Makefile
--- mplayer-svn-r19389.orig/TOOLS/Makefile	2006-08-13 22:30:01 +0300
+++ mplayer-svn-r19389/TOOLS/Makefile	2006-08-14 05:10:56 +0300
@@ -27,8 +27,7 @@
 
 subrip: subrip.c
 	$(CC) $(CFLAGS) -g -o $@ $< ../vobsub.o ../spudec.o ../mp_msg.o \
-          ../unrarlib.o ../libswscale/swscale.o ../libswscale/rgb2rgb.o \
-          ../libswscale/yuv2rgb.o ../libmpcodecs/img_format.o -lm
+          ../unrarlib.o ../libmpcodecs/img_format.o -lswscale -lm
 
 vivodump: vivodump.c
 	$(CC) $(CFLAGS) -o $@ $< ../mp_msg.o ../libmpdemux/libmpdemux.a ../stream/stream.a
diff -urN mplayer-svn-r19389.orig/TOOLS/mwallp/compile.sh mplayer-svn-r19389/TOOLS/mwallp/compile.sh
--- mplayer-svn-r19389.orig/TOOLS/mwallp/compile.sh	2006-08-13 22:28:38 +0300
+++ mplayer-svn-r19389/TOOLS/mwallp/compile.sh	2006-08-14 05:15:05 +0300
@@ -1,6 +1,6 @@
 echo
 gcc -g mwallp.c jpeg.c ../../libvo/aclib.o ../../libmpcodecs/img_format.o \
     ../../cpudetect.o ../../mp_msg-mencoder.o ../../libavcodec/libavcodec.a \
-    ../../libswscale/libswscale.a ../../osdep/libosdep.a \
-    -I../../libmpcodecs -I../.. -I../../libswscale \
-    -L/usr/X11/lib -L/usr/X11R6/lib -lX11 -lm -ljpeg -o mwallp
+    ../../osdep/libosdep.a \
+    -I../../libmpcodecs -I../.. \
+    -L/usr/X11/lib -L/usr/X11R6/lib -lswscale -lX11 -lm -ljpeg -o mwallp
diff -urN mplayer-svn-r19389.orig/TOOLS/mwallp/jpeg.c mplayer-svn-r19389/TOOLS/mwallp/jpeg.c
--- mplayer-svn-r19389.orig/TOOLS/mwallp/jpeg.c	2006-08-13 22:28:38 +0300
+++ mplayer-svn-r19389/TOOLS/mwallp/jpeg.c	2006-08-14 05:19:31 +0300
@@ -13,8 +13,8 @@
 
 #include "img_format.h"
 
-#include "swscale.h"
-#include "rgb2rgb.h"
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
 
 static struct SwsContext *swsContext=NULL;
 
diff -urN mplayer-svn-r19389.orig/TOOLS/subrip.c mplayer-svn-r19389/TOOLS/subrip.c
--- mplayer-svn-r19389.orig/TOOLS/subrip.c	2006-08-13 22:30:01 +0300
+++ mplayer-svn-r19389/TOOLS/subrip.c	2006-08-14 05:11:30 +0300
@@ -1,4 +1,4 @@
-/* -*- compile-command: "gcc -g -Wall -I.. -o subrip subrip.c ../vobsub.o ../spudec.o ../mp_msg.o ../unrarlib.o ../libswscale/swscale.o ../libswscale/rgb2rgb.o ../libswscale/yuv2rgb.o ../libmpcodecs/img_format.o -lm" -*- */
+/* -*- compile-command: "gcc -g -Wall -I.. -o subrip subrip.c ../vobsub.o ../spudec.o ../mp_msg.o ../unrarlib.o ../libmpcodecs/img_format.o -lswscale -lm" -*- */
 /*
  * Use with CVS JOCR/GOCR.
  *
