diff -urN mplayer-svn-r22915.orig/configure mplayer-svn-r22915/configure
--- mplayer-svn-r22915.orig/configure	2007-04-05 18:15:55 +0300
+++ mplayer-svn-r22915/configure	2007-04-05 19:26:02 +0300
@@ -292,10 +292,12 @@
   --disable-libavcodec      disable libavcodec [autodetect]
   --disable-libavformat     disable libavformat [autodetect]
   --disable-libpostproc     disable libpostproc [autodetect]
+  --disable-libswscale      disable libswscale [autodetect]
   --disable-libavutil_so    disable shared libavutil [autodetect]
   --disable-libavcodec_so   disable shared libavcodec [autodetect]
   --disable-libavformat_so  disable shared libavformat [autodetect]
   --disable-libpostproc_so  disable shared libpostproc [autodetect]
+  --disable-libswscale_so disable shared libswscale [autodetect]
   --disable-libavcodec_mpegaudio_hp disable high precision audio decoding
                                     in libavcodec [enabled]
   --disable-tremor-internal disable internal Tremor [enabled]
@@ -495,6 +497,8 @@
 _libavformat_so=auto
 _libpostproc=auto
 _libpostproc_so=auto
+_libswscale=auto
+_libswscale_so=auto
 _libavcodec_mpegaudio_hp=yes
 _mencoder=yes
 _mplayer=yes
@@ -998,6 +1002,10 @@
   --disable-libpostproc)	_libpostproc=no		;;
   --enable-libpostproc_so)	_libpostproc_so=yes	;;
   --disable-libpostproc_so)	_libpostproc_so=no	;;
+  --enable-libswscale)		_libswscale=yes		;;
+  --disable-libswscale)		_libswscale=no		;;
+  --enable-libswscale_so)	_libswscale_so=yes	;;
+  --disable-libswscale_so)	_libswscale_so=no	;;
   --enable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=yes	;;
   --disable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=no	;;
 
@@ -6204,6 +6212,15 @@
 fi
 echores "$_libpostproc"
 
+echocheck "FFmpeg libswscale (static)"
+if test "$_libswscale" = auto ; then
+_libswscale=no
+if test -d libswscale && test -f libswscale/swscale.h ; then
+   _libswscale='yes'
+fi
+fi
+echores "$_libswscale"
+
 
 if test "$_libavutil" != yes ; then
 echocheck "FFmpeg libavutil (dynamic)"
@@ -6302,6 +6319,27 @@
 echores "$_libpostproc_so"
 fi #if test "$_libpostproc" != yes ; then
 
+if test "$_libswscale" != yes ; then
+echocheck "FFmpeg libswscale (dynamic)"
+if test "$_libswscale_so" != no ; then
+  _libswscale_so=no
+  cat > $TMPC << EOF
+  #define USE_LIBPOSTPROC 1
+  #include <inttypes.h>
+  #include <ffmpeg/swscale.h>
+  int main(void) {
+      sws_getIdentityVec();
+      return 0;}
+EOF
+  if cc_check -lswscale $_ld_lm ; then
+    _ld_extra="$_ld_extra -lswscale"
+    _libswscale_so=yes
+    _res_comment="using libswscale.so, but static libswscale is recommended"
+  fi
+fi
+echores "$_libswscale_so"
+fi
+
 _def_libavutil='#undef USE_LIBAVUTIL'
 _def_libavutil_so='#undef USE_LIBAVUTIL_SO'
 if test "$_libavutil" = yes ; then
@@ -6355,6 +6393,19 @@
   fi
 fi
 
+_def_libswscale='#undef USE_LIBSWSCALE'
+_def_libswscale_so='#undef USE_LIBSWSCALE_SO'
+if test "$_libswscale" = yes ; then
+  _def_libswscale='#define USE_LIBSWSCALE 1'
+  _ld_libswscale='libswscale/libswscale.a'
+  _dep_libswscale='libswscale/libswscale.a'
+else
+  if test "$_libswscale_so" = yes ; then
+    _def_libswscale_so='#define USE_LIBSWSCALE_SO 1'
+    _ld_libswscale='-lswscale'
+  fi
+fi
+
 echocheck "md5sum support"
 if test "$_md5sum" = yes; then
   _def_md5sum="#define HAVE_MD5SUM"
@@ -7586,6 +7637,10 @@
 CONFIG_LIBAVFORMAT_SO = $_libavformat_so
 CONFIG_LIBPOSTPROC = $_libpostproc
 CONFIG_LIBPOSTPROC_SO = $_libpostproc_so
+CONFIG_LIBSWSCALE = $_libswscale
+CONFIG_SWSCALER = $_libswscale
+CONFIG_LIBSWSCALE_SO = $_libswscale_so
+SWSCALE_LIB = $_ld_libswscale
 ZORAN = $_zr
 LIBLZO = $_liblzo
 LIBDV = $_libdv
@@ -7968,6 +8023,9 @@
 $_def_libpostproc
 $_def_libpostproc_so
 
+$_def_libswscale
+$_def_libswscale_so
+
 /* Win32 DLL support */
 $_def_win32dll
 #define WIN32_PATH "$_win32codecsdir"
diff -urN mplayer-svn-r22915.orig/dvdread/ifo_read.c mplayer-svn-r22915/dvdread/ifo_read.c
--- mplayer-svn-r22915.orig/dvdread/ifo_read.c	2007-04-05 18:12:19 +0300
+++ mplayer-svn-r22915/dvdread/ifo_read.c	2007-04-05 19:26:02 +0300
@@ -29,7 +29,7 @@
 #include <inttypes.h>
 #include <string.h>
 
-#include "bswap.h"
+#include "mpbswap.h"
 #include "ifo_types.h"
 #include "ifo_read.h"
 #include "dvd_reader.h"
diff -urN mplayer-svn-r22915.orig/dvdread/nav_read.c mplayer-svn-r22915/dvdread/nav_read.c
--- mplayer-svn-r22915.orig/dvdread/nav_read.c	2007-04-05 18:12:19 +0300
+++ mplayer-svn-r22915/dvdread/nav_read.c	2007-04-05 19:26:02 +0300
@@ -22,7 +22,7 @@
 #include <string.h>
 #include <inttypes.h>
 
-#include "bswap.h"
+#include "mpbswap.h"
 #include "nav_types.h"
 #include "nav_read.h"
 #include "dvdread_internal.h"
diff -urN mplayer-svn-r22915.orig/Gui/wm/ws.c mplayer-svn-r22915/Gui/wm/ws.c
--- mplayer-svn-r22915.orig/Gui/wm/ws.c	2007-04-05 18:11:10 +0300
+++ mplayer-svn-r22915/Gui/wm/ws.c	2007-04-05 19:26:02 +0300
@@ -24,8 +24,13 @@
 #include "ws.h"
 #include "wsxdnd.h"
 #include "../cpudetect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "../libswscale/swscale.h"
 #include "../libswscale/rgb2rgb.h"
+#endif
 #include "../libmpcodecs/vf_scale.h"
 #include "../mp_msg.h"
 #include "../help_mp.h"
diff -urN mplayer-svn-r22915.orig/liba52/parse.c mplayer-svn-r22915/liba52/parse.c
--- mplayer-svn-r22915.orig/liba52/parse.c	2007-04-05 18:10:49 +0300
+++ mplayer-svn-r22915/liba52/parse.c	2007-04-05 19:26:02 +0300
@@ -37,7 +37,11 @@
 #include "bitstream.h"
 #include "tables.h"
 #include "mm_accel.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "libavutil/avutil.h"
+#endif
 
 #ifdef HAVE_MEMALIGN
 /* some systems have memalign() but no declaration for it */
diff -urN mplayer-svn-r22915.orig/libaf/af_format.c mplayer-svn-r22915/libaf/af_format.c
--- mplayer-svn-r22915.orig/libaf/af_format.c	2007-04-05 18:10:41 +0300
+++ mplayer-svn-r22915/libaf/af_format.c	2007-04-05 19:26:02 +0300
@@ -21,7 +21,11 @@
 #endif
 
 #include "af.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r22915.orig/libao2/ao_dxr2.c mplayer-svn-r22915/libao2/ao_dxr2.c
--- mplayer-svn-r22915.orig/libao2/ao_dxr2.c	2007-04-05 18:12:04 +0300
+++ mplayer-svn-r22915/libao2/ao_dxr2.c	2007-04-05 19:26:02 +0300
@@ -7,7 +7,11 @@
 #include "config.h"
 #include "mp_msg.h"
 #include "help_mp.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "audio_out.h"
diff -urN mplayer-svn-r22915.orig/libao2/ao_pcm.c mplayer-svn-r22915/libao2/ao_pcm.c
--- mplayer-svn-r22915.orig/libao2/ao_pcm.c	2007-04-05 18:12:05 +0300
+++ mplayer-svn-r22915/libao2/ao_pcm.c	2007-04-05 19:26:02 +0300
@@ -4,7 +4,11 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "subopt-helper.h"
 #include "libaf/af_format.h"
diff -urN mplayer-svn-r22915.orig/libass/mputils.h mplayer-svn-r22915/libass/mputils.h
--- mplayer-svn-r22915.orig/libass/mputils.h	2007-04-05 18:12:05 +0300
+++ mplayer-svn-r22915/libass/mputils.h	2007-04-05 19:26:02 +0300
@@ -6,6 +6,10 @@
 #include "libvo/font_load.h" // for blur()
 #include "subreader.h" // for guess_buffer_cp
 #include "libvo/sub.h" // for utf8_get_char
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #endif
diff -urN mplayer-svn-r22915.orig/libdha/sysdep/pci_linux.c mplayer-svn-r22915/libdha/sysdep/pci_linux.c
--- mplayer-svn-r22915.orig/libdha/sysdep/pci_linux.c	2007-04-05 18:10:58 +0300
+++ mplayer-svn-r22915/libdha/sysdep/pci_linux.c	2007-04-05 19:26:02 +0300
@@ -132,7 +132,11 @@
 #include <fcntl.h>
 //#include <sys/io.h>
 #include <linux/pci.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 static int pci_get_vendor(
diff -urN mplayer-svn-r22915.orig/libmpcodecs/ad_dk3adpcm.c mplayer-svn-r22915/libmpcodecs/ad_dk3adpcm.c
--- mplayer-svn-r22915.orig/libmpcodecs/ad_dk3adpcm.c	2007-04-05 18:12:06 +0300
+++ mplayer-svn-r22915/libmpcodecs/ad_dk3adpcm.c	2007-04-05 19:26:02 +0300
@@ -16,7 +16,11 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "ad_internal.h"
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/ad_imaadpcm.c mplayer-svn-r22915/libmpcodecs/ad_imaadpcm.c
--- mplayer-svn-r22915.orig/libmpcodecs/ad_imaadpcm.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/ad_imaadpcm.c	2007-04-05 19:26:02 +0300
@@ -21,7 +21,11 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "ad_internal.h"
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/ad_msadpcm.c mplayer-svn-r22915/libmpcodecs/ad_msadpcm.c
--- mplayer-svn-r22915.orig/libmpcodecs/ad_msadpcm.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/ad_msadpcm.c	2007-04-05 19:26:02 +0300
@@ -12,7 +12,11 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "ad_internal.h"
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/ad_qtaudio.c mplayer-svn-r22915/libmpcodecs/ad_qtaudio.c
--- mplayer-svn-r22915.orig/libmpcodecs/ad_qtaudio.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/ad_qtaudio.c	2007-04-05 19:26:02 +0300
@@ -14,7 +14,13 @@
 #include "loader/wine/windef.h"
 
 #include "ad_internal.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/common.h"
+#include "libavutil/bswap.h"
+#endif
 
 #ifdef WIN32_LOADER
 #include "loader/ldt_keeper.h"
diff -urN mplayer-svn-r22915.orig/libmpcodecs/native/nuppelvideo.c mplayer-svn-r22915/libmpcodecs/native/nuppelvideo.c
--- mplayer-svn-r22915.orig/libmpcodecs/native/nuppelvideo.c	2007-04-05 18:12:17 +0300
+++ mplayer-svn-r22915/libmpcodecs/native/nuppelvideo.c	2007-04-05 19:26:02 +0300
@@ -13,7 +13,11 @@
 
 #include "config.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "../libvo/fastmemcpy.h"
diff -urN mplayer-svn-r22915.orig/libmpcodecs/native/RTjpegN.c mplayer-svn-r22915/libmpcodecs/native/RTjpegN.c
--- mplayer-svn-r22915.orig/libmpcodecs/native/RTjpegN.c	2007-04-05 18:12:17 +0300
+++ mplayer-svn-r22915/libmpcodecs/native/RTjpegN.c	2007-04-05 19:26:02 +0300
@@ -30,7 +30,11 @@
 #define MMX
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "RTjpegN.h"
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vd_ffmpeg.c mplayer-svn-r22915/libmpcodecs/vd_ffmpeg.c
--- mplayer-svn-r22915.orig/libmpcodecs/vd_ffmpeg.c	2007-04-05 18:12:17 +0300
+++ mplayer-svn-r22915/libmpcodecs/vd_ffmpeg.c	2007-04-05 19:26:02 +0300
@@ -7,8 +7,15 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/avutil.h>
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/common.h"
+#include "libavutil/avutil.h"
 #include "libavutil/intreadwrite.h"
+#endif
 #include "mpbswap.h"
 
 #include "vd_internal.h"
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vd_ijpg.c mplayer-svn-r22915/libmpcodecs/vd_ijpg.c
--- mplayer-svn-r22915.orig/libmpcodecs/vd_ijpg.c	2007-04-05 18:12:16 +0300
+++ mplayer-svn-r22915/libmpcodecs/vd_ijpg.c	2007-04-05 19:26:02 +0300
@@ -11,7 +11,11 @@
 
 #include <setjmp.h>
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vd_mpng.c mplayer-svn-r22915/libmpcodecs/vd_mpng.c
--- mplayer-svn-r22915.orig/libmpcodecs/vd_mpng.c	2007-04-05 18:12:17 +0300
+++ mplayer-svn-r22915/libmpcodecs/vd_mpng.c	2007-04-05 19:26:02 +0300
@@ -6,7 +6,11 @@
 
 #include <png.h>
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vd_mtga.c mplayer-svn-r22915/libmpcodecs/vd_mtga.c
--- mplayer-svn-r22915.orig/libmpcodecs/vd_mtga.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/vd_mtga.c	2007-04-05 19:26:02 +0300
@@ -10,7 +10,11 @@
 #include "config.h"
 #include "mp_msg.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vd_qtvideo.c mplayer-svn-r22915/libmpcodecs/vd_qtvideo.c
--- mplayer-svn-r22915.orig/libmpcodecs/vd_qtvideo.c	2007-04-05 18:12:17 +0300
+++ mplayer-svn-r22915/libmpcodecs/vd_qtvideo.c	2007-04-05 19:26:02 +0300
@@ -28,7 +28,11 @@
 
 LIBVD_EXTERN(qtvideo)
 
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 
 #ifndef MACOSX
 #include "loader/qtx/qtxsdk/components.h"
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vd_sgi.c mplayer-svn-r22915/libmpcodecs/vd_sgi.c
--- mplayer-svn-r22915.orig/libmpcodecs/vd_sgi.c	2007-04-05 18:12:16 +0300
+++ mplayer-svn-r22915/libmpcodecs/vd_sgi.c	2007-04-05 19:26:02 +0300
@@ -7,7 +7,11 @@
 
 #include "config.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "vd_internal.h"
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vd_xanim.c mplayer-svn-r22915/libmpcodecs/vd_xanim.c
--- mplayer-svn-r22915.orig/libmpcodecs/vd_xanim.c	2007-04-05 18:15:24 +0300
+++ mplayer-svn-r22915/libmpcodecs/vd_xanim.c	2007-04-05 19:26:02 +0300
@@ -37,7 +37,11 @@
 #include <errno.h> /* strerror, errno */
 
 #include "mp_msg.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 
 #include "osdep/timer.h"
 #include "libvo/fastmemcpy.h"
diff -urN mplayer-svn-r22915.orig/libmpcodecs/ve_qtvideo.c mplayer-svn-r22915/libmpcodecs/ve_qtvideo.c
--- mplayer-svn-r22915.orig/libmpcodecs/ve_qtvideo.c	2007-04-05 18:12:16 +0300
+++ mplayer-svn-r22915/libmpcodecs/ve_qtvideo.c	2007-04-05 19:26:02 +0300
@@ -10,7 +10,11 @@
 
 #include "config.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #ifdef WIN32_LOADER 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_bmovl.c mplayer-svn-r22915/libmpcodecs/vf_bmovl.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_bmovl.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_bmovl.c	2007-04-05 19:26:02 +0300
@@ -69,7 +69,11 @@
 #include "img_format.h"
 
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_divtc.c mplayer-svn-r22915/libmpcodecs/vf_divtc.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_divtc.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_divtc.c	2007-04-05 19:26:02 +0300
@@ -7,7 +7,11 @@
 #include "config.h"
 #include "mp_msg.h"
 #include "cpudetect.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "img_format.h"
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_geq.c mplayer-svn-r22915/libmpcodecs/vf_geq.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_geq.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_geq.c	2007-04-05 19:26:02 +0300
@@ -30,10 +30,20 @@
 // Needed to bring in lrintf.
 #define HAVE_AV_CONFIG_H
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#include <ffmpeg/dsputil.h>
+#include <ffmpeg/eval.h>
+#else
 #include "libavcodec/avcodec.h"
 #include "libavcodec/dsputil.h"
 #include "libavcodec/eval.h"
+#endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 /* FIXME: common.h defines printf away when HAVE_AV_CONFIG
  * is defined, but mp_image.h needs printf.
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_halfpack.c mplayer-svn-r22915/libmpcodecs/vf_halfpack.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_halfpack.c	2007-04-05 18:12:16 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_halfpack.c	2007-04-05 19:26:02 +0300
@@ -12,7 +12,12 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
+
 
 struct vf_priv_s {
 	int field;
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_palette.c mplayer-svn-r22915/libmpcodecs/vf_palette.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_palette.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_palette.c	2007-04-05 19:26:02 +0300
@@ -11,7 +11,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 
 //===========================================================================//
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_qp.c mplayer-svn-r22915/libmpcodecs/vf_qp.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_qp.c	2007-04-05 18:12:16 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_qp.c	2007-04-05 19:26:02 +0300
@@ -37,9 +37,18 @@
 // Needed to bring in lrintf.
 #define HAVE_AV_CONFIG_H
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#include <ffmpeg/dsputil.h>
+#else
 #include "libavcodec/avcodec.h"
 #include "libavcodec/dsputil.h"
+#endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 /* FIXME: common.h defines printf away when HAVE_AV_CONFIG
  * is defined, but mp_image.h needs printf.
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_rectangle.c mplayer-svn-r22915/libmpcodecs/vf_rectangle.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_rectangle.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_rectangle.c	2007-04-05 19:26:02 +0300
@@ -6,7 +6,11 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 struct vf_priv_s {
     int x, y, w, h;
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_rgb2bgr.c mplayer-svn-r22915/libmpcodecs/vf_rgb2bgr.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_rgb2bgr.c	2007-04-05 18:12:16 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_rgb2bgr.c	2007-04-05 19:26:02 +0300
@@ -11,7 +11,12 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
+
 
 //===========================================================================//
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_sab.c mplayer-svn-r22915/libmpcodecs/vf_sab.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_sab.c	2007-04-05 18:12:16 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_sab.c	2007-04-05 19:26:02 +0300
@@ -38,7 +38,11 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_scale.c mplayer-svn-r22915/libmpcodecs/vf_scale.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_scale.c	2007-04-05 18:12:16 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_scale.c	2007-04-05 19:26:02 +0300
@@ -11,10 +11,14 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "fmt-conversion.h"
-#include "bswap.h"
+#include "mpbswap.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 #include "m_option.h"
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_screenshot.c mplayer-svn-r22915/libmpcodecs/vf_screenshot.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_screenshot.c	2007-04-05 18:12:15 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_screenshot.c	2007-04-05 19:26:02 +0300
@@ -22,7 +22,11 @@
 #include "vf_scale.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 
 struct vf_priv_s {
     int frameno;
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_smartblur.c mplayer-svn-r22915/libmpcodecs/vf_smartblur.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_smartblur.c	2007-04-05 18:12:15 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_smartblur.c	2007-04-05 19:26:02 +0300
@@ -38,7 +38,11 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_unsharp.c mplayer-svn-r22915/libmpcodecs/vf_unsharp.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_unsharp.c	2007-04-05 18:12:18 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_unsharp.c	2007-04-05 19:26:02 +0300
@@ -35,7 +35,11 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 //===========================================================================//
 
diff -urN mplayer-svn-r22915.orig/libmpcodecs/vf_yuy2.c mplayer-svn-r22915/libmpcodecs/vf_yuy2.c
--- mplayer-svn-r22915.orig/libmpcodecs/vf_yuy2.c	2007-04-05 18:12:15 +0300
+++ mplayer-svn-r22915/libmpcodecs/vf_yuy2.c	2007-04-05 19:26:02 +0300
@@ -12,7 +12,11 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r22915.orig/libmpdemux/asf.h mplayer-svn-r22915/libmpdemux/asf.h
--- mplayer-svn-r22915.orig/libmpdemux/asf.h	2007-04-05 18:12:01 +0300
+++ mplayer-svn-r22915/libmpdemux/asf.h	2007-04-05 19:26:02 +0300
@@ -3,7 +3,11 @@
 
 //#include "config.h"	/* for WORDS_BIGENDIAN */
 #include <inttypes.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 ///////////////////////
diff -urN mplayer-svn-r22915.orig/libmpdemux/asfheader.c mplayer-svn-r22915/libmpdemux/asfheader.c
--- mplayer-svn-r22915.orig/libmpdemux/asfheader.c	2007-04-05 18:12:02 +0300
+++ mplayer-svn-r22915/libmpdemux/asfheader.c	2007-04-05 19:26:02 +0300
@@ -6,8 +6,13 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/common.h"
 #include "libavutil/intreadwrite.h"
+#endif
 #include "mp_msg.h"
 #include "help_mp.h"
 
diff -urN mplayer-svn-r22915.orig/libmpdemux/aviheader.c mplayer-svn-r22915/libmpdemux/aviheader.c
--- mplayer-svn-r22915.orig/libmpdemux/aviheader.c	2007-04-05 18:12:01 +0300
+++ mplayer-svn-r22915/libmpdemux/aviheader.c	2007-04-05 19:26:02 +0300
@@ -13,7 +13,11 @@
 #include "stheader.h"
 
 #include "aviheader.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 static MainAVIHeader avih;
 
diff -urN mplayer-svn-r22915.orig/libmpdemux/aviheader.h mplayer-svn-r22915/libmpdemux/aviheader.h
--- mplayer-svn-r22915.orig/libmpdemux/aviheader.h	2007-04-05 18:12:02 +0300
+++ mplayer-svn-r22915/libmpdemux/aviheader.h	2007-04-05 19:26:02 +0300
@@ -1,8 +1,12 @@
 #ifndef _aviheader_h
 #define	_aviheader_h
 
-//#include "config.h"	/* get correct definition WORDS_BIGENDIAN */
+#include "config.h"	/* get correct definition WORDS_BIGENDIAN */
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #ifndef mmioFOURCC
diff -urN mplayer-svn-r22915.orig/libmpdemux/demux_lavf.c mplayer-svn-r22915/libmpdemux/demux_lavf.c
--- mplayer-svn-r22915.orig/libmpdemux/demux_lavf.c	2007-04-05 18:15:55 +0300
+++ mplayer-svn-r22915/libmpdemux/demux_lavf.c	2007-04-05 19:26:02 +0300
@@ -40,7 +40,11 @@
 #include "avi.h"
 #include "opt.h"
 #endif
-#include "riff.h"
+#ifdef USE_LIBAVFORMAT_SO
+#include <ffmpeg/riff.h>
+#else
+#include "libavformat/riff.h"
+#endif
 
 #define PROBE_BUF_SIZE 2048
 
diff -urN mplayer-svn-r22915.orig/libmpdemux/demux_mkv.c mplayer-svn-r22915/libmpdemux/demux_mkv.c
--- mplayer-svn-r22915.orig/libmpdemux/demux_mkv.c	2007-04-05 18:12:01 +0300
+++ mplayer-svn-r22915/libmpdemux/demux_mkv.c	2007-04-05 19:26:02 +0300
@@ -28,7 +28,11 @@
 #include "libass/ass.h"
 #include "libass/ass_mp.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #ifdef USE_QTX_CODECS
 #include "loader/qtx/qtxsdk/components.h"
diff -urN mplayer-svn-r22915.orig/libmpdemux/demux_mpc.c mplayer-svn-r22915/libmpdemux/demux_mpc.c
--- mplayer-svn-r22915.orig/libmpdemux/demux_mpc.c	2007-04-05 18:12:02 +0300
+++ mplayer-svn-r22915/libmpdemux/demux_mpc.c	2007-04-05 19:26:02 +0300
@@ -11,7 +11,11 @@
 #include <stdio.h>
 #include <string.h>
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "stream/stream.h"
 #include "demuxer.h"
diff -urN mplayer-svn-r22915.orig/libmpdemux/demux_rtp_codec.cpp mplayer-svn-r22915/libmpdemux/demux_rtp_codec.cpp
--- mplayer-svn-r22915.orig/libmpdemux/demux_rtp_codec.cpp	2007-04-05 18:15:55 +0300
+++ mplayer-svn-r22915/libmpdemux/demux_rtp_codec.cpp	2007-04-05 19:37:48 +0300
@@ -6,7 +6,11 @@
 #include <limits.h>
 #include <math.h>
 #include "stheader.h"
-#include "base64.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/base64.h>
+#else
+#include "libavutil/base64.h"
+#endif
 }
 
 #ifdef USE_LIBAVCODEC
diff -urN mplayer-svn-r22915.orig/libmpdemux/demux_vqf.c mplayer-svn-r22915/libmpdemux/demux_vqf.c
--- mplayer-svn-r22915.orig/libmpdemux/demux_vqf.c	2007-04-05 18:12:02 +0300
+++ mplayer-svn-r22915/libmpdemux/demux_vqf.c	2007-04-05 19:26:02 +0300
@@ -2,7 +2,11 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "stream/stream.h"
diff -urN mplayer-svn-r22915.orig/libmpdemux/ebml.c mplayer-svn-r22915/libmpdemux/ebml.c
--- mplayer-svn-r22915.orig/libmpdemux/ebml.c	2007-04-05 18:12:01 +0300
+++ mplayer-svn-r22915/libmpdemux/ebml.c	2007-04-05 19:26:02 +0300
@@ -11,9 +11,17 @@
 
 #include "stream/stream.h"
 #include "ebml.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intfloat_readwrite.h>
+#else
 #include "libavutil/intfloat_readwrite.h"
+#endif
 
 
 #ifndef SIZE_MAX
diff -urN mplayer-svn-r22915.orig/libmpdemux/nuppelvideo.h mplayer-svn-r22915/libmpdemux/nuppelvideo.h
--- mplayer-svn-r22915.orig/libmpdemux/nuppelvideo.h	2007-04-05 18:12:02 +0300
+++ mplayer-svn-r22915/libmpdemux/nuppelvideo.h	2007-04-05 19:26:02 +0300
@@ -1,6 +1,10 @@
 /* nuppelvideo.h  rh */
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intfloat_readwrite.h>
+#else
 #include "libavutil/intfloat_readwrite.h"
+#endif
 
 typedef struct  __attribute__((packed)) rtfileheader
 {
diff -urN mplayer-svn-r22915.orig/libswscale/rgb2rgb.c mplayer-svn-r22915/libswscale/rgb2rgb.c
--- mplayer-svn-r22915.orig/libswscale/rgb2rgb.c	2007-04-05 18:11:00 +0300
+++ mplayer-svn-r22915/libswscale/rgb2rgb.c	2007-04-05 19:26:02 +0300
@@ -32,7 +32,11 @@
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "x86_cpu.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 #ifdef USE_FASTMEMCPY
 #include "libvo/fastmemcpy.h"
 #endif
diff -urN mplayer-svn-r22915.orig/libswscale/swscale.c mplayer-svn-r22915/libswscale/swscale.c
--- mplayer-svn-r22915.orig/libswscale/swscale.c	2007-04-05 18:15:54 +0300
+++ mplayer-svn-r22915/libswscale/swscale.c	2007-04-05 19:26:02 +0300
@@ -70,7 +70,11 @@
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "x86_cpu.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 #include "rgb2rgb.h"
 #ifdef USE_FASTMEMCPY
 #include "libvo/fastmemcpy.h"
diff -urN mplayer-svn-r22915.orig/libswscale/swscale-example.c mplayer-svn-r22915/libswscale/swscale-example.c
--- mplayer-svn-r22915.orig/libswscale/swscale-example.c	2007-04-05 18:11:00 +0300
+++ mplayer-svn-r22915/libswscale/swscale-example.c	2007-04-05 19:26:02 +0300
@@ -25,7 +25,11 @@
 #include <stdarg.h>
 
 #undef HAVE_AV_CONFIG_H
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "rgb2rgb.h"
diff -urN mplayer-svn-r22915.orig/libswscale/swscale_internal.h mplayer-svn-r22915/libswscale/swscale_internal.h
--- mplayer-svn-r22915.orig/libswscale/swscale_internal.h	2007-04-05 18:10:59 +0300
+++ mplayer-svn-r22915/libswscale/swscale_internal.h	2007-04-05 19:26:02 +0300
@@ -25,7 +25,11 @@
 #include <altivec.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 
 #ifdef CONFIG_DARWIN
 #define AVV(x...) (x)
diff -urN mplayer-svn-r22915.orig/libvo/font_load_ft.c mplayer-svn-r22915/libvo/font_load_ft.c
--- mplayer-svn-r22915.orig/libvo/font_load_ft.c	2007-04-05 18:12:24 +0300
+++ mplayer-svn-r22915/libvo/font_load_ft.c	2007-04-05 19:26:02 +0300
@@ -28,7 +28,11 @@
 #include <fontconfig/fontconfig.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "font_load.h"
 #include "mp_msg.h"
diff -urN mplayer-svn-r22915.orig/libvo/mga_common.c mplayer-svn-r22915/libvo/mga_common.c
--- mplayer-svn-r22915.orig/libvo/mga_common.c	2007-04-05 18:12:21 +0300
+++ mplayer-svn-r22915/libvo/mga_common.c	2007-04-05 19:26:02 +0300
@@ -1,8 +1,12 @@
 
 #include "fastmemcpy.h"
 #include "cpudetect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #include "mp_msg.h"
 #include "help_mp.h"
diff -urN mplayer-svn-r22915.orig/libvo/sub.c mplayer-svn-r22915/libvo/sub.c
--- mplayer-svn-r22915.orig/libvo/sub.c	2007-04-05 18:12:24 +0300
+++ mplayer-svn-r22915/libvo/sub.c	2007-04-05 19:26:02 +0300
@@ -20,7 +20,12 @@
 #include "font_load.h"
 #include "sub.h"
 #include "spudec.h"
+
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #define NEW_SPLITTING
 
diff -urN mplayer-svn-r22915.orig/libvo/vo_aa.c mplayer-svn-r22915/libvo/vo_aa.c
--- mplayer-svn-r22915.orig/libvo/vo_aa.c	2007-04-05 18:15:51 +0300
+++ mplayer-svn-r22915/libvo/vo_aa.c	2007-04-05 19:26:02 +0300
@@ -27,7 +27,11 @@
 #include "video_out.h"
 #include "video_out_internal.h"
 #include "aspect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #include "font_load.h"
 #include "sub.h"
diff -urN mplayer-svn-r22915.orig/libvo/vo_vesa.c mplayer-svn-r22915/libvo/vo_vesa.c
--- mplayer-svn-r22915.orig/libvo/vo_vesa.c	2007-04-05 18:12:24 +0300
+++ mplayer-svn-r22915/libvo/vo_vesa.c	2007-04-05 19:26:02 +0300
@@ -39,7 +39,11 @@
 
 #include "fastmemcpy.h"
 #include "sub.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "aspect.h"
 #include "vesa_lvo.h"
@@ -48,7 +52,11 @@
 #endif
 #include "mp_msg.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 
 
diff -urN mplayer-svn-r22915.orig/libvo/vo_x11.c mplayer-svn-r22915/libvo/vo_x11.c
--- mplayer-svn-r22915.orig/libvo/vo_x11.c	2007-04-05 18:15:51 +0300
+++ mplayer-svn-r22915/libvo/vo_x11.c	2007-04-05 19:26:02 +0300
@@ -38,7 +38,11 @@
 #include "fastmemcpy.h"
 #include "sub.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #define MODE_RGB  0x1
 #define MODE_BGR  0x2
diff -urN mplayer-svn-r22915.orig/libvo/vo_xv.c mplayer-svn-r22915/libvo/vo_xv.c
--- mplayer-svn-r22915.orig/libvo/vo_xv.c	2007-04-05 18:12:21 +0300
+++ mplayer-svn-r22915/libvo/vo_xv.c	2007-04-05 19:26:02 +0300
@@ -43,7 +43,11 @@
 #include "Gui/interface.h"
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 static vo_info_t info = {
     "X11/Xv",
diff -urN mplayer-svn-r22915.orig/libvo/vo_xvmc.c mplayer-svn-r22915/libvo/vo_xvmc.c
--- mplayer-svn-r22915.orig/libvo/vo_xvmc.c	2007-04-05 18:12:24 +0300
+++ mplayer-svn-r22915/libvo/vo_xvmc.c	2007-04-05 19:26:51 +0300
@@ -36,7 +36,11 @@
 #include "Gui/interface.h"
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 //no chanse xinerama to be suported in near future
 #undef HAVE_XINERAMA
@@ -47,7 +51,11 @@
 
 #define UNUSED(x) ((void)(x))
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#else
 #include "libavcodec/avcodec.h"
+#endif
 #if LIBAVCODEC_BUILD < ((51<<16)+(40<<8)+2)
 #error You need at least libavcodecs v51.40.2
 #endif
diff -urN mplayer-svn-r22915.orig/libvo/vo_yuv4mpeg.c mplayer-svn-r22915/libvo/vo_yuv4mpeg.c
--- mplayer-svn-r22915.orig/libvo/vo_yuv4mpeg.c	2007-04-05 18:12:23 +0300
+++ mplayer-svn-r22915/libvo/vo_yuv4mpeg.c	2007-04-05 19:26:02 +0300
@@ -40,10 +40,19 @@
 #include "sub.h"
 
 #include "fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/swscale.h"
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/rational.h>
+#else
 #include "libavutil/rational.h"
+#endif
 
 static vo_info_t info = 
 {
diff -urN mplayer-svn-r22915.orig/Makefile mplayer-svn-r22915/Makefile
--- mplayer-svn-r22915.orig/Makefile	2007-04-05 18:15:55 +0300
+++ mplayer-svn-r22915/Makefile	2007-04-05 19:26:02 +0300
@@ -57,13 +57,15 @@
               libaf/libaf.a \
               libmpdemux/libmpdemux.a \
               stream/stream.a \
-              libswscale/libswscale.a \
+              $(SWSCALE_LIB) \
               libvo/libosd.a \
 
 COMMON_LIBS-$(CONFIG_LIBAVFORMAT) += libavformat/libavformat.a
 COMMON_LIBS-$(CONFIG_LIBAVCODEC)  += libavcodec/libavcodec.a
 COMMON_LIBS-$(CONFIG_LIBAVUTIL)   += libavutil/libavutil.a
 COMMON_LIBS-$(CONFIG_LIBPOSTPROC) += libpostproc/libpostproc.a
+COMMON_LIBS-$(CONFIG_LIBSWSCALE)  += libswscale/libswscale.a
+PARTS-$(CONFIG_LIBSWSCALE)        += libswscale
 COMMON_LIBS-$(WIN32DLL)           += loader/libloader.a
 COMMON_LIBS-$(MP3LIB)             += mp3lib/libmp3.a
 COMMON_LIBS-$(LIBA52)             += liba52/liba52.a
diff -urN mplayer-svn-r22915.orig/mp3lib/sr1.c mplayer-svn-r22915/mp3lib/sr1.c
--- mplayer-svn-r22915.orig/mp3lib/sr1.c	2007-04-05 18:10:51 +0300
+++ mplayer-svn-r22915/mp3lib/sr1.c	2007-04-05 19:26:02 +0300
@@ -24,7 +24,11 @@
 #include "mpg123.h"
 #include "huffman.h"
 #include "mp3.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "cpudetect.h"
 //#include "liba52/mm_accel.h"
diff -urN mplayer-svn-r22915.orig/mpbswap.h mplayer-svn-r22915/mpbswap.h
--- mplayer-svn-r22915.orig/mpbswap.h	2007-04-05 18:11:32 +0300
+++ mplayer-svn-r22915/mpbswap.h	2007-04-05 19:26:02 +0300
@@ -1,6 +1,11 @@
 #ifndef MPBSWAP_H
 #define MPBSWAP_H
 
+#include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
 #include "libavutil/bswap.h"
+#endif
 
 #endif /* MPBSWAP_H */
diff -urN mplayer-svn-r22915.orig/spudec.c mplayer-svn-r22915/spudec.c
--- mplayer-svn-r22915.orig/spudec.c	2007-04-05 18:11:25 +0300
+++ mplayer-svn-r22915/spudec.c	2007-04-05 19:26:02 +0300
@@ -28,7 +28,11 @@
 #else
 #include "avutil.h"
 #endif
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 
 /* Valid values for spu_aamode:
    0: none (fastest, most ugly)
diff -urN mplayer-svn-r22915.orig/stream/realrtsp/real.c mplayer-svn-r22915/stream/realrtsp/real.c
--- mplayer-svn-r22915.orig/stream/realrtsp/real.c	2007-04-05 18:15:24 +0300
+++ mplayer-svn-r22915/stream/realrtsp/real.c	2007-04-05 19:26:02 +0300
@@ -31,17 +31,22 @@
 #include <string.h>
 
 #include "../config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "real.h"
 #include "asmrp.h"
 #include "sdpplin.h"
 #include "xbuffer.h"
 #if USE_LIBAVUTIL_SO
-#include "ffmpeg/md5.h"
+#include <ffmpeg/md5.h>
+#include <ffmpeg/intreadwrite.h>
 #else
 #include "libavutil/md5.h"
-#endif
 #include "libavutil/intreadwrite.h"
+#endif
 #include "../http.h"
 #include "mp_msg.h"
 
diff -urN mplayer-svn-r22915.orig/stream/realrtsp/rmff.c mplayer-svn-r22915/stream/realrtsp/rmff.c
--- mplayer-svn-r22915.orig/stream/realrtsp/rmff.c	2007-04-05 18:11:31 +0300
+++ mplayer-svn-r22915/stream/realrtsp/rmff.c	2007-04-05 19:26:02 +0300
@@ -29,7 +29,11 @@
 #include "rmff.h"
 #include "xbuffer.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 /*
 #define LOG
diff -urN mplayer-svn-r22915.orig/stream/stream_cdda.c mplayer-svn-r22915/stream/stream_cdda.c
--- mplayer-svn-r22915.orig/stream/stream_cdda.c	2007-04-05 18:11:31 +0300
+++ mplayer-svn-r22915/stream/stream_cdda.c	2007-04-05 19:26:02 +0300
@@ -6,7 +6,11 @@
 #include "stream.h"
 #include "m_option.h"
 #include "m_struct.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libmpdemux/demuxer.h"
 
diff -urN mplayer-svn-r22915.orig/stream/stream_dvdnav.c mplayer-svn-r22915/stream/stream_dvdnav.c
--- mplayer-svn-r22915.orig/stream/stream_dvdnav.c	2007-04-05 18:11:32 +0300
+++ mplayer-svn-r22915/stream/stream_dvdnav.c	2007-04-05 19:26:02 +0300
@@ -11,7 +11,11 @@
 #include "libmpdemux/demuxer.h"
 #include "stream_dvdnav.h"
 #include "libvo/video_out.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "spudec.h"
 #include "m_option.h"
 #include "m_struct.h"
diff -urN mplayer-svn-r22915.orig/stream/stream_netstream.c mplayer-svn-r22915/stream/stream_netstream.c
--- mplayer-svn-r22915.orig/stream/stream_netstream.c	2007-04-05 18:11:32 +0300
+++ mplayer-svn-r22915/stream/stream_netstream.c	2007-04-05 19:26:02 +0300
@@ -58,7 +58,11 @@
 #include "help_mp.h"
 #include "m_option.h"
 #include "m_struct.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "netstream.h"
diff -urN mplayer-svn-r22915.orig/TOOLS/Makefile mplayer-svn-r22915/TOOLS/Makefile
--- mplayer-svn-r22915.orig/TOOLS/Makefile	2007-04-05 18:15:55 +0300
+++ mplayer-svn-r22915/TOOLS/Makefile	2007-04-05 19:26:02 +0300
@@ -15,8 +15,8 @@
 
 subrip$(EXESUF): subrip.c
 	$(CC) $(CFLAGS) -g -o $@ $< ../vobsub.o ../spudec.o ../mp_msg.o \
-          ../unrarlib.o ../libswscale/libswscale.a ../libavutil/libavutil.a \
-          ../libmpcodecs/img_format.o ../osdep/getch2.o -ltermcap -lm
+          ../unrarlib.o ../libmpcodecs/img_format.o ../osdep/getch2.o \
+          -L../libswscale -lswscale -L../libavutil -lavutil -ltinfo -lm
 
 #FIXME: Linking is broken, help welcome.
 vivodump$(EXESUF): vivodump.c
diff -urN mplayer-svn-r22915.orig/vobsub.c mplayer-svn-r22915/vobsub.c
--- mplayer-svn-r22915.orig/vobsub.c	2007-04-05 18:10:42 +0300
+++ mplayer-svn-r22915/vobsub.c	2007-04-05 19:26:02 +0300
@@ -23,7 +23,11 @@
 #ifdef USE_UNRARLIB
 #include "unrarlib.h"
 #endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 extern int vobsub_id;
 
