diff -urN MPlayer-svn-20060621.orig/configure MPlayer-svn-20060621/configure
--- MPlayer-svn-20060621.orig/configure	2006-06-20 13:10:58 +0300
+++ MPlayer-svn-20060621/configure	2006-06-21 18:35:54 +0300
@@ -4143,6 +4143,8 @@
   _def_aa='#define HAVE_AA 1'
   if cygwin ; then
     _ld_aa=`aalib-config --libs | cut -d " " -f 2,5,6`
+  else
+    _ld_aa='-laa'
   fi
   _vosrc="$_vosrc vo_aa.c"
   _vomodules="aa $_vomodules"
@@ -4919,7 +4921,7 @@
 echores "$_jack"
 
 echocheck "OpenAL"
-if test "$_openal" = auto ; then
+if test "$_openal" != no ; then
   _openal=no
 cat > $TMPC << EOF
 #include <AL/al.h>
@@ -5375,7 +5377,7 @@
     _res_comment="freetype support needed"
 fi
 echocheck "fontconfig"
-if test "$_fontconfig" = auto ; then
+if test "$_fontconfig" != no ; then
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <fontconfig/fontconfig.h>
