diff -urN mplayer-svn-r27482.orig/configure mplayer-svn-r27482/configure
--- mplayer-svn-r27482.orig/configure	2008-08-26 04:01:15 +0300
+++ mplayer-svn-r27482/configure	2008-08-26 04:17:43 +0300
@@ -5458,10 +5458,10 @@
   if test "$_dl" = yes; then
     cat > $TMPC << EOF
 #include <inttypes.h>
-#include <libdvdread/dvd_reader.h>
-#include <libdvdread/ifo_types.h>
-#include <libdvdread/ifo_read.h>
-#include <libdvdread/nav_read.h>
+#include <dvdread/dvd_reader.h>
+#include <dvdread/ifo_types.h>
+#include <dvdread/ifo_read.h>
+#include <dvdread/nav_read.h>
 int main(void) { return 0; }
 EOF
   fi
diff -urN mplayer-svn-r27482.orig/stream/stream_dvd_common.c mplayer-svn-r27482/stream/stream_dvd_common.c
--- mplayer-svn-r27482.orig/stream/stream_dvd_common.c	2008-08-26 04:01:10 +0300
+++ mplayer-svn-r27482/stream/stream_dvd_common.c	2008-08-26 04:18:57 +0300
@@ -1,9 +1,9 @@
 #include "config.h"
 #include <inttypes.h>
 #ifdef CONFIG_DVDREAD_INTERNAL
-#include <dvdread/ifo_types.h>
+#include "dvdread/ifo_types.h"
 #else
-#include <libdvdread/ifo_types.h>
+#include <dvdread/ifo_types.h>
 #endif
 #include "stream_dvd_common.h"
 
diff -urN mplayer-svn-r27482.orig/stream/stream_dvd_common.h mplayer-svn-r27482/stream/stream_dvd_common.h
--- mplayer-svn-r27482.orig/stream/stream_dvd_common.h	2008-08-26 04:01:10 +0300
+++ mplayer-svn-r27482/stream/stream_dvd_common.h	2008-08-26 04:18:25 +0300
@@ -4,9 +4,9 @@
 #include "config.h"
 #include <inttypes.h>
 #ifdef CONFIG_DVDREAD_INTERNAL
-#include <dvdread/ifo_types.h>
+#include "dvdread/ifo_types.h"
 #else
-#include <libdvdread/ifo_types.h>
+#include <dvdread/ifo_types.h>
 #endif
 
 int mp_dvdtimetomsec(dvd_time_t *dt);
diff -urN mplayer-svn-r27482.orig/stream/stream_dvd.h mplayer-svn-r27482/stream/stream_dvd.h
--- mplayer-svn-r27482.orig/stream/stream_dvd.h	2008-08-26 04:01:10 +0300
+++ mplayer-svn-r27482/stream/stream_dvd.h	2008-08-26 04:17:43 +0300
@@ -9,10 +9,10 @@
 #include "dvdread/ifo_read.h"
 #include "dvdread/nav_read.h"
 #else
-#include <libdvdread/dvd_reader.h>
-#include <libdvdread/ifo_types.h>
-#include <libdvdread/ifo_read.h>
-#include <libdvdread/nav_read.h>
+#include <dvdread/dvd_reader.h>
+#include <dvdread/ifo_types.h>
+#include <dvdread/ifo_read.h>
+#include <dvdread/nav_read.h>
 #endif
 #include "stream.h"
 
