--- MPlayer-cvs-20060331.orig/DOCS/xml/configure	2006-03-31 16:28:57 +0300
+++ MPlayer-cvs-20060331/DOCS/xml/configure	2006-03-31 20:02:58 +0300
@@ -8,6 +8,7 @@
 
 echo "Searching for XML catalogs..."
 for _try_catalog in \
+  `pwd`/catalog \
   /etc/sgml/catalog \
   /usr/share/xml/docbook/*/catalog.xml \
   /usr/share/sgml/docbook/xml-dtd-4.1.2/xmlcatalog \
@@ -274,6 +275,7 @@
     echo "I will try to use OpenJade or Jade (using '$_jade')."
     echo "They aren't (currently) fully supported, however."
     for _try_docbook_dsl in \
+      /usr/share/sgml/docbook/dsssl-stylesheets-1.79/html/docbook.dsl \
       /usr/share/sgml/docbook/stylesheet/dsssl/modular/html/docbook.dsl \
       /usr/local/share/sgml/docbook/stylesheet/dsssl/modular/html/docbook.dsl
     do
@@ -284,6 +286,7 @@
       fi
     done
     for _try_xml_dcl in \
+      /usr/share/sgml/xml.dcl \
       /usr/share/sgml/declaration/xml.dcl \
       /usr/local/share/sgml/declaration/xml.dcl
     do
