diff -urN mplayer-svn-r21128.orig/configure mplayer-svn-r21128/configure
--- mplayer-svn-r21128.orig/configure	2006-11-21 15:37:21 +0200
+++ mplayer-svn-r21128/configure	2006-11-21 15:52:28 +0200
@@ -4790,7 +4790,7 @@
   _def_arts='#define USE_ARTS 1'
   _aosrc="$_aosrc ao_arts.c"
   _aomodules="arts $_aomodules"
-  _libs_mplayer="$_libs_mplayer `artsc-config --libs`"
+  _libs_mplayer="$_libs_mplayer -lartsc"
   _inc_extra="$_inc_extra `artsc-config --cflags`"
 else
   _noaomodules="arts $_noaomodules"
