--- MPlayer-1.0pre4.orig/help/help_mp-ru.h	2003-12-09 21:16:01 +0600
+++ MPlayer-1.0pre4/help/help_mp-ru.h	2004-05-03 05:27:00 +0700
@@ -67,9 +67,9 @@
 #define MSGTR_CmdlineDVDkey "Для расшифровки используется затребованный DVD ключ.\n"
 #define MSGTR_DVDauthOk "Авторизация DVD выглядит OK.\n"
 #define MSGTR_DumpSelectedStreamMissing "dump: FATAL: Выбранный поток потерян!\n"
-#define MSGTR_CantOpenDumpfile "Не могу открыть файл для дампирования!!!\n"
+#define MSGTR_CantOpenDumpfile "Не могу открыть файл для сохранения дампа!!!\n"
 #define MSGTR_CoreDumped "Core dumped ;)\n"
-#define MSGTR_FPSnotspecified "В заголовке кадры/сек не указаны (или недопустимые)! Используйте -fps опцию!\n"
+#define MSGTR_FPSnotspecified "В заголовке не указаны кадры/сек (или они недопустимы)! Используйте опцию -fps!\n"
 #define MSGTR_TryForceAudioFmtStr "Попытка форсировать семейство аудио кодеков %s...\n"
 #define MSGTR_CantFindAfmtFallback "Не могу найти аудио кодек для форсированного семейства драйверов, переход на другие драйвера.\n"
 #define MSGTR_CantFindAudioCodec "Не могу найти кодек для аудио формата 0x%X!\n"
@@ -82,9 +82,9 @@
 #define MSGTR_CannotInitAO "не могу открыть/инициализировать аудио устройство -> БЕЗ ЗВУКА\n"
 #define MSGTR_StartPlaying "Начало воcпроизведения...\n"
 #define MSGTR_SystemTooSlow "\n\n"\
-"         *****************************************************************\n"\
-"         **** Ваша система слишком МЕДЛЕННА чтобы воспроизводить это! ****\n"\
-"         *****************************************************************\n"\
+"         ******************************************************************\n"\
+"         **** Ваша система слишком МЕДЛЕННА, чтобы воспроизводить это! ****\n"\
+"         ******************************************************************\n"\
 "Возможные причины, проблемы, обходы: \n"\
 "- Наиболее частая: плохой/сырой _аудио_ драйвер\n"\
 "  - Попытайтесь -ao sdl или используйте ALSA 0.5 или эмуляцию oss на ALSA 0.9.\n"\
@@ -107,7 +107,7 @@
 #define MSGTR_Playing "Проигрывание %s.\n"
 #define MSGTR_NoSound "Аудио: без звука!!!\n"
 #define MSGTR_FPSforced "Кадры/сек форсированы в %5.3f (ftime: %5.3f).\n"
-#define MSGTR_CompiledWithRuntimeDetection "Скомпилировано для определениея типа процессора во время выполнения - ПРЕДУПРЕЖДЕНИЕ - это не оптимально!\nДля получения максимальной производительности, перекомпилируйте MPlayer c --disable-runtime-cpudetection.\n"
+#define MSGTR_CompiledWithRuntimeDetection "Скомпилировано с определением типа процессора во время выполнения - ПРЕДУПРЕЖДЕНИЕ - это не оптимально!\nДля получения максимальной производительности, перекомпилируйте MPlayer c --disable-runtime-cpudetection.\n"
 #define MSGTR_CompiledWithCPUExtensions "Скомпилировано для x86 CPU со следующими расширениями:"
 #define MSGTR_AvailableVideoOutputPlugins "Доступные плагины вывода видео:\n"
 #define MSGTR_AvailableVideoOutputDrivers "Доступные драйвера вывода видео:\n"
