diff -urN mplayer-svn-r23722.orig/etc/codecs.conf mplayer-svn-r23722/etc/codecs.conf
--- mplayer-svn-r23722.orig/etc/codecs.conf	2007-06-09 04:09:24 +0300
+++ mplayer-svn-r23722/etc/codecs.conf	2007-07-06 00:18:38 +0300
@@ -805,6 +805,14 @@
   out YUY2,UYVY
 ;  out BGR32,BGR24,BGR16 ;,BGR15
 
+videocodec vidm
+ info "VIDM"
+ status working
+ fourcc "VIDM"
+ driver vfw
+ dll "vidm.dll"
+ out YV12
+
 videocodec wmvvc1dmo
   info "Windows Media Video (VC-1) Advanced Profile Decoder"
   status working
