diff -urN mplayer-svn-r23560.orig/configure mplayer-svn-r23560/configure
--- mplayer-svn-r23560.orig/configure	2007-06-13 23:25:26 +0300
+++ mplayer-svn-r23560/configure	2007-06-16 03:06:50 +0300
@@ -128,7 +128,7 @@
 
 x86_64() {
   case "$host_arch" in
-    x86_64|amd64) return 0 ;;
+    x86-64|x86_64|amd64) return 0 ;;
     *) return 1 ;;
   esac
 }
@@ -1831,7 +1831,7 @@
     _optimizing=''
     ;;
 
-  x86_64|amd64)
+  x86-64|x86_64|amd64)
     _def_arch='#define ARCH_X86_64 1'
     _target_arch='TARGET_ARCH_X86_64 = yes'
     _def_arch_x86="#define ARCH_X86 1"
@@ -2836,6 +2836,7 @@
   _noinputmodules="network $_noinputmodules"
   _def_network='#undef MPLAYER_NETWORK'
   _ftp=no
+  _live=no
 fi
 echores "$_network"
 
@@ -3289,7 +3290,7 @@
 
 
 echocheck "termcap"
-if test "$_termcap" = auto ; then
+if test "$_termcap" != no ; then
   cat > $TMPC <<EOF
 int main(void) { tgetent(); return 0; }
 EOF
@@ -3309,7 +3310,7 @@
 
 
 echocheck "termios"
-if test "$_termios" = auto ; then
+if test "$_termios" != no ; then
   cat > $TMPC <<EOF
 #include <sys/termios.h>
 int main(void) { return 0; }
@@ -4264,7 +4265,9 @@
 if test "$_aa" = yes ; then
   _def_aa='#define HAVE_AA 1'
   if cygwin ; then
-    _libs_mplayer="$_libs_mplayer `aalib-config --libs | cut -d " " -f 2,5,6`"
+    _libs_mplayer="$_libs_mplayer `aalib-config --libs | cut -d ' ' -f 2,5,6`"
+  else
+    _libs_mplayer="$_libs_mplayer `aalib-config --libs | cut -d ' ' -f 1,2,4-6`"
   fi
   _vosrc="$_vosrc vo_aa.c"
   _vomodules="aa $_vomodules"
@@ -4292,7 +4295,7 @@
 if test "$_caca" = yes ; then
   _def_caca='#define HAVE_CACA 1'
   _inc_extra="$_inc_extra `caca-config --cflags`"
-  _libs_mplayer="$_libs_mplayer `caca-config --libs`"
+  _libs_mplayer="$_libs_mplayer `caca-config --libs | cut -d ' ' -f 1-4`"
   _vosrc="$_vosrc vo_caca.c"
   _vomodules="caca $_vomodules"
 else
@@ -4957,7 +4960,7 @@
 
 
 echocheck "JACK"
-if test "$_jack" = auto ; then
+if test "$_jack" != no ; then
   _jack=yes
 
 cat > $TMPC << EOF
@@ -4984,7 +4987,7 @@
 echores "$_jack"
 
 echocheck "OpenAL"
-if test "$_openal" = auto ; then
+if test "$_openal" != no ; then
   _openal=no
 cat > $TMPC << EOF
 #ifdef OPENAL_AL_H
@@ -5410,7 +5413,7 @@
     _res_comment="freetype support needed"
 fi
 echocheck "fontconfig"
-if test "$_fontconfig" = auto ; then
+if test "$_fontconfig" != no ; then
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <fontconfig/fontconfig.h>
@@ -5733,7 +5736,7 @@
 echores "$_speex"
 
 echocheck "OggTheora support"
-if test "$_theora" = auto ; then
+if test "$_theora" != no ; then
   _theora=no
     cat > $TMPC << EOF
 #include <theora/theora.h>
@@ -5849,7 +5852,7 @@
 
 
 echocheck "FAAC (AAC encoder) support"
-if test "$_faac" = auto ; then
+if test "$_faac" != no ; then
   cat > $TMPC <<EOF
 #include <inttypes.h> 
 #include <faac.h>
@@ -5970,7 +5973,7 @@
 
 
 echocheck "Win32 codecs"
-if test "$_win32dll" = auto ; then
+if test "$_win32dll" != no ; then
   _win32dll=no
   if x86_32 && not qnx; then
     _win32dll=yes
@@ -6040,56 +6043,6 @@
 echores "$_real"
 
 
-echocheck "LIVE555 Streaming Media libraries"
-if test "$_live" = auto  && test "$_network" = yes ; then
-  cat > $TMPCPP << EOF
-#include <liveMedia.hh>
-#if (LIVEMEDIA_LIBRARY_VERSION_INT < 1141257600)
-#error Please upgrade to version 2006.03.03 or later of the "LIVE555 Streaming Media" libraries - available from <www.live555.com/liveMedia/>
-#endif
-int main(void) {}
-EOF
-
-  _live=no
-  for I in $_inc_extra "-I$_libdir/live" "-I/usr/lib/live" "-I/usr/local/live" "-I/usr/local/lib/live" ; do
-    cxx_check $I/liveMedia/include $I/UsageEnvironment/include \
-      $I/groupsock/include && _livelibdir=`echo $I| sed s/-I//` && \
-      _ld_extra="$_livelibdir/liveMedia/libliveMedia.a \
-                 $_livelibdir/groupsock/libgroupsock.a \
-                 $_livelibdir/UsageEnvironment/libUsageEnvironment.a \
-                 $_livelibdir/BasicUsageEnvironment/libBasicUsageEnvironment.a \
-                 $_ld_extra  -lstdc++" \
-      _inc_extraxx="-I$_livelibdir/liveMedia/include \
-                  -I$_livelibdir/UsageEnvironment/include \
-                  -I$_livelibdir/BasicUsageEnvironment/include \
-                  -I$_livelibdir/groupsock/include" && \
-      _live=yes && break
-  done
-  if test "$_live" != yes ; then
-      if cxx_check -I/usr/include/liveMedia -I/usr/include/UsageEnvironment -I/usr/include/groupsock; then
-	  _live_dist=yes
-      fi
-  fi
-fi
-if test "$_live" = yes && test "$_network" = yes ; then
-  _res_comment="using $_livelibdir"
-  _def_live='#define STREAMING_LIVE555 1'
- _inputmodules="live555 $_inputmodules"
-elif test "$_live_dist" = yes && test "$_network" = yes ; then
-  _res_comment="using distribution version"
-  _live="yes"
-  _def_live='#define STREAMING_LIVE555 1'
-  _ld_extra="$_ld_extra -lliveMedia -lgroupsock -lUsageEnvironment -lBasicUsageEnvironment -lstdc++"
-  _inc_extraxx="-I/usr/include/liveMedia -I/usr/include/UsageEnvironment \
-             -I/usr/include/BasicUsageEnvironment -I/usr/include/groupsock"
- _inputmodules="live555 $_inputmodules"
-else
-  _def_live='#undef STREAMING_LIVE555'
-  _noinputmodules="live555 $_noinputmodules"
-fi
-echores "$_live"
-
-
 echocheck "FFmpeg libavutil"
 if test "$_libavutil_a" = auto ; then
   if test -d libavutil ; then
@@ -6098,7 +6051,7 @@
   else
     die "MPlayer will not compile without libavutil in the source tree."
   fi
-elif test "$_libavutil_so" = auto ; then
+elif test "$_libavutil_so" != no ; then
   _libavutil_so=no
   cat > $TMPC << EOF
 #include <ffmpeg/common.h>
@@ -6136,7 +6089,7 @@
     _libavcodec_a="yes"
     _res_comment="static"
   fi
-elif test "$_libavcodec_so" = auto ; then
+elif test "$_libavcodec_so" != no ; then
   _libavcodec_so=no
   _res_comment="libavcodec.so is discouraged over static libavcodec"
   cat > $TMPC << EOF
@@ -6180,7 +6133,7 @@
     _libavformat_a=yes
     _res_comment="static"
   fi
-elif test "$_libavformat_so" = auto ; then
+elif test "$_libavformat_so" != no ; then
   _libavformat_so=no
   cat > $TMPC <<EOF
   #include <ffmpeg/avformat.h>
@@ -6216,7 +6169,7 @@
     _libpostproc_a='yes'
     _res_comment="static"
   fi
-elif test "$_libpostproc_so" = auto ; then
+elif test "$_libpostproc_so" != no ; then
   _libpostproc_so=no
   cat > $TMPC << EOF
   #define USE_LIBPOSTPROC 1
@@ -6254,7 +6207,7 @@
 echores "$_md5sum"
 
 echocheck "libamr narrowband"
-if test "$_libamr_nb" = auto ; then
+if test "$_libavcodec_a" = yes -a "$_libamr_nb" = auto ; then
   _libamr_nb=no
   cat > $TMPC << EOF
 #include <amrnb/interf_dec.h>
@@ -6282,7 +6235,7 @@
 
 
 echocheck "libamr wideband"
-if test "$_libamr_wb" = auto ; then
+if test "$_libavcodec_a" = yes -a "$_libamr_wb" = auto ; then
   _libamr_wb=no
   cat > $TMPC << EOF
 #include <amrwb/dec_if.h>
@@ -6366,7 +6319,7 @@
 
 
 echocheck "XviD"
-if test "$_xvid" = auto ; then
+if test "$_xvid" != no ; then
   _xvid=no
   cat > $TMPC << EOF
 #include <xvid.h>
@@ -6405,7 +6358,7 @@
 
 
 echocheck "x264"
-if test "$_x264" = auto ; then
+if test "$_x264" != no ; then
   cat > $TMPC << EOF
 #include <inttypes.h>
 #include <x264.h>
@@ -6444,7 +6397,7 @@
 
 
 echocheck "nut"
-if test "$_nut" = auto ; then
+if test "$_nut" != no ; then
   cat > $TMPC << EOF
 #include <stdio.h>
 #include <stdlib.h>
@@ -6846,6 +6799,55 @@
 fi
 echores "$_ftp"
 
+echocheck "LIVE555 Streaming Media libraries"
+if test "$_live" != no  && test "$_network" = yes ; then
+  cat > $TMPCPP << EOF
+#include <liveMedia.hh>
+#if (LIVEMEDIA_LIBRARY_VERSION_INT < 1141257600)
+#error Please upgrade to version 2006.03.03 or later of the "LIVE555 Streaming Media" libraries - available from <www.live555.com/liveMedia/>
+#endif
+int main(void) {}
+EOF
+
+  _live=no
+  for I in "$_livelibdir" "$_libdir/live" "/usr/lib/live" "/usr/local/live" "/usr/local/lib/live" ; do
+    cxx_check -I$I/liveMedia/include -I$I/UsageEnvironment/include \
+      -I$I/groupsock/include && _livelibdir=`echo $I| sed s/-I//` && \
+      _ld_extra="$_livelibdir/liveMedia/libliveMedia.a \
+                 $_livelibdir/groupsock/libgroupsock.a \
+                 $_livelibdir/UsageEnvironment/libUsageEnvironment.a \
+                 $_livelibdir/BasicUsageEnvironment/libBasicUsageEnvironment.a \
+                 $_ld_extra  -lstdc++" \
+      _inc_extraxx="-I$_livelibdir/liveMedia/include \
+                  -I$_livelibdir/UsageEnvironment/include \
+                  -I$_livelibdir/BasicUsageEnvironment/include \
+                  -I$_livelibdir/groupsock/include" && \
+      _live=yes && break
+  done
+  if test "$_live" != yes ; then
+      if cxx_check -I/usr/include/liveMedia -I/usr/include/UsageEnvironment -I/usr/include/groupsock; then
+	  _live_dist=yes
+      fi
+  fi
+fi
+if test "$_live" = yes && test "$_network" = yes ; then
+  _res_comment="using $_livelibdir"
+  _def_live='#define STREAMING_LIVE555 1'
+ _inputmodules="live555 $_inputmodules"
+elif test "$_live_dist" = yes && test "$_network" = yes ; then
+  _res_comment="using distribution version"
+  _live="yes"
+  _def_live='#define STREAMING_LIVE555 1'
+  _ld_extra="$_ld_extra -lliveMedia -lgroupsock -lUsageEnvironment -lBasicUsageEnvironment -lstdc++"
+  _inc_extraxx="-I/usr/include/liveMedia -I/usr/include/UsageEnvironment \
+             -I/usr/include/BasicUsageEnvironment -I/usr/include/groupsock"
+ _inputmodules="live555 $_inputmodules"
+else
+  _def_live='#undef STREAMING_LIVE555'
+  _noinputmodules="live555 $_noinputmodules"
+fi
+echores "$_live"
+
 echocheck "vstream client"
 if test "$_vstream" = auto ; then
   _vstream=no
@@ -6941,7 +6943,7 @@
   if darwin ; then
      _ld_extra="$_ld_extra ${_xmmslibdir}/libxmms.dylib"
   else
-     _ld_extra="$_ld_extra ${_xmmslibdir}/libxmms.so.1 -export-dynamic"
+     _ld_extra="$_ld_extra `xmms-config --libs`"
   fi
 else
   _def_xmms='#undef HAVE_XMMS'
@@ -6997,10 +6999,7 @@
 if test "$_gui" = yes ; then
 
   # Required libraries
-  if test "$_libavcodec" != yes ||
-      not "echo $_libavdecoders | grep PNG_DECODER >/dev/null 2>&1" ; then
-    die "The GUI requires libavcodec with PNG support."
-  fi
+  test "$_libavcodec" = yes || die "The GUI requires libavcodec with PNG support."
 if not win32 ; then
   test "$_x11" != yes && die "X11 support required for GUI compilation."
 
@@ -7247,7 +7246,7 @@
   _res_comment="internal"
   _vidix_external=no
   _vidix=yes
-elif test "$_vidix_external" = auto; then
+elif test "$_vidix_external" != no; then
  _vidix_external=no
  cat > $TMPC <<EOF
 #include <vidix/vidix.h>
@@ -7278,7 +7277,7 @@
   _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/mplayer/vidix/\" "
 elif test "$_vidix_external" = yes ; then
   _libs_mplayer="$_libs_mplayer -lvidix"
-  _def_vidix_pfx='#define MP_VIDIX_PFX "" '
+  _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/vidix/\" "
 fi
 
 if test "$_vidix" = yes && win32; then
@@ -8349,6 +8348,7 @@
 $_def_vidix_drv_sis
 $_def_vidix_drv_unichrome
 $_def_vidix_pfx
+$_def_vidix_ext
 
 /* enables / disables new input joystick support */
 $_def_joystick
diff -urN mplayer-svn-r23560.orig/Makefile mplayer-svn-r23560/Makefile
--- mplayer-svn-r23560.orig/Makefile	2007-05-12 02:32:51 +0300
+++ mplayer-svn-r23560/Makefile	2007-06-16 03:03:38 +0300
@@ -73,7 +73,7 @@
                libao2/libao2.a \
                input/libinput.a \
 
-LIBS_MPLAYER-$(VIDIX)             += vidix/libvidix.a
+LIBS_MPLAYER-$(VIDIX_INTERNAL)    += vidix/libvidix.a
 LIBS_MPLAYER-$(GUI)               += gui/libgui.a
 LIBS_MPLAYER-$(LIBMENU)           += libmenu/libmenu.a
 
