diff -urN mplayer-svn-r22753.orig/libdha/Makefile mplayer-svn-r22753/libdha/Makefile
--- mplayer-svn-r22753.orig/libdha/Makefile	2007-03-19 00:20:12 +0200
+++ mplayer-svn-r22753/libdha/Makefile	2007-03-20 00:13:24 +0200
@@ -4,10 +4,7 @@
 MINOR_VERSION = 0
 VERSION = $(MAJOR_VERSION).$(MINOR_VERSION)
 
-LIBNAME = libdha.so
-ifeq ($(TARGET_WIN32),yes)
 LIBNAME = libdha.a
-endif
 
 SRCS_MPLAYER = libdha.c \
                mtrr.c \
@@ -34,13 +31,7 @@
 include ../mpcommon.mak
 
 $(LIBNAME): $(OBJS_MPLAYER)
-ifeq ($(TARGET_WIN32),yes)
 	$(AR) r $@ $^
-else
-	$(CC) -shared -Wl,-soname -Wl,$@ -o $@ $^ $(EXTRALIBS)
-	ln -sf $@ $@.$(VERSION)
-	ln -sf $@ $@.$(MAJOR_VERSION)
-endif
 
 pci_names.c:	oth/pci.db
 	LC_ALL=C $(AWK) -f pci_db2c.awk $<
@@ -59,7 +50,6 @@
 install:
 	-mkdir -p $(LIBDIR)
 	$(INSTALL) -m 755 $(INSTALLSTRIP) -p $(LIBNAME) $(LIBDIR)/$(LIBNAME)
-	ln -sf $(LIBNAME) $(LIBDIR)/$(LIBNAME).$(MAJOR_VERSION)
 	-$(LDCONFIG)
 
 uninstall:
diff -urN mplayer-svn-r22753.orig/Makefile mplayer-svn-r22753/Makefile
--- mplayer-svn-r22753.orig/Makefile	2007-03-19 21:51:02 +0200
+++ mplayer-svn-r22753/Makefile	2007-03-20 00:07:43 +0200
@@ -205,10 +205,10 @@
 tremor/libvorbisidec.a:
 	$(MAKE) -C tremor
 
-libdha/libdha.so:
+libdha/libdha.a:
 	$(MAKE) -C libdha
 
-vidix/libvidix.a: libdha/libdha.so
+vidix/libvidix.a: libdha/libdha.a
 	$(MAKE) -C vidix
 
 Gui/libgui.a:
@@ -295,7 +295,6 @@
 	$(INSTALL) -m 644 etc/mplayer.desktop $(prefix)/share/applications/
 
 install-vidix:
-	$(MAKE) -C libdha install
 	mkdir -p $(LIBDIR)/mplayer/vidix
 	$(INSTALL) -m 755 $(INSTALLSTRIP) -p vidix/drivers/*.so $(LIBDIR)/mplayer/vidix
 
@@ -415,7 +414,7 @@
 libfaad2/libfaad2.a: .norecurse $(wildcard libfaad2/*.[ch] libfaad2/*/*.[ch])
 
 loader/libloader.a: .norecurse $(wildcard loader/*.[chSs])
-libdha/libdha.so: .norecurse $(wildcard libdha/*.[ch])
+libdha/libdha.a: .norecurse $(wildcard libdha/*.[ch])
 vidix/libvidix.a: .norecurse $(wildcard vidix/*.[ch])
 Gui/libgui.a: .norecurse $(wildcard Gui/*.[ch] Gui/*/*.[ch] Gui/*/*/*.[ch])
 
