diff -urN MPlayer-1.0rc1.orig/configure MPlayer-1.0rc1/configure
--- MPlayer-1.0rc1.orig/configure	2006-10-23 01:32:31 +0300
+++ MPlayer-1.0rc1/configure	2006-10-25 14:24:15 +0300
@@ -3272,7 +3272,7 @@
 
 
 echocheck "termios"
-if test "$_termios" = auto ; then
+if test "$_termios" != no ; then
   cat > $TMPC <<EOF
 #include <sys/termios.h>
 int main(void) { return 0; }
@@ -4193,6 +4193,8 @@
   _def_aa='#define HAVE_AA 1'
   if cygwin ; then
     _ld_aa=`aalib-config --libs | cut -d " " -f 2,5,6`
+  else
+    _ld_aa=`aalib-config --libs | cut -d " " -f 1,2,4-6`
   fi
   _vosrc="$_vosrc vo_aa.c"
   _vomodules="aa $_vomodules"
@@ -4220,7 +4222,7 @@
 if test "$_caca" = yes ; then
   _def_caca='#define HAVE_CACA 1'
   _inc_extra="$_inc_extra `caca-config --cflags`"
-  _ld_caca=`caca-config --libs`
+  _ld_caca=`caca-config --libs | cut -d " " -f 1-4`
   _vosrc="$_vosrc vo_caca.c"
   _vomodules="caca $_vomodules"
 else
@@ -4910,7 +4912,7 @@
 
 
 echocheck "JACK"
-if test "$_jack" = auto ; then
+if test "$_jack" != no ; then
   _jack=yes
 
 cat > $TMPC << EOF
@@ -4937,7 +4939,7 @@
 echores "$_jack"
 
 echocheck "OpenAL"
-if test "$_openal" = auto ; then
+if test "$_openal" != no ; then
   _openal=no
 cat > $TMPC << EOF
 #include <AL/al.h>
@@ -5398,7 +5400,7 @@
     _res_comment="freetype support needed"
 fi
 echocheck "fontconfig"
-if test "$_fontconfig" = auto ; then
+if test "$_fontconfig" != no ; then
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <fontconfig/fontconfig.h>
@@ -5845,7 +5847,7 @@
 
 
 echocheck "FAAC (AAC encoder) support"
-if test "$_faac" = auto ; then
+if test "$_faac" != no ; then
   cat > $TMPC <<EOF
 #include <inttypes.h> 
 #include <faac.h>
@@ -5945,7 +5947,7 @@
 
 if x86 && not qnx; then
 
-if test "$_win32" = auto ; then
+if test "$_win32" != no ; then
     if test -z "$_win32libdir" ; then
       for I in "$_libdir/codecs" "$_libdir/win32" /usr/local/lib/codecs /usr/lib/codecs /usr/local/lib/win32 /usr/lib/win32 ; do
         if test -d "$I" ; then
@@ -6062,55 +6064,6 @@
 fi
 echores "$_real"
 
-
-echocheck "LIVE555 Streaming Media libraries"
-if test "$_live" = auto  && test "$_network" = yes ; then
-  cat > $TMPCPP << EOF
-#include <liveMedia.hh>
-#if (LIVEMEDIA_LIBRARY_VERSION_INT < 1090195200)
-#error Please upgrade to version 2004.07.19 or later of the "LIVE555 Streaming Media" libraries - available from <www.live555.com/liveMedia/>
-#endif
-int main(void) {}
-EOF
-
-  _live=no
-  for I in "$_livelibdir" "$_libdir/live" "/usr/lib/live" "/usr/local/live" "/usr/local/lib/live" ; do
-    cxx_check -I$I/liveMedia/include -I$I/UsageEnvironment/include -I$I/groupsock/include && _livelibdir=$I && _live=yes && break
-  done
-  if test "$_live" != yes ; then
-      if cxx_check -I/usr/include/liveMedia -I/usr/include/UsageEnvironment -I/usr/include/groupsock; then
-	  _live_dist=yes
-      fi
-  fi
-fi
-if test "$_live" = yes && test "$_network" = yes ; then
-  _res_comment="using $_livelibdir"
-  _def_live='#define STREAMING_LIVE555 1'
-  _ld_live="$_livelibdir/liveMedia/libliveMedia.a \
-           $_livelibdir/groupsock/libgroupsock.a \
-           $_livelibdir/UsageEnvironment/libUsageEnvironment.a \
-           $_livelibdir/BasicUsageEnvironment/libBasicUsageEnvironment.a \
-           -lstdc++"
-  _inc_extra="$_inc_extra -I$_livelibdir/liveMedia/include \
-             -I$_livelibdir/UsageEnvironment/include \
-             -I$_livelibdir/BasicUsageEnvironment/include \
-             -I$_livelibdir/groupsock/include"
- _inputmodules="live555 $_inputmodules"
-elif test "$_live_dist" = yes && test "$_network" = yes ; then
-  _res_comment="using distribution version"
-  _live="yes"
-  _def_live='#define STREAMING_LIVE555 1'
-  _ld_live="-lliveMedia -lgroupsock -lUsageEnvironment -lBasicUsageEnvironment -lstdc++"
-  _inc_extra="$_inc_extra -I/usr/include/liveMedia -I/usr/include/UsageEnvironment \
-             -I/usr/include/BasicUsageEnvironment -I/usr/include/groupsock"
- _inputmodules="live555 $_inputmodules"
-else
-  _def_live='#undef STREAMING_LIVE555'
-  _noinputmodules="live555 $_noinputmodules"
-fi
-echores "$_live"
-
-
 echocheck "FFmpeg libavutil (static)"
 if test "$_libavutil" = auto ; then
   if test -d libavutil ; then
@@ -6546,7 +6499,7 @@
 fi
 
 echocheck "x264"
-if test "$_x264" = auto ; then
+if test "$_x264" != no ; then
   cat > $TMPC << EOF
 #include <inttypes.h>
 #include <x264.h>
@@ -6581,7 +6534,7 @@
 
 
 echocheck "nut"
-if test "$_nut" = auto ; then
+if test "$_nut" != no ; then
   cat > $TMPC << EOF
 #include <stdio.h>
 #include <stdlib.h>
@@ -6863,6 +6816,7 @@
   _noinputmodules="network $_noinputmodules"
   _def_network='#undef MPLAYER_NETWORK'
   _ftp=no
+  _live=no
 fi
 echores "$_network"
 
@@ -6876,6 +6830,53 @@
 fi
 echores "$_ftp"
 
+echocheck "LIVE555 Streaming Media libraries"
+if test "$_live" != no  && test "$_network" = yes ; then
+  cat > $TMPCPP << EOF
+#include <liveMedia.hh>
+#if (LIVEMEDIA_LIBRARY_VERSION_INT < 1090195200)
+#error Please upgrade to version 2004.07.19 or later of the "LIVE555 Streaming Media" libraries - available from <www.live555.com/liveMedia/>
+#endif
+int main(void) {}
+EOF
+
+  _live=no
+  for I in "$_livelibdir" "$_libdir/live" "/usr/lib/live" "/usr/local/live" "/usr/local/lib/live" ; do
+    cxx_check -I$I/liveMedia/include -I$I/UsageEnvironment/include -I$I/groupsock/include && _livelibdir=$I && _live=yes && break
+  done
+  if test "$_live" != yes ; then
+      if cxx_check -I/usr/include/liveMedia -I/usr/include/UsageEnvironment -I/usr/include/groupsock; then
+	  _live_dist=yes
+      fi
+  fi
+fi
+if test "$_live" = yes && test "$_network" = yes ; then
+  _res_comment="using $_livelibdir"
+  _def_live='#define STREAMING_LIVE555 1'
+  _ld_live="$_livelibdir/liveMedia/libliveMedia.a \
+           $_livelibdir/groupsock/libgroupsock.a \
+           $_livelibdir/UsageEnvironment/libUsageEnvironment.a \
+           $_livelibdir/BasicUsageEnvironment/libBasicUsageEnvironment.a \
+           -lstdc++"
+  _inc_extra="$_inc_extra -I$_livelibdir/liveMedia/include \
+             -I$_livelibdir/UsageEnvironment/include \
+             -I$_livelibdir/BasicUsageEnvironment/include \
+             -I$_livelibdir/groupsock/include"
+ _inputmodules="live555 $_inputmodules"
+elif test "$_live_dist" = yes && test "$_network" = yes ; then
+  _res_comment="using distribution version"
+  _live="yes"
+  _def_live='#define STREAMING_LIVE555 1'
+  _ld_live="-lliveMedia -lgroupsock -lUsageEnvironment -lBasicUsageEnvironment -lstdc++"
+  _inc_extra="$_inc_extra -I/usr/include/liveMedia -I/usr/include/UsageEnvironment \
+             -I/usr/include/BasicUsageEnvironment -I/usr/include/groupsock"
+ _inputmodules="live555 $_inputmodules"
+else
+  _def_live='#undef STREAMING_LIVE555'
+  _noinputmodules="live555 $_noinputmodules"
+fi
+echores "$_live"
+
 echocheck "vstream client"
 if test "$_vstream" = auto ; then
   _vstream=no
@@ -7327,6 +7328,9 @@
 if test "$_vidix_internal" = yes || test "$_vidix_external" = yes ; then
   _vidix=yes
   _def_vidix='#define CONFIG_VIDIX 1'
+  if test "$_vidix_external" = yes ; then
+    _def_vidix_ext='#define VIDIX_EXT 1'
+  fi
 else
   _vidix=no
   _def_vidix='#undef CONFIG_VIDIX'
@@ -7336,7 +7340,7 @@
   _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/mplayer/vidix/\" "
 elif test "$_vidix_external" = yes ; then
   _ld_vidix_external="-lvidix"
-  _def_vidix_pfx='#define MP_VIDIX_PFX "" '
+  _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/vidix/\" "
 fi
 
 if test "$_vidix" = yes; then
@@ -8289,6 +8293,7 @@
 /* enables / disables VIDIX usage */
 $_def_vidix
 $_def_vidix_pfx
+$_def_vidix_ext
 
 /* enables / disables new input joystick support */
 $_def_joystick
