diff -urN mplayer-svn-r23547.orig/gui/bitmap.c mplayer-svn-r23547/gui/bitmap.c
--- mplayer-svn-r23547.orig/gui/bitmap.c	2007-06-12 22:14:49 +0300
+++ mplayer-svn-r23547/gui/bitmap.c	2007-06-12 22:31:32 +0300
@@ -10,7 +10,11 @@
 #else
 #include "libavcodec/avcodec.h"
 #endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 #include "libvo/fastmemcpy.h"
 
 static int pngRead( unsigned char * fname,txSample * bf )
diff -urN mplayer-svn-r23547.orig/gui/mplayer/play.h mplayer-svn-r23547/gui/mplayer/play.h
--- mplayer-svn-r23547.orig/gui/mplayer/play.h	2007-04-23 23:01:47 +0300
+++ mplayer-svn-r23547/gui/mplayer/play.h	2007-06-12 22:31:06 +0300
@@ -2,7 +2,7 @@
 #ifndef __GUI_PLAY_H
 #define __GUI_PLAY_H
 
-#include "../config.h"
+#include "../../config.h"
 
 extern int mplGotoTheNext;
 
