--- MPlayer-1.0pre7try2/configure.orig	2006-01-13 17:50:58 +0300
+++ MPlayer-1.0pre7try2/configure	2006-01-13 17:59:42 +0300
@@ -1890,7 +1890,7 @@ test -z "$_bindir" && _bindir="$_prefix/
 test -z "$_datadir" && _datadir="$_prefix/share/mplayer"
 test -z "$_mandir" && _mandir="$_prefix/man"
 test -z "$_confdir" && _confdir="$_prefix/etc/mplayer"
-test -z "$_libdir" && _libdir="$_prefix/lib"
+test -z "$_libdir" && _libdir="$_libdir"
 test -z "$_mlibdir" && _mlibdir="$MLIBHOME"
 
 if x86 ; then
