diff -urN mplayer-svn-r26107.orig/configure mplayer-svn-r26107/configure
--- mplayer-svn-r26107.orig/configure	2008-02-27 01:48:47 +0200
+++ mplayer-svn-r26107/configure	2008-02-27 02:05:37 +0200
@@ -292,10 +292,12 @@
   --disable-libavcodec_a    disable static libavcodec [autodetect]
   --disable-libavformat_a   disable static libavformat [autodetect]
   --disable-libpostproc_a   disable static libpostproc [autodetect]
+  --disable-libswscale_a    disable static libswscale [autodetect]
   --disable-libavutil_so    disable shared libavutil [autodetect]
   --disable-libavcodec_so   disable shared libavcodec [autodetect]
   --disable-libavformat_so  disable shared libavformat [autodetect]
   --disable-libpostproc_so  disable shared libpostproc [autodetect]
+  --disable-libswscale_so   disable shared libswscale [autodetect]
   --disable-libavcodec_mpegaudio_hp disable high precision audio decoding
                                     in libavcodec [enabled]
   --disable-tremor-internal disable internal Tremor [enabled]
@@ -512,6 +514,8 @@
 _libavformat_so=auto
 _libpostproc_a=auto
 _libpostproc_so=auto
+_libswscale_a=auto
+_libswscale_so=auto
 _libavcodec_mpegaudio_hp=yes
 _mencoder=yes
 _mplayer=yes
@@ -1047,6 +1051,10 @@
   --disable-libpostproc_a)	_libpostproc_a=no	;;
   --enable-libpostproc_so)	_libpostproc_so=yes	;;
   --disable-libpostproc_so)	_libpostproc_so=no	;;
+  --enable-libswscale_a)	_libswscale_a=yes	;;
+  --disable-libswscale_a)	_libswscale_a=no	;;
+  --enable-libswscale_so)	_libswscale_so=yes	;;
+  --disable-libswscale_so)	_libswscale_so=no	;;
   --enable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=yes	;;
   --disable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=no	;;
 
@@ -6266,6 +6274,45 @@
 fi
 echores "$_real"
 
+echocheck "FFmpeg libswscale"
+if test "$_libswscale_a" = auto ; then
+  if test -d libswascale ; then
+    _libswscale_a=yes
+    _res_comment="static"
+  else
+    die "MPlayer will not compile without libswscale in the source tree."
+  fi
+elif test "$_libswscale_so" != no ; then
+  _libswscale_so=no
+  cat > $TMPC << EOF
+#include <ffmpeg/swscale.h>
+int main(void) { sws_getIdentityVec(); return 0; }
+EOF
+  if $_pkg_config --exists libswscale ; then
+    _inc_libswscale=`$_pkg_config --cflags libswscale`
+    _ld_tmp=`$_pkg_config --libs libswscale`
+    cc_check $_inc_libswscale $_ld_tmp && _ld_extra="$_ld_extra $_ld_tmp" \
+      && _libswscale_so=yes
+  elif cc_check -lswscale $_ld_lm ; then
+    _ld_extra="$_ld_extra -lswscale"
+    _libswscale_so=yes
+    _res_comment="using libswscale.so, but static libswscale is recommended"
+  fi
+fi
+_libswscale=no
+_def_libswscale='#undef USE_LIBSWSCALE'
+_def_libswscale_a='#undef USE_LIBSWSCALE_A'
+_def_libswscale_so='#undef USE_LIBSWSCALE_SO'
+test "$_libswscale_a" = yes || test "$_libswscale_so" = yes && _libswscale=yes
+test "$_libswscale"    = yes && _def_libswscale='#define USE_LIBSWSCALE 1'
+test "$_libswscale_a"  = yes && _def_libswscale_a='#define USE_LIBSWSCALE_A 1'
+test "$_libswscale_so" = yes && _def_libswscale_so='#define USE_LIBSWSCALE_SO 1'
+# neither static nor shared libswscale is available, but it is mandatory ...
+if test "$_libswscale" = no ; then
+  die "You need static or shared libswscale, MPlayer will not compile without!"
+fi
+echores "$_libswscale"
+
 
 echocheck "QuickTime codecs"
 if test "$_qtx" = auto ; then
@@ -7760,6 +7807,10 @@
 LIBPOSTPROC = $_libpostproc
 LIBPOSTPROC_A = $_libpostproc_a
 LIBPOSTPROC_SO = $_libpostproc_so
+LIBSWSCALE = $_libswscale
+LIBSWSCALE_A = $_libswscale_a
+LIBSWSCALE_SO = $_libswscale_so
+CONFIG_SWSCALER = $_libswscale
 ZORAN = $_zr
 LIBLZO = $_liblzo
 LIBDV = $_libdv
@@ -8151,6 +8202,10 @@
 $_def_libpostproc_a
 $_def_libpostproc_so
 
+$_def_libswscale
+$_def_libswscale_a
+$_def_libswscale_so
+
 /* Win32 DLL support */
 $_def_win32dll
 #define WIN32_PATH "$_win32codecsdir"
diff -urN mplayer-svn-r26107.orig/gui/mplayer/gui_common.c mplayer-svn-r26107/gui/mplayer/gui_common.c
--- mplayer-svn-r26107.orig/gui/mplayer/gui_common.c	2008-01-11 02:30:35 +0200
+++ mplayer-svn-r26107/gui/mplayer/gui_common.c	2008-02-27 02:05:37 +0200
@@ -25,7 +25,11 @@
 #include "../libmpdemux/stheader.h"
 #include "../codec-cfg.h"
 #include "../access_mpcontext.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "../libavutil/avstring.h"
+#endif
 
 
 #include "play.h"
diff -urN mplayer-svn-r26107.orig/gui/skin/font.c mplayer-svn-r26107/gui/skin/font.c
--- mplayer-svn-r26107.orig/gui/skin/font.c	2007-07-06 23:36:36 +0300
+++ mplayer-svn-r26107/gui/skin/font.c	2008-02-27 02:05:37 +0200
@@ -10,7 +10,11 @@
 #include "font.h"
 #include "cut.h"
 #include "../mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "../libavutil/avstring.h"
+#endif
 
 int items;
 
diff -urN mplayer-svn-r26107.orig/gui/skin/skin.c mplayer-svn-r26107/gui/skin/skin.c
--- mplayer-svn-r26107.orig/gui/skin/skin.c	2007-07-06 23:36:36 +0300
+++ mplayer-svn-r26107/gui/skin/skin.c	2008-02-27 02:05:37 +0200
@@ -11,7 +11,11 @@
 #include "../mp_msg.h"
 #include "../help_mp.h"
 #include "mplayer/widgets.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/avstring.h"
+#endif
 
 //#define MSGL_DBG2 MSGL_STATUS
 
diff -urN mplayer-svn-r26107.orig/gui/wm/ws.c mplayer-svn-r26107/gui/wm/ws.c
--- mplayer-svn-r26107.orig/gui/wm/ws.c	2007-07-19 01:16:01 +0300
+++ mplayer-svn-r26107/gui/wm/ws.c	2008-02-27 02:05:37 +0200
@@ -24,8 +24,13 @@
 #include "ws.h"
 #include "wsxdnd.h"
 #include "../cpudetect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "../libswscale/swscale.h"
 #include "../libswscale/rgb2rgb.h"
+#endif
 #include "../libmpcodecs/vf_scale.h"
 #include "../mp_msg.h"
 #include "../help_mp.h"
diff -urN mplayer-svn-r26107.orig/input/input.c mplayer-svn-r26107/input/input.c
--- mplayer-svn-r26107.orig/input/input.c	2008-01-26 00:38:06 +0200
+++ mplayer-svn-r26107/input/input.c	2008-02-27 02:05:37 +0200
@@ -20,7 +20,11 @@
 #include "osdep/getch2.h"
 #include "osdep/keycodes.h"
 #include "osdep/timer.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "avstring.h"
+#endif
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "m_config.h"
diff -urN mplayer-svn-r26107.orig/liba52/parse.c mplayer-svn-r26107/liba52/parse.c
--- mplayer-svn-r26107.orig/liba52/parse.c	2006-11-09 16:34:19 +0200
+++ mplayer-svn-r26107/liba52/parse.c	2008-02-27 02:05:37 +0200
@@ -37,7 +37,11 @@
 #include "bitstream.h"
 #include "tables.h"
 #include "mm_accel.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "libavutil/avutil.h"
+#endif
 
 #ifdef HAVE_MEMALIGN
 /* some systems have memalign() but no declaration for it */
diff -urN mplayer-svn-r26107.orig/libaf/af_format.c mplayer-svn-r26107/libaf/af_format.c
--- mplayer-svn-r26107.orig/libaf/af_format.c	2007-11-10 23:20:51 +0200
+++ mplayer-svn-r26107/libaf/af_format.c	2008-02-27 02:05:37 +0200
@@ -21,7 +21,11 @@
 #endif
 
 #include "af.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r26107.orig/libaf/af_resample.c mplayer-svn-r26107/libaf/af_resample.c
--- mplayer-svn-r26107.orig/libaf/af_resample.c	2007-11-10 23:20:51 +0200
+++ mplayer-svn-r26107/libaf/af_resample.c	2008-02-27 02:05:37 +0200
@@ -13,7 +13,12 @@
 #include <stdlib.h>
 #include <inttypes.h>
 
+#include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "af.h"
 #include "dsp.h"
 
diff -urN mplayer-svn-r26107.orig/libaf/af_scaletempo.c mplayer-svn-r26107/libaf/af_scaletempo.c
--- mplayer-svn-r26107.orig/libaf/af_scaletempo.c	2007-12-02 17:50:51 +0200
+++ mplayer-svn-r26107/libaf/af_scaletempo.c	2008-02-27 02:05:37 +0200
@@ -36,7 +36,12 @@
 #include <limits.h>
 
 #include "af.h"
+#include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "subopt-helper.h"
 #include "help_mp.h"
 
diff -urN mplayer-svn-r26107.orig/libao2/ao_dxr2.c mplayer-svn-r26107/libao2/ao_dxr2.c
--- mplayer-svn-r26107.orig/libao2/ao_dxr2.c	2008-01-21 04:18:26 +0200
+++ mplayer-svn-r26107/libao2/ao_dxr2.c	2008-02-27 02:05:37 +0200
@@ -7,7 +7,11 @@
 #include "config.h"
 #include "mp_msg.h"
 #include "help_mp.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "audio_out.h"
diff -urN mplayer-svn-r26107.orig/libao2/ao_pcm.c mplayer-svn-r26107/libao2/ao_pcm.c
--- mplayer-svn-r26107.orig/libao2/ao_pcm.c	2007-12-11 00:55:26 +0200
+++ mplayer-svn-r26107/libao2/ao_pcm.c	2008-02-27 02:05:37 +0200
@@ -4,7 +4,11 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "subopt-helper.h"
 #include "libaf/af_format.h"
diff -urN mplayer-svn-r26107.orig/libass/mputils.h mplayer-svn-r26107/libass/mputils.h
--- mplayer-svn-r26107.orig/libass/mputils.h	2008-02-06 00:33:19 +0200
+++ mplayer-svn-r26107/libass/mputils.h	2008-02-27 02:05:37 +0200
@@ -6,6 +6,10 @@
 #include "libvo/font_load.h" // for blur()
 #include "subreader.h" // for guess_buffer_cp
 #include "libvo/sub.h" // for utf8_get_char
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #endif /* LIBASS_MPUTILS_H */
diff -urN mplayer-svn-r26107.orig/libmenu/menu.c mplayer-svn-r26107/libmenu/menu.c
--- mplayer-svn-r26107.orig/libmenu/menu.c	2008-02-23 00:42:57 +0200
+++ mplayer-svn-r26107/libmenu/menu.c	2008-02-27 02:05:37 +0200
@@ -405,7 +405,11 @@
 
 #ifdef USE_FRIBIDI
 #include <fribidi/fribidi.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 char *menu_fribidi_charset = NULL;
 int menu_flip_hebrew = 0;
 int menu_fribidi_flip_commas = 0;
diff -urN mplayer-svn-r26107.orig/libmpcodecs/ad_dk3adpcm.c mplayer-svn-r26107/libmpcodecs/ad_dk3adpcm.c
--- mplayer-svn-r26107.orig/libmpcodecs/ad_dk3adpcm.c	2006-12-08 20:58:10 +0200
+++ mplayer-svn-r26107/libmpcodecs/ad_dk3adpcm.c	2008-02-27 02:05:37 +0200
@@ -16,7 +16,11 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "ad_internal.h"
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/ad_imaadpcm.c mplayer-svn-r26107/libmpcodecs/ad_imaadpcm.c
--- mplayer-svn-r26107.orig/libmpcodecs/ad_imaadpcm.c	2006-12-08 20:58:09 +0200
+++ mplayer-svn-r26107/libmpcodecs/ad_imaadpcm.c	2008-02-27 02:05:37 +0200
@@ -21,7 +21,11 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "ad_internal.h"
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/ad_msadpcm.c mplayer-svn-r26107/libmpcodecs/ad_msadpcm.c
--- mplayer-svn-r26107.orig/libmpcodecs/ad_msadpcm.c	2006-12-08 20:58:10 +0200
+++ mplayer-svn-r26107/libmpcodecs/ad_msadpcm.c	2008-02-27 02:05:37 +0200
@@ -12,7 +12,11 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "ad_internal.h"
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/ad_qtaudio.c mplayer-svn-r26107/libmpcodecs/ad_qtaudio.c
--- mplayer-svn-r26107.orig/libmpcodecs/ad_qtaudio.c	2007-03-19 00:20:09 +0200
+++ mplayer-svn-r26107/libmpcodecs/ad_qtaudio.c	2008-02-27 02:05:37 +0200
@@ -14,7 +14,13 @@
 #include "loader/wine/windef.h"
 
 #include "ad_internal.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/common.h"
+#include "libavutil/bswap.h"
+#endif
 
 #ifdef WIN32_LOADER
 #include "loader/ldt_keeper.h"
diff -urN mplayer-svn-r26107.orig/libmpcodecs/native/nuppelvideo.c mplayer-svn-r26107/libmpcodecs/native/nuppelvideo.c
--- mplayer-svn-r26107.orig/libmpcodecs/native/nuppelvideo.c	2008-02-27 01:48:44 +0200
+++ mplayer-svn-r26107/libmpcodecs/native/nuppelvideo.c	2008-02-27 02:05:37 +0200
@@ -13,7 +13,11 @@
 
 #include "config.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "../libvo/fastmemcpy.h"
diff -urN mplayer-svn-r26107.orig/libmpcodecs/native/RTjpegN.c mplayer-svn-r26107/libmpcodecs/native/RTjpegN.c
--- mplayer-svn-r26107.orig/libmpcodecs/native/RTjpegN.c	2007-01-22 19:01:51 +0200
+++ mplayer-svn-r26107/libmpcodecs/native/RTjpegN.c	2008-02-27 02:05:37 +0200
@@ -30,7 +30,11 @@
 #define MMX
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "RTjpegN.h"
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vd_ffmpeg.c mplayer-svn-r26107/libmpcodecs/vd_ffmpeg.c
--- mplayer-svn-r26107.orig/libmpcodecs/vd_ffmpeg.c	2008-02-27 01:48:45 +0200
+++ mplayer-svn-r26107/libmpcodecs/vd_ffmpeg.c	2008-02-27 02:05:37 +0200
@@ -7,8 +7,15 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/avutil.h>
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/common.h"
+#include "libavutil/avutil.h"
 #include "libavutil/intreadwrite.h"
+#endif
 #include "mpbswap.h"
 
 #include "vd_internal.h"
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vd_ijpg.c mplayer-svn-r26107/libmpcodecs/vd_ijpg.c
--- mplayer-svn-r26107.orig/libmpcodecs/vd_ijpg.c	2008-01-21 04:18:25 +0200
+++ mplayer-svn-r26107/libmpcodecs/vd_ijpg.c	2008-02-27 02:05:37 +0200
@@ -12,7 +12,11 @@
 
 #include <setjmp.h>
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "vd_internal.h"
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vd_mpng.c mplayer-svn-r26107/libmpcodecs/vd_mpng.c
--- mplayer-svn-r26107.orig/libmpcodecs/vd_mpng.c	2007-06-05 23:28:24 +0300
+++ mplayer-svn-r26107/libmpcodecs/vd_mpng.c	2008-02-27 02:05:37 +0200
@@ -6,7 +6,11 @@
 
 #include <png.h>
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vd_mtga.c mplayer-svn-r26107/libmpcodecs/vd_mtga.c
--- mplayer-svn-r26107.orig/libmpcodecs/vd_mtga.c	2007-06-05 23:28:24 +0300
+++ mplayer-svn-r26107/libmpcodecs/vd_mtga.c	2008-02-27 02:05:37 +0200
@@ -11,7 +11,11 @@
 #include "config.h"
 #include "mp_msg.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vd_qtvideo.c mplayer-svn-r26107/libmpcodecs/vd_qtvideo.c
--- mplayer-svn-r26107.orig/libmpcodecs/vd_qtvideo.c	2008-02-15 02:21:54 +0200
+++ mplayer-svn-r26107/libmpcodecs/vd_qtvideo.c	2008-02-27 02:05:37 +0200
@@ -28,7 +28,7 @@
 
 LIBVD_EXTERN(qtvideo)
 
-#include "bswap.h"
+#include "mpbswap.h"
 
 #ifndef MACOSX
 #include "loader/qtx/qtxsdk/components.h"
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vd_sgi.c mplayer-svn-r26107/libmpcodecs/vd_sgi.c
--- mplayer-svn-r26107.orig/libmpcodecs/vd_sgi.c	2006-12-08 20:58:10 +0200
+++ mplayer-svn-r26107/libmpcodecs/vd_sgi.c	2008-02-27 02:05:37 +0200
@@ -7,7 +7,11 @@
 
 #include "config.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "vd_internal.h"
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vd_xanim.c mplayer-svn-r26107/libmpcodecs/vd_xanim.c
--- mplayer-svn-r26107.orig/libmpcodecs/vd_xanim.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r26107/libmpcodecs/vd_xanim.c	2008-02-27 02:05:37 +0200
@@ -37,7 +37,11 @@
 #include <errno.h> /* strerror, errno */
 
 #include "mp_msg.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 
 #include "osdep/timer.h"
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/ve_qtvideo.c mplayer-svn-r26107/libmpcodecs/ve_qtvideo.c
--- mplayer-svn-r26107.orig/libmpcodecs/ve_qtvideo.c	2008-02-15 02:21:54 +0200
+++ mplayer-svn-r26107/libmpcodecs/ve_qtvideo.c	2008-02-27 02:05:37 +0200
@@ -10,7 +10,11 @@
 
 #include "config.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #ifdef WIN32_LOADER 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_bmovl.c mplayer-svn-r26107/libmpcodecs/vf_bmovl.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_bmovl.c	2007-11-10 23:20:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_bmovl.c	2008-02-27 02:05:37 +0200
@@ -69,7 +69,11 @@
 #include "img_format.h"
 
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_divtc.c mplayer-svn-r26107/libmpcodecs/vf_divtc.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_divtc.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_divtc.c	2008-02-27 02:05:37 +0200
@@ -7,7 +7,11 @@
 #include "config.h"
 #include "mp_msg.h"
 #include "cpudetect.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "img_format.h"
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_geq.c mplayer-svn-r26107/libmpcodecs/vf_geq.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_geq.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_geq.c	2008-02-27 02:05:37 +0200
@@ -32,8 +32,13 @@
 #include "vf.h"
 
 #define HAVE_AV_CONFIG_H
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#include <ffmpeg/eval.h>
+#else
 #include "libavcodec/avcodec.h"
 #include "libavcodec/eval.h"
+#endif
 
 struct vf_priv_s {
     AVEvalExpr * e[3];
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_halfpack.c mplayer-svn-r26107/libmpcodecs/vf_halfpack.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_halfpack.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_halfpack.c	2008-02-27 02:05:37 +0200
@@ -11,7 +11,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 
 struct vf_priv_s {
 	int field;
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_palette.c mplayer-svn-r26107/libmpcodecs/vf_palette.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_palette.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_palette.c	2008-02-27 02:05:37 +0200
@@ -11,7 +11,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 
 //===========================================================================//
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_qp.c mplayer-svn-r26107/libmpcodecs/vf_qp.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_qp.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_qp.c	2008-02-27 02:05:37 +0200
@@ -33,8 +33,13 @@
 
 // Needed to bring in lrintf.
 #define HAVE_AV_CONFIG_H
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#include <ffmpeg/eval.h>
+#else
 #include "libavcodec/avcodec.h"
 #include "libavcodec/eval.h"
+#endif
 
 #ifdef HAVE_MALLOC_H
 #include <malloc.h>
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_rectangle.c mplayer-svn-r26107/libmpcodecs/vf_rectangle.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_rectangle.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_rectangle.c	2008-02-27 02:05:37 +0200
@@ -6,7 +6,11 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 struct vf_priv_s {
     int x, y, w, h;
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_rgb2bgr.c mplayer-svn-r26107/libmpcodecs/vf_rgb2bgr.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_rgb2bgr.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_rgb2bgr.c	2008-02-27 02:05:37 +0200
@@ -10,7 +10,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 
 //===========================================================================//
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_sab.c mplayer-svn-r26107/libmpcodecs/vf_sab.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_sab.c	2008-02-27 01:48:45 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_sab.c	2008-02-27 02:05:37 +0200
@@ -33,7 +33,11 @@
 #include "img_format.h"
 #include "mp_image.h"
 #include "vf.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_scale.c mplayer-svn-r26107/libmpcodecs/vf_scale.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_scale.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_scale.c	2008-02-27 02:05:37 +0200
@@ -11,9 +11,13 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "fmt-conversion.h"
-#include "bswap.h"
-
+#include "mpbswap.h"
+  
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 #include "m_option.h"
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_screenshot.c mplayer-svn-r26107/libmpcodecs/vf_screenshot.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_screenshot.c	2008-02-27 01:48:45 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_screenshot.c	2008-02-27 02:08:33 +0200
@@ -19,8 +19,16 @@
 #include "vf.h"
 #include "vf_scale.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#else
 #include "libavcodec/avcodec.h"
+#endif
 
 struct vf_priv_s {
     int frameno;
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_smartblur.c mplayer-svn-r26107/libmpcodecs/vf_smartblur.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_smartblur.c	2008-02-27 01:48:45 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_smartblur.c	2008-02-27 02:05:37 +0200
@@ -33,7 +33,11 @@
 #include "img_format.h"
 #include "mp_image.h"
 #include "vf.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_unsharp.c mplayer-svn-r26107/libmpcodecs/vf_unsharp.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_unsharp.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_unsharp.c	2008-02-27 02:05:37 +0200
@@ -35,7 +35,11 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 //===========================================================================//
 
diff -urN mplayer-svn-r26107.orig/libmpcodecs/vf_yuy2.c mplayer-svn-r26107/libmpcodecs/vf_yuy2.c
--- mplayer-svn-r26107.orig/libmpcodecs/vf_yuy2.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26107/libmpcodecs/vf_yuy2.c	2008-02-27 02:05:37 +0200
@@ -11,7 +11,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r26107.orig/libmpdemux/asf.h mplayer-svn-r26107/libmpdemux/asf.h
--- mplayer-svn-r26107.orig/libmpdemux/asf.h	2008-02-23 00:42:57 +0200
+++ mplayer-svn-r26107/libmpdemux/asf.h	2008-02-27 02:05:37 +0200
@@ -3,7 +3,11 @@
 
 //#include "config.h"	/* for WORDS_BIGENDIAN */
 #include <inttypes.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 ///////////////////////
diff -urN mplayer-svn-r26107.orig/libmpdemux/asfheader.c mplayer-svn-r26107/libmpdemux/asfheader.c
--- mplayer-svn-r26107.orig/libmpdemux/asfheader.c	2007-07-26 01:19:56 +0300
+++ mplayer-svn-r26107/libmpdemux/asfheader.c	2008-02-27 02:05:37 +0200
@@ -6,8 +6,13 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/common.h"
 #include "libavutil/intreadwrite.h"
+#endif
 #include "mp_msg.h"
 #include "help_mp.h"
 
diff -urN mplayer-svn-r26107.orig/libmpdemux/aviheader.c mplayer-svn-r26107/libmpdemux/aviheader.c
--- mplayer-svn-r26107.orig/libmpdemux/aviheader.c	2007-12-02 17:50:51 +0200
+++ mplayer-svn-r26107/libmpdemux/aviheader.c	2008-02-27 02:05:37 +0200
@@ -13,7 +13,11 @@
 #include "stheader.h"
 
 #include "aviheader.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 static MainAVIHeader avih;
 
diff -urN mplayer-svn-r26107.orig/libmpdemux/aviheader.h mplayer-svn-r26107/libmpdemux/aviheader.h
--- mplayer-svn-r26107.orig/libmpdemux/aviheader.h	2008-02-23 00:42:57 +0200
+++ mplayer-svn-r26107/libmpdemux/aviheader.h	2008-02-27 02:05:37 +0200
@@ -1,8 +1,12 @@
 #ifndef MPLAYER_AVIHEADER_H
 #define MPLAYER_AVIHEADER_H
 
-//#include "config.h"	/* get correct definition WORDS_BIGENDIAN */
+#include "config.h"	/* get correct definition WORDS_BIGENDIAN */
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #ifndef mmioFOURCC
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_asf.c mplayer-svn-r26107/libmpdemux/demux_asf.c
--- mplayer-svn-r26107.orig/libmpdemux/demux_asf.c	2008-02-27 01:48:46 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_asf.c	2008-02-27 02:05:37 +0200
@@ -13,7 +13,11 @@
 #include "demuxer.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 // defined at asfheader.c:
 
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_audio.c mplayer-svn-r26107/libmpdemux/demux_audio.c
--- mplayer-svn-r26107.orig/libmpdemux/demux_audio.c	2008-02-06 00:33:19 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_audio.c	2008-02-27 02:05:37 +0200
@@ -10,7 +10,11 @@
 #include "stheader.h"
 #include "genres.h"
 #include "mp3_hdr.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 #include <string.h>
 
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_lavf.c mplayer-svn-r26107/libmpdemux/demux_lavf.c
--- mplayer-svn-r26107.orig/libmpdemux/demux_lavf.c	2008-02-27 01:48:46 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_lavf.c	2008-02-27 02:16:17 +0200
@@ -31,11 +31,31 @@
 #include "m_option.h"
 #include "libvo/sub.h"
 
+#ifdef USE_LIBAVFORMAT_SO
+#include <ffmpeg/avformat.h>
+#else
 #include "libavformat/avformat.h"
+#endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "libavutil/avutil.h"
+#endif
+#ifdef USE_LIBAVFORMAT_SO
+#include <ffmpeg/avi.h>
+#else
 #include "libavformat/avi.h"
+#endif
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/opt.h>
+#else
 #include "libavcodec/opt.h"
+#endif
+#ifdef USE_LIBAVFORMAT_SO
+#include <ffmpeg/riff.h>
+#else
 #include "libavformat/riff.h"
+#endif
 
 #define PROBE_BUF_SIZE 2048
 
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_mkv.c mplayer-svn-r26107/libmpdemux/demux_mkv.c
--- mplayer-svn-r26107.orig/libmpdemux/demux_mkv.c	2008-02-27 01:48:46 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_mkv.c	2008-02-27 02:18:03 +0200
@@ -29,7 +29,11 @@
 #include "libass/ass.h"
 #include "libass/ass_mp.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #ifdef USE_QTX_CODECS
 #include "loader/qtx/qtxsdk/components.h"
@@ -39,9 +43,15 @@
 #include <zlib.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/lzo.h>
+#include <ffmpeg/intreadwrite.h>
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/lzo.h"
 #include "libavutil/intreadwrite.h"
 #include "libavutil/avstring.h"
+#endif
 
 static unsigned char sipr_swaps[38][2]={
     {0,63},{1,22},{2,44},{3,90},{5,81},{7,31},{8,86},{9,58},{10,36},{12,68},
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_mov.c mplayer-svn-r26107/libmpdemux/demux_mov.c
--- mplayer-svn-r26107.orig/libmpdemux/demux_mov.c	2008-02-10 22:29:52 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_mov.c	2008-02-27 02:05:37 +0200
@@ -41,7 +41,11 @@
 #include "stheader.h"
 
 #include "libmpcodecs/img_format.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 #include "libvo/sub.h"
 
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_mpc.c mplayer-svn-r26107/libmpdemux/demux_mpc.c
--- mplayer-svn-r26107.orig/libmpdemux/demux_mpc.c	2008-02-06 00:33:19 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_mpc.c	2008-02-27 02:05:37 +0200
@@ -11,7 +11,11 @@
 #include <stdio.h>
 #include <string.h>
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "stream/stream.h"
 #include "demuxer.h"
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_ogg.c mplayer-svn-r26107/libmpdemux/demux_ogg.c
--- mplayer-svn-r26107.orig/libmpdemux/demux_ogg.c	2008-02-15 02:21:55 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_ogg.c	2008-02-27 02:05:37 +0200
@@ -13,7 +13,11 @@
 #include "stream/stream.h"
 #include "demuxer.h"
 #include "stheader.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 #define FOURCC_VORBIS mmioFOURCC('v', 'r', 'b', 's')
 #define FOURCC_SPEEX  mmioFOURCC('s', 'p', 'x', ' ')
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_rtp_codec.cpp mplayer-svn-r26107/libmpdemux/demux_rtp_codec.cpp
--- mplayer-svn-r26107.orig/libmpdemux/demux_rtp_codec.cpp	2008-02-15 02:21:55 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_rtp_codec.cpp	2008-02-27 02:05:37 +0200
@@ -6,7 +6,11 @@
 #include <limits.h>
 #include <math.h>
 #include "stheader.h"
-#include "base64.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/base64.h>
+#else
+#include "libavutil/base64.h"
+#endif
 }
 
 #ifdef USE_LIBAVCODEC
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_ty.c mplayer-svn-r26107/libmpdemux/demux_ty.c
--- mplayer-svn-r26107.orig/libmpdemux/demux_ty.c	2008-02-06 00:33:19 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_ty.c	2008-02-27 02:05:37 +0200
@@ -44,8 +44,14 @@
 #include "parse_es.h"
 #include "stheader.h"
 #include "sub_cc.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/avstring.h"
 #include "libavutil/intreadwrite.h"
+#endif
+
 
 extern void skip_audio_frame( sh_audio_t *sh_audio );
 extern int sub_justify;
diff -urN mplayer-svn-r26107.orig/libmpdemux/demux_vqf.c mplayer-svn-r26107/libmpdemux/demux_vqf.c
--- mplayer-svn-r26107.orig/libmpdemux/demux_vqf.c	2008-02-06 00:33:19 +0200
+++ mplayer-svn-r26107/libmpdemux/demux_vqf.c	2008-02-27 02:05:37 +0200
@@ -2,7 +2,11 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "stream/stream.h"
diff -urN mplayer-svn-r26107.orig/libmpdemux/ebml.c mplayer-svn-r26107/libmpdemux/ebml.c
--- mplayer-svn-r26107.orig/libmpdemux/ebml.c	2007-03-19 00:20:28 +0200
+++ mplayer-svn-r26107/libmpdemux/ebml.c	2008-02-27 02:05:37 +0200
@@ -11,9 +11,17 @@
 
 #include "stream/stream.h"
 #include "ebml.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intfloat_readwrite.h>
+#else
 #include "libavutil/intfloat_readwrite.h"
+#endif
 
 
 #ifndef SIZE_MAX
diff -urN mplayer-svn-r26107.orig/libmpdemux/muxer_lavf.c mplayer-svn-r26107/libmpdemux/muxer_lavf.c
--- mplayer-svn-r26107.orig/libmpdemux/muxer_lavf.c	2008-02-27 01:48:46 +0200
+++ mplayer-svn-r26107/libmpdemux/muxer_lavf.c	2008-02-27 02:18:57 +0200
@@ -17,8 +17,16 @@
 #include "demuxer.h"
 #include "stheader.h"
 #include "m_option.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avformat.h>
+#else
 #include "libavformat/avformat.h"
+#endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/avstring.h"
+#endif
 
 enum PixelFormat imgfmt2pixfmt(int fmt);
 
diff -urN mplayer-svn-r26107.orig/libmpdemux/nuppelvideo.h mplayer-svn-r26107/libmpdemux/nuppelvideo.h
--- mplayer-svn-r26107.orig/libmpdemux/nuppelvideo.h	2008-02-23 00:42:57 +0200
+++ mplayer-svn-r26107/libmpdemux/nuppelvideo.h	2008-02-27 02:05:37 +0200
@@ -3,7 +3,11 @@
 #ifndef MPLAYER_NUPPELVIDEO_H
 #define MPLAYER_NUPPELVIDEO_H
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intfloat_readwrite.h>
+#else
 #include "libavutil/intfloat_readwrite.h"
+#endif
 
 typedef struct  __attribute__((packed)) rtfileheader
 {
diff -urN mplayer-svn-r26107.orig/libswscale/rgb2rgb.c mplayer-svn-r26107/libswscale/rgb2rgb.c
--- mplayer-svn-r26107.orig/libswscale/rgb2rgb.c	2008-02-06 00:33:19 +0200
+++ mplayer-svn-r26107/libswscale/rgb2rgb.c	2008-02-27 02:05:37 +0200
@@ -31,7 +31,11 @@
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "x86_cpu.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 
 #define FAST_BGR2YV12 // use 7 bit coeffs instead of 15bit
 
diff -urN mplayer-svn-r26107.orig/libswscale/swscale.c mplayer-svn-r26107/libswscale/swscale.c
--- mplayer-svn-r26107.orig/libswscale/swscale.c	2008-02-23 00:42:56 +0200
+++ mplayer-svn-r26107/libswscale/swscale.c	2008-02-27 02:05:37 +0200
@@ -70,7 +70,11 @@
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "x86_cpu.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 #include "rgb2rgb.h"
 #include "libavcodec/opt.h"
 
diff -urN mplayer-svn-r26107.orig/libswscale/swscale-example.c mplayer-svn-r26107/libswscale/swscale-example.c
--- mplayer-svn-r26107.orig/libswscale/swscale-example.c	2008-01-21 04:18:27 +0200
+++ mplayer-svn-r26107/libswscale/swscale-example.c	2008-02-27 02:05:37 +0200
@@ -25,7 +25,11 @@
 #include <stdarg.h>
 
 #undef HAVE_AV_CONFIG_H
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "rgb2rgb.h"
diff -urN mplayer-svn-r26107.orig/libswscale/swscale_internal.h mplayer-svn-r26107/libswscale/swscale_internal.h
--- mplayer-svn-r26107.orig/libswscale/swscale_internal.h	2008-02-23 00:42:56 +0200
+++ mplayer-svn-r26107/libswscale/swscale_internal.h	2008-02-27 02:05:37 +0200
@@ -27,7 +27,11 @@
 #include <altivec.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 
 #define MAX_FILTER_SIZE 256
 
diff -urN mplayer-svn-r26107.orig/libvo/font_load_ft.c mplayer-svn-r26107/libvo/font_load_ft.c
--- mplayer-svn-r26107.orig/libvo/font_load_ft.c	2008-01-28 01:03:56 +0200
+++ mplayer-svn-r26107/libvo/font_load_ft.c	2008-02-27 02:05:37 +0200
@@ -28,7 +28,11 @@
 #include <fontconfig/fontconfig.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "font_load.h"
 #include "mp_msg.h"
diff -urN mplayer-svn-r26107.orig/libvo/mga_common.c mplayer-svn-r26107/libvo/mga_common.c
--- mplayer-svn-r26107.orig/libvo/mga_common.c	2008-02-15 02:21:55 +0200
+++ mplayer-svn-r26107/libvo/mga_common.c	2008-02-27 02:05:37 +0200
@@ -1,8 +1,12 @@
 
 #include "fastmemcpy.h"
 #include "cpudetect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #include "mp_msg.h"
 #include "help_mp.h"
diff -urN mplayer-svn-r26107.orig/libvo/sub.c mplayer-svn-r26107/libvo/sub.c
--- mplayer-svn-r26107.orig/libvo/sub.c	2008-01-28 01:03:56 +0200
+++ mplayer-svn-r26107/libvo/sub.c	2008-02-27 02:05:37 +0200
@@ -26,7 +26,12 @@
 #include "font_load.h"
 #include "sub.h"
 #include "spudec.h"
+
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #define NEW_SPLITTING
 
diff -urN mplayer-svn-r26107.orig/libvo/vo_aa.c mplayer-svn-r26107/libvo/vo_aa.c
--- mplayer-svn-r26107.orig/libvo/vo_aa.c	2007-12-02 17:50:50 +0200
+++ mplayer-svn-r26107/libvo/vo_aa.c	2008-02-27 02:05:37 +0200
@@ -27,7 +27,11 @@
 #include "video_out.h"
 #include "video_out_internal.h"
 #include "aspect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #include "font_load.h"
 #include "sub.h"
diff -urN mplayer-svn-r26107.orig/libvo/vo_dxr3.c mplayer-svn-r26107/libvo/vo_dxr3.c
--- mplayer-svn-r26107.orig/libvo/vo_dxr3.c	2008-01-11 02:30:36 +0200
+++ mplayer-svn-r26107/libvo/vo_dxr3.c	2008-02-27 02:05:37 +0200
@@ -38,7 +38,11 @@
 #ifdef HAVE_X11
 #include "x11_common.h"
 #endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/avstring.h"
+#endif
 
 #define SPU_SUPPORT
 
diff -urN mplayer-svn-r26107.orig/libvo/vo_vesa.c mplayer-svn-r26107/libvo/vo_vesa.c
--- mplayer-svn-r26107.orig/libvo/vo_vesa.c	2008-02-15 02:21:55 +0200
+++ mplayer-svn-r26107/libvo/vo_vesa.c	2008-02-27 02:05:37 +0200
@@ -39,7 +39,11 @@
 
 #include "fastmemcpy.h"
 #include "sub.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "aspect.h"
 #include "vesa_lvo.h"
@@ -48,7 +52,11 @@
 #endif
 #include "mp_msg.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 
 
diff -urN mplayer-svn-r26107.orig/libvo/vo_x11.c mplayer-svn-r26107/libvo/vo_x11.c
--- mplayer-svn-r26107.orig/libvo/vo_x11.c	2008-02-15 02:21:55 +0200
+++ mplayer-svn-r26107/libvo/vo_x11.c	2008-02-27 02:05:37 +0200
@@ -36,7 +36,11 @@
 
 #include "sub.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #define MODE_RGB  0x1
 #define MODE_BGR  0x2
diff -urN mplayer-svn-r26107.orig/libvo/vo_xv.c mplayer-svn-r26107/libvo/vo_xv.c
--- mplayer-svn-r26107.orig/libvo/vo_xv.c	2008-02-15 02:21:55 +0200
+++ mplayer-svn-r26107/libvo/vo_xv.c	2008-02-27 02:05:37 +0200
@@ -45,7 +45,11 @@
 #include "gui/interface.h"
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 static const vo_info_t info = {
     "X11/Xv",
diff -urN mplayer-svn-r26107.orig/libvo/vo_xvmc.c mplayer-svn-r26107/libvo/vo_xvmc.c
--- mplayer-svn-r26107.orig/libvo/vo_xvmc.c	2007-12-02 17:50:50 +0200
+++ mplayer-svn-r26107/libvo/vo_xvmc.c	2008-02-27 02:05:37 +0200
@@ -35,7 +35,11 @@
 #include "gui/interface.h"
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 //no chance for xinerama to be supported in the near future
 #undef HAVE_XINERAMA
@@ -46,7 +50,11 @@
 
 #define UNUSED(x) ((void)(x))
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#else
 #include "libavcodec/avcodec.h"
+#endif
 #if LIBAVCODEC_BUILD < ((51<<16)+(40<<8)+2)
 #error You need at least libavcodecs v51.40.2
 #endif
diff -urN mplayer-svn-r26107.orig/libvo/vo_yuv4mpeg.c mplayer-svn-r26107/libvo/vo_yuv4mpeg.c
--- mplayer-svn-r26107.orig/libvo/vo_yuv4mpeg.c	2007-12-02 17:50:50 +0200
+++ mplayer-svn-r26107/libvo/vo_yuv4mpeg.c	2008-02-27 02:05:37 +0200
@@ -40,10 +40,19 @@
 #include "sub.h"
 
 #include "fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/swscale.h"
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/rational.h>
+#else
 #include "libavutil/rational.h"
+#endif
 
 static const vo_info_t info = 
 {
diff -urN mplayer-svn-r26107.orig/Makefile mplayer-svn-r26107/Makefile
--- mplayer-svn-r26107.orig/Makefile	2008-02-27 01:48:47 +0200
+++ mplayer-svn-r26107/Makefile	2008-02-27 02:05:37 +0200
@@ -53,13 +53,14 @@
               libaf/libaf.a \
               libmpdemux/libmpdemux.a \
               stream/stream.a \
-              libswscale/libswscale.a \
               libvo/libosd.a \
 
 COMMON_LIBS-$(LIBAVFORMAT_A)      += libavformat/libavformat.a
 COMMON_LIBS-$(LIBAVCODEC_A)       += libavcodec/libavcodec.a
 COMMON_LIBS-$(LIBAVUTIL_A)        += libavutil/libavutil.a
 COMMON_LIBS-$(LIBPOSTPROC_A)      += libpostproc/libpostproc.a
+COMMON_LIBS-$(LIBSWSCALE_A)       += libswscale/libswscale.a
+PARTS-$(CONFIG_LIBSWSCALE)        += libswscale
 COMMON_LIBS-$(WIN32DLL)           += loader/libloader.a
 COMMON_LIBS-$(MP3LIB)             += mp3lib/libmp3.a
 COMMON_LIBS-$(LIBA52)             += liba52/liba52.a
diff -urN mplayer-svn-r26107.orig/m_option.c mplayer-svn-r26107/m_option.c
--- mplayer-svn-r26107.orig/m_option.c	2007-12-25 04:33:43 +0200
+++ mplayer-svn-r26107/m_option.c	2008-02-27 02:05:37 +0200
@@ -16,7 +16,11 @@
 //#include "m_config.h"
 #include "mp_msg.h"
 #include "stream/url.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/avstring.h"
+#endif
 
 // Don't free for 'production' atm
 #ifndef MP_DEBUG
diff -urN mplayer-svn-r26107.orig/mp3lib/dct64_sse.c mplayer-svn-r26107/mp3lib/dct64_sse.c
--- mplayer-svn-r26107.orig/mp3lib/dct64_sse.c	2007-12-17 18:36:04 +0200
+++ mplayer-svn-r26107/mp3lib/dct64_sse.c	2008-02-27 02:05:37 +0200
@@ -5,7 +5,12 @@
  * and mp3lib/dct64_mmx.c
  */
 
-#include <libavutil/mem.h>
+#include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/mem.h>
+#else
+#include "libavutil/mem.h"
+#endif
 
 typedef float real;
 
diff -urN mplayer-svn-r26107.orig/mp3lib/sr1.c mplayer-svn-r26107/mp3lib/sr1.c
--- mplayer-svn-r26107.orig/mp3lib/sr1.c	2008-01-28 01:03:56 +0200
+++ mplayer-svn-r26107/mp3lib/sr1.c	2008-02-27 02:05:37 +0200
@@ -19,7 +19,11 @@
 #include "mpg123.h"
 #include "huffman.h"
 #include "mp3.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "cpudetect.h"
 //#include "liba52/mm_accel.h"
diff -urN mplayer-svn-r26107.orig/mpbswap.h mplayer-svn-r26107/mpbswap.h
--- mplayer-svn-r26107.orig/mpbswap.h	2008-02-23 00:42:57 +0200
+++ mplayer-svn-r26107/mpbswap.h	2008-02-27 02:05:37 +0200
@@ -1,7 +1,12 @@
 #ifndef MPLAYER_MPBSWAP_H
 #define MPLAYER_MPBSWAP_H
 
+#include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
 #include "libavutil/bswap.h"
+#endif
 #ifndef HAVE_SWAB
 void swab(const void *from, void *to, ssize_t n);
 #endif
diff -urN mplayer-svn-r26107.orig/mpcommon.c mplayer-svn-r26107/mpcommon.c
--- mplayer-svn-r26107.orig/mpcommon.c	2008-01-28 01:03:57 +0200
+++ mplayer-svn-r26107/mpcommon.c	2008-02-27 02:05:37 +0200
@@ -10,7 +10,11 @@
 #ifdef HAVE_TV_TELETEXT
 #include "stream/tv.h"
 #endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 double sub_last_pts = -303;
 
diff -urN mplayer-svn-r26107.orig/mplayer.c mplayer-svn-r26107/mplayer.c
--- mplayer-svn-r26107.orig/mplayer.c	2008-02-23 00:42:57 +0200
+++ mplayer-svn-r26107/mplayer.c	2008-02-27 02:05:37 +0200
@@ -49,7 +49,11 @@
 
 #include "cfg-mplayer-def.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 #include "subreader.h"
 
diff -urN mplayer-svn-r26107.orig/spudec.c mplayer-svn-r26107/spudec.c
--- mplayer-svn-r26107.orig/spudec.c	2008-02-27 01:48:47 +0200
+++ mplayer-svn-r26107/spudec.c	2008-02-27 02:06:53 +0200
@@ -23,8 +23,16 @@
 #include <math.h>
 #include "libvo/video_out.h"
 #include "spudec.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "libavutil/avutil.h"
+#endif
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 
 /* Valid values for spu_aamode:
    0: none (fastest, most ugly)
diff -urN mplayer-svn-r26107.orig/stream/asf_streaming.c mplayer-svn-r26107/stream/asf_streaming.c
--- mplayer-svn-r26107.orig/stream/asf_streaming.c	2008-02-06 00:33:17 +0200
+++ mplayer-svn-r26107/stream/asf_streaming.c	2008-02-27 02:05:37 +0200
@@ -25,7 +25,11 @@
 #include "network.h"
 #include "tcp.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 #ifdef ARCH_X86
 #define	ASF_LOAD_GUID_PREFIX(guid)	(*(uint32_t *)(guid))
diff -urN mplayer-svn-r26107.orig/stream/pnm.c mplayer-svn-r26107/stream/pnm.c
--- mplayer-svn-r26107.orig/stream/pnm.c	2008-01-21 04:18:26 +0200
+++ mplayer-svn-r26107/stream/pnm.c	2008-02-27 02:05:37 +0200
@@ -44,7 +44,11 @@
 #include <winsock2.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 #include "stream.h"
 #include "libmpdemux/demuxer.h"
diff -urN mplayer-svn-r26107.orig/stream/realrtsp/real.c mplayer-svn-r26107/stream/realrtsp/real.c
--- mplayer-svn-r26107.orig/stream/realrtsp/real.c	2008-02-27 01:48:45 +0200
+++ mplayer-svn-r26107/stream/realrtsp/real.c	2008-02-27 02:10:23 +0200
@@ -36,8 +36,13 @@
 #include "asmrp.h"
 #include "sdpplin.h"
 #include "xbuffer.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/md5.h>
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/md5.h"
 #include "libavutil/intreadwrite.h"
+#endif
 #include "../http.h"
 #include "mp_msg.h"
 
diff -urN mplayer-svn-r26107.orig/stream/realrtsp/rmff.c mplayer-svn-r26107/stream/realrtsp/rmff.c
--- mplayer-svn-r26107.orig/stream/realrtsp/rmff.c	2007-03-27 00:16:50 +0300
+++ mplayer-svn-r26107/stream/realrtsp/rmff.c	2008-02-27 02:05:37 +0200
@@ -29,7 +29,11 @@
 #include "rmff.h"
 #include "xbuffer.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 /*
 #define LOG
diff -urN mplayer-svn-r26107.orig/stream/stream_cdda.c mplayer-svn-r26107/stream/stream_cdda.c
--- mplayer-svn-r26107.orig/stream/stream_cdda.c	2008-01-21 04:18:26 +0200
+++ mplayer-svn-r26107/stream/stream_cdda.c	2008-02-27 02:05:37 +0200
@@ -6,7 +6,11 @@
 #include "stream.h"
 #include "m_option.h"
 #include "m_struct.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libmpdemux/demuxer.h"
 
diff -urN mplayer-svn-r26107.orig/stream/stream_cddb.c mplayer-svn-r26107/stream/stream_cddb.c
--- mplayer-svn-r26107.orig/stream/stream_cddb.c	2008-02-12 02:04:35 +0200
+++ mplayer-svn-r26107/stream/stream_cddb.c	2008-02-27 02:05:37 +0200
@@ -58,7 +58,11 @@
 #include "version.h"
 #include "stream.h"
 #include "network.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #define DEFAULT_FREEDB_SERVER	"freedb.freedb.org"
 #define DEFAULT_CACHE_DIR	"/.cddb/"
diff -urN mplayer-svn-r26107.orig/stream/stream_cue.c mplayer-svn-r26107/stream/stream_cue.c
--- mplayer-svn-r26107.orig/stream/stream_cue.c	2008-01-21 04:18:26 +0200
+++ mplayer-svn-r26107/stream/stream_cue.c	2008-02-27 02:05:37 +0200
@@ -18,7 +18,11 @@
 #include "help_mp.h"
 #include "m_option.h"
 #include "m_struct.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/avstring.h"
+#endif
 
 #define byte    unsigned char
 #define SIZERAW 2352
diff -urN mplayer-svn-r26107.orig/stream/stream_dvb.c mplayer-svn-r26107/stream/stream_dvb.c
--- mplayer-svn-r26107.orig/stream/stream_dvb.c	2008-01-21 04:18:26 +0200
+++ mplayer-svn-r26107/stream/stream_dvb.c	2008-02-27 02:05:37 +0200
@@ -45,7 +45,12 @@
 #include "m_option.h"
 #include "m_struct.h"
 #include "get_path.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/avstring.h"
+#endif
+
 
 #include "dvbin.h"
 
diff -urN mplayer-svn-r26107.orig/stream/stream_dvd.c mplayer-svn-r26107/stream/stream_dvd.c
--- mplayer-svn-r26107.orig/stream/stream_dvd.c	2008-02-12 02:04:35 +0200
+++ mplayer-svn-r26107/stream/stream_dvd.c	2008-02-27 02:05:37 +0200
@@ -39,7 +39,11 @@
 #include "stream_dvd.h"
 #include "stream_dvd_common.h"
 #include "libmpdemux/demuxer.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 extern int stream_cache_size;
 extern char* dvd_device;
diff -urN mplayer-svn-r26107.orig/stream/stream_dvdnav.c mplayer-svn-r26107/stream/stream_dvdnav.c
--- mplayer-svn-r26107.orig/stream/stream_dvdnav.c	2008-02-12 02:04:35 +0200
+++ mplayer-svn-r26107/stream/stream_dvdnav.c	2008-02-27 02:05:37 +0200
@@ -13,7 +13,11 @@
 #include <dvdnav/dvdnav.h>
 #include "stream_dvdnav.h"
 #include "libvo/video_out.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "spudec.h"
 #include "m_option.h"
 #include "m_struct.h"
diff -urN mplayer-svn-r26107.orig/stream/stream_netstream.c mplayer-svn-r26107/stream/stream_netstream.c
--- mplayer-svn-r26107.orig/stream/stream_netstream.c	2008-01-21 04:18:26 +0200
+++ mplayer-svn-r26107/stream/stream_netstream.c	2008-02-27 02:05:37 +0200
@@ -58,7 +58,11 @@
 #include "help_mp.h"
 #include "m_option.h"
 #include "m_struct.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "netstream.h"
diff -urN mplayer-svn-r26107.orig/stream/stream_pvr.c mplayer-svn-r26107/stream/stream_pvr.c
--- mplayer-svn-r26107.orig/stream/stream_pvr.c	2007-12-02 17:50:49 +0200
+++ mplayer-svn-r26107/stream/stream_pvr.c	2008-02-27 02:05:37 +0200
@@ -44,8 +44,13 @@
 #include "pvr.h"
 
 #include "frequencies.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/common.h"
 #include "libavutil/avstring.h"
+#endif
 
 #define PVR_DEFAULT_DEVICE "/dev/video0"
 #define PVR_MAX_CONTROLS 10
diff -urN mplayer-svn-r26107.orig/stream/stream_radio.c mplayer-svn-r26107/stream/stream_radio.c
--- mplayer-svn-r26107.orig/stream/stream_radio.c	2008-02-15 02:21:54 +0200
+++ mplayer-svn-r26107/stream/stream_radio.c	2008-02-27 02:05:37 +0200
@@ -63,7 +63,11 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "stream_radio.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/avstring.h"
+#endif
 
 #ifdef USE_RADIO_CAPTURE
 #include "audio_in.h"
diff -urN mplayer-svn-r26107.orig/stream/tv.c mplayer-svn-r26107/stream/tv.c
--- mplayer-svn-r26107.orig/stream/tv.c	2008-01-21 04:18:26 +0200
+++ mplayer-svn-r26107/stream/tv.c	2008-02-27 02:05:37 +0200
@@ -29,7 +29,11 @@
 
 #include "libaf/af_format.h"
 #include "libmpcodecs/img_format.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/avstring.h"
+#endif
 #include "osdep/timer.h"
 
 #include "tv.h"
diff -urN mplayer-svn-r26107.orig/stream/tvi_vbi.c mplayer-svn-r26107/stream/tvi_vbi.c
--- mplayer-svn-r26107.orig/stream/tvi_vbi.c	2007-12-03 01:37:09 +0200
+++ mplayer-svn-r26107/stream/tvi_vbi.c	2008-02-27 02:05:37 +0200
@@ -94,7 +94,11 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "libmpcodecs/img_format.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "input/input.h"
 #include "osdep/timer.h"
 
diff -urN mplayer-svn-r26107.orig/stream/vcd_read.h mplayer-svn-r26107/stream/vcd_read.h
--- mplayer-svn-r26107.orig/stream/vcd_read.h	2008-02-23 00:42:53 +0200
+++ mplayer-svn-r26107/stream/vcd_read.h	2008-02-27 02:07:32 +0200
@@ -1,7 +1,11 @@
 #ifndef MPLAYER_VCD_READ_H
 #define MPLAYER_VCD_READ_H
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 //=================== VideoCD ==========================
 #if	defined(linux) || defined(sun) || defined(__bsdi__)
 
diff -urN mplayer-svn-r26107.orig/TOOLS/Makefile mplayer-svn-r26107/TOOLS/Makefile
--- mplayer-svn-r26107.orig/TOOLS/Makefile	2008-01-24 02:33:26 +0200
+++ mplayer-svn-r26107/TOOLS/Makefile	2008-02-27 02:05:37 +0200
@@ -13,14 +13,15 @@
 	subrip$(EXESUF) \
 #	vivodump$(EXESUF) \
 
-COMMON_LIBS = ../mp_msg.o ../mp_fifo.o ../osdep/libosdep.a -ltermcap -lm
+COMMON_LIBS = ../mp_msg.o ../mp_fifo.o ../osdep/libosdep.a -ltinfo -lm
 
 all: $(OBJS)
 
 subrip$(EXESUF): subrip.c
 	$(CC) $(CFLAGS) -o $@ $< ../vobsub.o ../spudec.o \
-          ../unrar_exec.o ../libswscale/libswscale.a ../libavutil/libavutil.a \
-          ../libmpcodecs/img_format.o $(COMMON_LIBS)
+          ../unrarlib.o ../libmpcodecs/img_format.o $(COMMON_LIBS) \
+          -L../libswscale -lswscale -L../libavutil -lavutil
+
 
 #FIXME: Linking is broken, help welcome.
 vivodump$(EXESUF): vivodump.c
diff -urN mplayer-svn-r26107.orig/vidix/drivers.c mplayer-svn-r26107/vidix/drivers.c
--- mplayer-svn-r26107.orig/vidix/drivers.c	2007-05-11 01:00:46 +0300
+++ mplayer-svn-r26107/vidix/drivers.c	2008-02-27 02:05:37 +0200
@@ -26,7 +26,11 @@
 
 #include "vidixlib.h"
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "config.h"
 
diff -urN mplayer-svn-r26107.orig/vidix/mach64_vid.c mplayer-svn-r26107/vidix/mach64_vid.c
--- mplayer-svn-r26107.orig/vidix/mach64_vid.c	2008-01-28 01:03:56 +0200
+++ mplayer-svn-r26107/vidix/mach64_vid.c	2008-02-27 02:05:37 +0200
@@ -31,7 +31,11 @@
 #include <fcntl.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "vidix.h"
 #include "vidixlib.h"
diff -urN mplayer-svn-r26107.orig/vidix/nvidia_vid.c mplayer-svn-r26107/vidix/nvidia_vid.c
--- mplayer-svn-r26107.orig/vidix/nvidia_vid.c	2007-10-02 15:01:56 +0300
+++ mplayer-svn-r26107/vidix/nvidia_vid.c	2008-02-27 02:05:37 +0200
@@ -38,7 +38,11 @@
 #include "pci_ids.h"
 #include "pci_names.h"
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 
diff -urN mplayer-svn-r26107.orig/vidix/radeon_vid.c mplayer-svn-r26107/vidix/radeon_vid.c
--- mplayer-svn-r26107.orig/vidix/radeon_vid.c	2008-01-28 01:03:56 +0200
+++ mplayer-svn-r26107/vidix/radeon_vid.c	2008-02-27 02:05:37 +0200
@@ -37,7 +37,11 @@
 #include <inttypes.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "pci_ids.h"
 #include "pci_names.h"
diff -urN mplayer-svn-r26107.orig/vidix/sysdep/pci_linux.c mplayer-svn-r26107/vidix/sysdep/pci_linux.c
--- mplayer-svn-r26107.orig/vidix/sysdep/pci_linux.c	2007-12-02 17:50:49 +0200
+++ mplayer-svn-r26107/vidix/sysdep/pci_linux.c	2008-02-27 02:05:37 +0200
@@ -155,7 +155,11 @@
 #include <fcntl.h>
 //#include <sys/io.h>
 #include <linux/pci.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 static int pci_get_vendor(
diff -urN mplayer-svn-r26107.orig/vidix/vidixlib.c mplayer-svn-r26107/vidix/vidixlib.c
--- mplayer-svn-r26107.orig/vidix/vidixlib.c	2007-04-23 01:35:47 +0300
+++ mplayer-svn-r26107/vidix/vidixlib.c	2008-02-27 02:05:37 +0200
@@ -40,7 +40,11 @@
 #include "vidixlib.h"
 #include "drivers.h"
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 extern unsigned int  vdlGetVersion( void )
diff -urN mplayer-svn-r26107.orig/vobsub.c mplayer-svn-r26107/vobsub.c
--- mplayer-svn-r26107.orig/vobsub.c	2008-01-21 04:18:30 +0200
+++ mplayer-svn-r26107/vobsub.c	2008-02-27 02:05:37 +0200
@@ -23,7 +23,11 @@
 #ifdef USE_UNRAR_EXEC
 #include "unrar_exec.h"
 #endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 extern int vobsub_id;
 // Record the original -vobsubid set by commandline, since vobsub_id will be
