diff -urN mplayer-svn-r20117.orig/configure mplayer-svn-r20117/configure
--- mplayer-svn-r20117.orig/configure	2006-10-09 11:28:59 +0300
+++ mplayer-svn-r20117/configure	2006-10-09 12:03:19 +0300
@@ -5271,6 +5271,8 @@
 fi
 
 
+_ld_swscale="-lswscale"
+
 echocheck "cdparanoia"
 if test "$_cdparanoia" = auto ; then
     cat > $TMPC <<EOF
@@ -7541,6 +7543,7 @@
 DVDKIT2 = $_mpdvdkit
 DVDNAV = $_dvdnav
 DVDNAV_LIB = $_ld_dvdnav
+SWSCALE_LIB = $_ld_swscale
 WIN32DLL = $_win32
 W32_DEP = $_dep_win32
 W32_LIB = $_ld_win32
diff -urN mplayer-svn-r20117.orig/fmt-conversion.h mplayer-svn-r20117/fmt-conversion.h
--- mplayer-svn-r20117.orig/fmt-conversion.h	2006-09-24 19:47:48 +0300
+++ mplayer-svn-r20117/fmt-conversion.h	2006-10-09 12:04:41 +0300
@@ -1,4 +1,8 @@
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 #include "img_format.h"
 
 enum PixelFormat imgfmt2pixfmt(int fmt)
diff -urN mplayer-svn-r20117.orig/Gui/wm/ws.c mplayer-svn-r20117/Gui/wm/ws.c
--- mplayer-svn-r20117.orig/Gui/wm/ws.c	2006-07-12 16:44:27 +0300
+++ mplayer-svn-r20117/Gui/wm/ws.c	2006-10-09 12:06:50 +0300
@@ -24,8 +24,8 @@
 #include "ws.h"
 #include "wsxdnd.h"
 #include "../cpudetect.h"
-#include "../libswscale/swscale.h"
-#include "../libswscale/rgb2rgb.h"
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
 #include "../libmpcodecs/vf_scale.h"
 #include "../mp_msg.h"
 #include "../help_mp.h"
diff -urN mplayer-svn-r20117.orig/libmpcodecs/vf_halfpack.c mplayer-svn-r20117/libmpcodecs/vf_halfpack.c
--- mplayer-svn-r20117.orig/libmpcodecs/vf_halfpack.c	2006-08-17 13:36:26 +0300
+++ mplayer-svn-r20117/libmpcodecs/vf_halfpack.c	2006-10-09 12:07:29 +0300
@@ -12,7 +12,7 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/rgb2rgb.h"
+#include <ffmpeg/rgb2rgb.h>
 
 struct vf_priv_s {
 	int field;
diff -urN mplayer-svn-r20117.orig/libmpcodecs/vf_palette.c mplayer-svn-r20117/libmpcodecs/vf_palette.c
--- mplayer-svn-r20117.orig/libmpcodecs/vf_palette.c	2006-07-03 14:16:54 +0300
+++ mplayer-svn-r20117/libmpcodecs/vf_palette.c	2006-10-09 12:20:25 +0300
@@ -11,7 +11,7 @@
 #include "mp_image.h"
 #include "vf.h"
 
-#include "libswscale/rgb2rgb.h"
+#include <ffmpeg/rgb2rgb.h>
 
 //===========================================================================//
 
diff -urN mplayer-svn-r20117.orig/libmpcodecs/vf_qp.c mplayer-svn-r20117/libmpcodecs/vf_qp.c
--- mplayer-svn-r20117.orig/libmpcodecs/vf_qp.c	2006-06-13 15:10:35 +0300
+++ mplayer-svn-r20117/libmpcodecs/vf_qp.c	2006-10-09 12:21:08 +0300
@@ -37,9 +37,15 @@
 // Needed to bring in lrintf.
 #define HAVE_AV_CONFIG_H
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#include <ffmpeg/dsputil.h>
+#include <ffmpeg/common.h>
+#else
 #include "libavcodec/avcodec.h"
 #include "libavcodec/dsputil.h"
 #include "libavutil/common.h"
+#endif
 
 /* FIXME: common.h defines printf away when HAVE_AV_CONFIG
  * is defined, but mp_image.h needs printf.
diff -urN mplayer-svn-r20117.orig/libmpcodecs/vf_rgb2bgr.c mplayer-svn-r20117/libmpcodecs/vf_rgb2bgr.c
--- mplayer-svn-r20117.orig/libmpcodecs/vf_rgb2bgr.c	2006-07-03 14:16:54 +0300
+++ mplayer-svn-r20117/libmpcodecs/vf_rgb2bgr.c	2006-10-09 12:23:35 +0300
@@ -11,7 +11,7 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/rgb2rgb.h"
+#include <ffmpeg/rgb2rgb.h>
 
 //===========================================================================//
 
diff -urN mplayer-svn-r20117.orig/libmpcodecs/vf_sab.c mplayer-svn-r20117/libmpcodecs/vf_sab.c
--- mplayer-svn-r20117.orig/libmpcodecs/vf_sab.c	2006-09-18 11:03:58 +0300
+++ mplayer-svn-r20117/libmpcodecs/vf_sab.c	2006-10-09 12:24:34 +0300
@@ -29,12 +29,16 @@
 #include <malloc.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 #include "img_format.h"
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "vf_scale.h"
 
 
diff -urN mplayer-svn-r20117.orig/libmpcodecs/vf_scale.c mplayer-svn-r20117/libmpcodecs/vf_scale.c
--- mplayer-svn-r20117.orig/libmpcodecs/vf_scale.c	2006-09-18 11:03:58 +0300
+++ mplayer-svn-r20117/libmpcodecs/vf_scale.c	2006-10-09 12:25:06 +0300
@@ -13,7 +13,7 @@
 #include "fmt-conversion.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "vf_scale.h"
 
 #include "m_option.h"
diff -urN mplayer-svn-r20117.orig/libmpcodecs/vf_screenshot.c mplayer-svn-r20117/libmpcodecs/vf_screenshot.c
--- mplayer-svn-r20117.orig/libmpcodecs/vf_screenshot.c	2006-09-18 11:03:58 +0300
+++ mplayer-svn-r20117/libmpcodecs/vf_screenshot.c	2006-10-09 12:25:24 +0300
@@ -22,7 +22,7 @@
 #include "vf_scale.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 
 struct vf_priv_s {
     int frameno;
diff -urN mplayer-svn-r20117.orig/libmpcodecs/vf_smartblur.c mplayer-svn-r20117/libmpcodecs/vf_smartblur.c
--- mplayer-svn-r20117.orig/libmpcodecs/vf_smartblur.c	2006-09-18 11:03:58 +0300
+++ mplayer-svn-r20117/libmpcodecs/vf_smartblur.c	2006-10-09 12:26:12 +0300
@@ -29,12 +29,17 @@
 #include <malloc.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
+
 #include "img_format.h"
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r20117.orig/libmpcodecs/vf_yuy2.c mplayer-svn-r20117/libmpcodecs/vf_yuy2.c
--- mplayer-svn-r20117.orig/libmpcodecs/vf_yuy2.c	2006-07-03 14:16:54 +0300
+++ mplayer-svn-r20117/libmpcodecs/vf_yuy2.c	2006-10-09 12:26:41 +0300
@@ -12,7 +12,7 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
-#include "libswscale/rgb2rgb.h"
+#include <ffmpeg/swscale.h>
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r20117.orig/libswscale/swscale-example.c mplayer-svn-r20117/libswscale/swscale-example.c
--- mplayer-svn-r20117.orig/libswscale/swscale-example.c	2006-10-09 11:28:56 +0300
+++ mplayer-svn-r20117/libswscale/swscale-example.c	2006-10-09 12:29:31 +0300
@@ -25,10 +25,14 @@
 #include <stdarg.h>
 
 #undef HAVE_AV_CONFIG_H
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
-#include "swscale.h"
+#endif
+#include <ffmpeg/swscale.h>
 #include "swscale_internal.h"
-#include "rgb2rgb.h"
+#include <ffmpeg/rgb2rgb.h>
 
 static uint64_t getSSD(uint8_t *src1, uint8_t *src2, int stride1, int stride2, int w, int h){
 	int x,y;
diff -urN mplayer-svn-r20117.orig/libswscale/swscale_internal.h mplayer-svn-r20117/libswscale/swscale_internal.h
--- mplayer-svn-r20117.orig/libswscale/swscale_internal.h	2006-10-09 11:28:56 +0300
+++ mplayer-svn-r20117/libswscale/swscale_internal.h	2006-10-09 12:30:00 +0300
@@ -28,7 +28,11 @@
 #include <altivec.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 
 #ifdef CONFIG_DARWIN
 #define AVV(x...) (x)
diff -urN mplayer-svn-r20117.orig/libvo/mga_common.c mplayer-svn-r20117/libvo/mga_common.c
--- mplayer-svn-r20117.orig/libvo/mga_common.c	2006-07-03 14:16:55 +0300
+++ mplayer-svn-r20117/libvo/mga_common.c	2006-10-09 12:30:29 +0300
@@ -1,7 +1,7 @@
 
 #include "fastmemcpy.h"
 #include "cpudetect.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 #include "mp_msg.h"
 #include "help_mp.h"
diff -urN mplayer-svn-r20117.orig/libvo/sub.c mplayer-svn-r20117/libvo/sub.c
--- mplayer-svn-r20117.orig/libvo/sub.c	2006-08-17 13:36:33 +0300
+++ mplayer-svn-r20117/libvo/sub.c	2006-10-09 12:31:28 +0300
@@ -14,7 +14,12 @@
 #include "font_load.h"
 #include "sub.h"
 #include "spudec.h"
+
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #define NEW_SPLITTING
 
diff -urN mplayer-svn-r20117.orig/libvo/vo_aa.c mplayer-svn-r20117/libvo/vo_aa.c
--- mplayer-svn-r20117.orig/libvo/vo_aa.c	2006-07-10 14:48:33 +0300
+++ mplayer-svn-r20117/libvo/vo_aa.c	2006-10-09 12:31:57 +0300
@@ -27,7 +27,7 @@
 #include "video_out.h"
 #include "video_out_internal.h"
 #include "aspect.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 #include "font_load.h"
 #include "sub.h"
diff -urN mplayer-svn-r20117.orig/libvo/vo_vesa.c mplayer-svn-r20117/libvo/vo_vesa.c
--- mplayer-svn-r20117.orig/libvo/vo_vesa.c	2006-08-28 14:40:31 +0300
+++ mplayer-svn-r20117/libvo/vo_vesa.c	2006-10-09 12:41:19 +0300
@@ -47,7 +47,7 @@
 #endif
 #include "mp_msg.h"
 
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 
 
diff -urN mplayer-svn-r20117.orig/libvo/vo_x11.c mplayer-svn-r20117/libvo/vo_x11.c
--- mplayer-svn-r20117.orig/libvo/vo_x11.c	2006-07-10 14:48:33 +0300
+++ mplayer-svn-r20117/libvo/vo_x11.c	2006-10-09 12:33:29 +0300
@@ -38,7 +38,7 @@
 #include "fastmemcpy.h"
 #include "sub.h"
 
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 #define MODE_RGB  0x1
 #define MODE_BGR  0x2
diff -urN mplayer-svn-r20117.orig/libvo/vo_yuv4mpeg.c mplayer-svn-r20117/libvo/vo_yuv4mpeg.c
--- mplayer-svn-r20117.orig/libvo/vo_yuv4mpeg.c	2006-07-03 14:16:55 +0300
+++ mplayer-svn-r20117/libvo/vo_yuv4mpeg.c	2006-10-09 12:34:06 +0300
@@ -39,7 +39,7 @@
 #include "sub.h"
 
 #include "fastmemcpy.h"
-#include "libswscale/swscale.h"
+#include <ffmpeg/swscale.h>
 #include "libmpcodecs/vf_scale.h"
 
 static vo_info_t info = 
diff -urN mplayer-svn-r20117.orig/Makefile mplayer-svn-r20117/Makefile
--- mplayer-svn-r20117.orig/Makefile	2006-10-09 11:28:59 +0300
+++ mplayer-svn-r20117/Makefile	2006-10-09 12:41:45 +0300
@@ -132,7 +132,7 @@
               libaf/libaf.a \
               libmpdemux/libmpdemux.a \
               stream/stream.a \
-              libswscale/libswscale.a \
+              $(SWSCALE_LIB) \
               osdep/libosdep.a \
               $(EXTRA_LIB)\
               $(DVDREAD_LIB) \
@@ -163,7 +163,6 @@
         libavcodec \
         libpostproc \
         libavformat \
-        libswscale \
         libao2 \
         osdep \
         input \
@@ -217,7 +216,6 @@
               libmpcodecs/libmpcodecs.a \
               libao2/libao2.a \
               osdep/libosdep.a \
-              libswscale/libswscale.a \
               input/libinput.a \
               libvo/libvo.a \
               libaf/libaf.a \
@@ -317,9 +315,6 @@
 libavformat/libavformat.a:
 	$(MAKE) -C libavformat LIBPREF=lib LIBSUF=.a
 
-libswscale/libswscale.a:
-	$(MAKE) -C libswscale LIBPREF=lib LIBSUF=.a
-
 libmpeg2/libmpeg2.a:
 	$(MAKE) -C libmpeg2
 
@@ -564,7 +559,6 @@
 libavutil/libavutil.a: .norecurse $(wildcard libavutil/*.[ch])
 libavcodec/libavcodec.a: .norecurse $(wildcard libavcodec/*.[ch] libavcodec/*/*.[chS])
 libavformat/libavformat.a: .norecurse $(wildcard libavformat/*.[ch])
-libswscale/libswscale.a: .norecurse $(wildcard libswscale/*.[ch])
 
 libmpeg2/libmpeg2.a: .norecurse $(wildcard libmpeg2/*.[ch])
 liba52/liba52.a: .norecurse $(wildcard liba52/*.[ch])
diff -urN mplayer-svn-r20117.orig/spudec.c mplayer-svn-r20117/spudec.c
--- mplayer-svn-r20117.orig/spudec.c	2006-09-18 11:04:02 +0300
+++ mplayer-svn-r20117/spudec.c	2006-10-09 12:37:18 +0300
@@ -23,8 +23,12 @@
 #include <math.h>
 #include "libvo/video_out.h"
 #include "spudec.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
-#include "libswscale/swscale.h"
+#endif
+#include <ffmpeg/swscale.h>
 
 #define MIN(a, b)	((a)<(b)?(a):(b))
 
diff -urN mplayer-svn-r20117.orig/TOOLS/Makefile mplayer-svn-r20117/TOOLS/Makefile
--- mplayer-svn-r20117.orig/TOOLS/Makefile	2006-08-17 13:36:37 +0300
+++ mplayer-svn-r20117/TOOLS/Makefile	2006-10-09 12:38:05 +0300
@@ -27,8 +27,7 @@
 
 subrip: subrip.c
 	$(CC) $(CFLAGS) -g -o $@ $< ../vobsub.o ../spudec.o ../mp_msg.o \
-          ../unrarlib.o ../libswscale/swscale.o ../libswscale/rgb2rgb.o \
-          ../libswscale/yuv2rgb.o ../libmpcodecs/img_format.o -lm
+          ../unrarlib.o ../libmpcodecs/img_format.o -lswscale -lm
 
 vivodump: vivodump.c
 	$(CC) $(CFLAGS) -o $@ $< ../mp_msg.o ../libmpdemux/libmpdemux.a ../stream/stream.a
diff -urN mplayer-svn-r20117.orig/TOOLS/mwallp/compile.sh mplayer-svn-r20117/TOOLS/mwallp/compile.sh
--- mplayer-svn-r20117.orig/TOOLS/mwallp/compile.sh	2006-07-03 14:16:57 +0300
+++ mplayer-svn-r20117/TOOLS/mwallp/compile.sh	2006-10-09 12:39:47 +0300
@@ -1,6 +1,6 @@
 echo
 gcc -g mwallp.c jpeg.c ../../libvo/aclib.o ../../libmpcodecs/img_format.o \
     ../../cpudetect.o ../../mp_msg-mencoder.o ../../libavcodec/libavcodec.a \
-    ../../libswscale/libswscale.a ../../osdep/libosdep.a \
-    -I../../libmpcodecs -I../.. -I../../libswscale \
-    -L/usr/X11/lib -L/usr/X11R6/lib -lX11 -lm -ljpeg -o mwallp
+    ../../osdep/libosdep.a \
+    -I../../libmpcodecs -I../.. \
+    -L/usr/X11/lib -L/usr/X11R6/lib -lswscale -lX11 -lm -ljpeg -o mwallp
diff -urN mplayer-svn-r20117.orig/TOOLS/mwallp/jpeg.c mplayer-svn-r20117/TOOLS/mwallp/jpeg.c
--- mplayer-svn-r20117.orig/TOOLS/mwallp/jpeg.c	2006-06-13 15:12:11 +0300
+++ mplayer-svn-r20117/TOOLS/mwallp/jpeg.c	2006-10-09 12:40:11 +0300
@@ -13,8 +13,8 @@
 
 #include "img_format.h"
 
-#include "swscale.h"
-#include "rgb2rgb.h"
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
 
 static struct SwsContext *swsContext=NULL;
 
diff -urN mplayer-svn-r20117.orig/TOOLS/subrip.c mplayer-svn-r20117/TOOLS/subrip.c
--- mplayer-svn-r20117.orig/TOOLS/subrip.c	2006-07-03 14:16:57 +0300
+++ mplayer-svn-r20117/TOOLS/subrip.c	2006-10-09 12:40:43 +0300
@@ -1,4 +1,4 @@
-/* -*- compile-command: "gcc -g -Wall -I.. -o subrip subrip.c ../vobsub.o ../spudec.o ../mp_msg.o ../unrarlib.o ../libswscale/swscale.o ../libswscale/rgb2rgb.o ../libswscale/yuv2rgb.o ../libmpcodecs/img_format.o -lm" -*- */
+/* -*- compile-command: "gcc -g -Wall -I.. -o subrip subrip.c ../vobsub.o ../spudec.o ../mp_msg.o ../unrarlib.o ../libmpcodecs/img_format.o -lswscale -lm" -*- */
 /*
  * Use with CVS JOCR/GOCR.
  *
