--- MPlayer-1.0pre4.orig/configure	2004-05-03 00:18:16 +0700
+++ MPlayer-1.0pre4/configure	2004-05-03 00:57:35 +0700
@@ -3595,6 +3595,20 @@
     _gif=yes
     _ld_gif="-lgif $_ld_x11"
   fi
+elif test "$_gif" = yes ; then
+  # check existance of gif/ungif lib and complain if it can' be found...
+  if cc_check -lungif && "$TMPO" >> "$TMPLOG" ; then
+    _ld_gif="-lungif"
+  elif cc_check -lungif $_ld_x11 && "$TMPO" >> "$TMPLOG" ; then
+    _ld_gif="-lungif $_ld_x11"
+  elif cc_check -lgif && "$TMPO" >> "$TMPLOG" ; then
+    _ld_gif="-lgif"
+  elif cc_check -lgif $_ld_x11 && "$TMPO" >> "$TMPLOG" ; then
+    _ld_gif="-lgif $_ld_x11"
+  else
+    die "GIF support is requested but neither libungif nor libgif can't be found"
+  fi
+  
 fi
 
 if test "$_gif" = yes ; then
