diff -urN mplayer-svn-r20777.orig/configure mplayer-svn-r20777/configure
--- mplayer-svn-r20777.orig/configure	2006-11-08 14:49:41 +0200
+++ mplayer-svn-r20777/configure	2006-11-09 01:25:36 +0200
@@ -218,6 +218,7 @@
   --enable-linux-devfs   set default devices to devfs [disable]
   --enable-termcap       use termcap database for key codes [autodetect]
   --enable-termios       use termios database for key codes [autodetect]
+  --enable-nls           use Native Language Support [disable]
   --disable-iconv        disable iconv for encoding conversion [autodetect]
   --disable-langinfo     do not use langinfo [autodetect]
   --enable-lirc          enable LIRC (remote control) support [autodetect]
@@ -1591,6 +1592,7 @@
 _dxr2=auto
 _dxr3=auto
 _ivtv=auto
+_nls=no
 _iconv=auto
 _langinfo=auto
 _rtc=auto
@@ -1790,6 +1792,8 @@
   --disable-dxr3)	_dxr3=no	;;
   --enable-ivtv)        _ivtv=yes       ;;
   --disable-ivtv)       _ivtv=no        ;;
+  --enable-nls)		_nls=yes	;;
+  --disable-nls)	_nls=no		;;
   --enable-iconv)	_iconv=yes	;;
   --disable-iconv)	_iconv=no	;;
   --enable-langinfo)	_langinfo=yes	;;
@@ -3025,6 +3029,14 @@
 fi
 echores "$_rpath"
 
+echocheck "NLS"
+if test "$_nls" = yes ; then
+  _def_nls='#define ENABLE_NLS 1'
+else
+  _def_nls='#undef ENABLE_NLS'
+fi
+echores "$_nls"
+
 echocheck "iconv"
 if test "$_iconv" = auto ; then
   _iconv_tmp='#include <iconv.h>'
@@ -7657,6 +7669,10 @@
    try to recompile MPlayer with this option disabled! */
 $_def_select
 
+$_def_nls
+#define PACKAGE "mplayer"
+#define LOCALEDIR "$_prefix/share/locale"
+
 /* define this to use iconv(3) function to codepage conversions */
 $_def_iconv
 
diff -urN mplayer-svn-r20777.orig/mp_msg.c mplayer-svn-r20777/mp_msg.c
--- mplayer-svn-r20777.orig/mp_msg.c	2006-08-28 14:40:34 +0300
+++ mplayer-svn-r20777/mp_msg.c	2006-11-09 01:24:19 +0200
@@ -8,6 +8,13 @@
 
 #include "config.h"
 
+#ifdef ENABLE_NLS
+#include <libintl.h>
+#define _(x) gettext(x)
+#else
+#define _(x) (x)
+#endif
+
 #ifdef USE_LANGINFO
 #include <locale.h>
 #include <langinfo.h>
@@ -27,7 +34,7 @@
 #include "mp_msg.h"
 
 /* maximum message length of mp_msg */
-#define MSGSIZE_MAX 3072
+#define MSGSIZE_MAX 4096
 
 int mp_msg_levels[MSGT_MAX]; // verbose level of this module. inited to -2
 int mp_msg_level_all = MSGL_STATUS;
@@ -54,6 +61,11 @@
       setlocale(LC_CTYPE, "C");
     }
 #endif
+#ifdef ENABLE_NLS
+    setlocale(LC_ALL, "");
+    bindtextdomain(PACKAGE, LOCALEDIR);
+    textdomain(PACKAGE);
+#endif
 #endif
 }
 
@@ -68,14 +80,14 @@
     
     if (!mp_msg_test(mod, lev)) return; // do not display
     va_start(va, format);
-    vsnprintf(tmp, MSGSIZE_MAX, format, va);
+    vsnprintf(tmp, MSGSIZE_MAX, _(format), va);
     va_end(va);
     tmp[MSGSIZE_MAX-2] = '\n';
     tmp[MSGSIZE_MAX-1] = 0;
 
 #ifdef HAVE_NEW_GUI
     if(use_gui)
-        guiMessageBox(lev, tmp);
+        guiMessageBox(lev, _(tmp));
 #endif
 
 #if defined(USE_ICONV) && defined(MSG_CHARSET)
