diff -urN mplayer-svn-r19447.orig/libvo/vo_cvidix.c mplayer-svn-r19447/libvo/vo_cvidix.c
--- mplayer-svn-r19447.orig/libvo/vo_cvidix.c	2006-06-13 15:11:09 +0300
+++ mplayer-svn-r19447/libvo/vo_cvidix.c	2006-08-21 19:08:50 +0300
@@ -23,7 +23,11 @@
 #include "mp_msg.h"
 
 #include "vosub_vidix.h"
+#ifdef VIDIX_EXT
+#include <vidixlib.h>
+#else
 #include "vidix/vidixlib.h"
+#endif
 
 
 static vo_info_t info = {
diff -urN mplayer-svn-r19447.orig/libvo/vosub_vidix.c mplayer-svn-r19447/libvo/vosub_vidix.c
--- mplayer-svn-r19447.orig/libvo/vosub_vidix.c	2006-06-13 15:11:09 +0300
+++ mplayer-svn-r19447/libvo/vosub_vidix.c	2006-08-21 19:08:50 +0300
@@ -28,7 +28,11 @@
 #include "help_mp.h"
 
 #include "vosub_vidix.h"
+#ifdef VIDIX_EXT
+#include <vidixlib.h>
+#else
 #include "vidix/vidixlib.h"
+#endif
 #include "fastmemcpy.h"
 #include "osd.h"
 #include "video_out.h"
diff -urN mplayer-svn-r19447.orig/libvo/vosub_vidix.h mplayer-svn-r19447/libvo/vosub_vidix.h
--- mplayer-svn-r19447.orig/libvo/vosub_vidix.h	2006-06-13 15:11:09 +0300
+++ mplayer-svn-r19447/libvo/vosub_vidix.h	2006-08-21 19:08:50 +0300
@@ -29,7 +29,11 @@
 void     vidix_flip_page(void);
 void     vidix_draw_osd(void);
 
+#ifdef VIDIX_EXT
+#include <vidix.h>
+#else
 #include "vidix/vidix.h"
+#endif
 /* graphic keys */
 int vidix_grkey_support(void);
 int vidix_grkey_get(vidix_grkey_t *gr_key);
diff -urN mplayer-svn-r19447.orig/libvo/vo_winvidix.c mplayer-svn-r19447/libvo/vo_winvidix.c
--- mplayer-svn-r19447.orig/libvo/vo_winvidix.c	2006-06-13 15:11:09 +0300
+++ mplayer-svn-r19447/libvo/vo_winvidix.c	2006-08-21 19:08:50 +0300
@@ -24,7 +24,11 @@
 #include "mp_msg.h"
 
 #include "vosub_vidix.h"
+#ifdef VIDIX_EXT
+#include <vidixlib.h>
+#else
 #include "vidix/vidixlib.h"
+#endif
 
 extern void mplayer_put_key(int code);
 
diff -urN mplayer-svn-r19447.orig/libvo/vo_xvidix.c mplayer-svn-r19447/libvo/vo_xvidix.c
--- mplayer-svn-r19447.orig/libvo/vo_xvidix.c	2006-06-20 13:10:53 +0300
+++ mplayer-svn-r19447/libvo/vo_xvidix.c	2006-08-21 19:08:50 +0300
@@ -31,7 +31,11 @@
 #include "mp_msg.h"
 
 #include "vosub_vidix.h"
+#ifdef VIDIX_EXT
+#include <vidixlib.h>
+#else
 #include "vidix/vidixlib.h"
+#endif
 
 #ifdef HAVE_NEW_GUI
 #include "Gui/interface.h"
