diff -urN mplayer-svn-r23664.orig/configure mplayer-svn-r23664/configure
--- mplayer-svn-r23664.orig/configure	2007-06-26 00:49:09 +0300
+++ mplayer-svn-r23664/configure	2007-06-26 01:24:03 +0300
@@ -290,10 +290,12 @@
   --disable-libavcodec_a    disable static libavcodec [autodetect]
   --disable-libavformat_a   disable static libavformat [autodetect]
   --disable-libpostproc_a   disable static libpostproc [autodetect]
+  --disable-libswscale_a    disable static libswscale [autodetect]
   --disable-libavutil_so    disable shared libavutil [autodetect]
   --disable-libavcodec_so   disable shared libavcodec [autodetect]
   --disable-libavformat_so  disable shared libavformat [autodetect]
   --disable-libpostproc_so  disable shared libpostproc [autodetect]
+  --disable-libswscale_so   disable shared libswscale [autodetect]
   --disable-libavcodec_mpegaudio_hp disable high precision audio decoding
                                     in libavcodec [enabled]
   --disable-tremor-internal disable internal Tremor [enabled]
@@ -499,6 +501,8 @@
 _libavformat_so=auto
 _libpostproc_a=auto
 _libpostproc_so=auto
+_libswscale_a=auto
+_libswscale_so=auto
 _libavcodec_mpegaudio_hp=yes
 _mencoder=yes
 _mplayer=yes
@@ -1000,6 +1004,10 @@
   --disable-libpostproc_a)	_libpostproc_a=no	;;
   --enable-libpostproc_so)	_libpostproc_so=yes	;;
   --disable-libpostproc_so)	_libpostproc_so=no	;;
+  --enable-libswscale_a)	_libswscale_a=yes	;;
+  --disable-libswscale_a)	_libswscale_a=no	;;
+  --enable-libswscale_so)	_libswscale_so=yes	;;
+  --disable-libswscale_so)	_libswscale_so=no	;;
   --enable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=yes	;;
   --disable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=no	;;
 
@@ -6225,6 +6233,45 @@
   && _def_libpostproc_so='#define USE_LIBPOSTPROC_SO 1'
 echores "$_libpostproc"
 
+echocheck "FFmpeg libswscale"
+if test "$_libswscale_a" = auto ; then
+  if test -d libswascale ; then
+    _libswscale_a=yes
+    _res_comment="static"
+  else
+    die "MPlayer will not compile without libswscale in the source tree."
+  fi
+elif test "$_libswscale_so" != no ; then
+  _libswscale_so=no
+  cat > $TMPC << EOF
+#include <ffmpeg/swscale.h>
+int main(void) { sws_getIdentityVec(); return 0; }
+EOF
+  if $_pkg_config --exists libswscale ; then
+    _inc_libswscale=`$_pkg_config --cflags libswscale`
+    _ld_tmp=`$_pkg_config --libs libswscale`
+    cc_check $_inc_libswscale $_ld_tmp && _ld_extra="$_ld_extra $_ld_tmp" \
+      && _libswscale_so=yes
+  elif cc_check -lswscale $_ld_lm ; then
+    _ld_extra="$_ld_extra -lswscale"
+    _libswscale_so=yes
+    _res_comment="using libswscale.so, but static libswscale is recommended"
+  fi
+fi
+_libswscale=no
+_def_libswscale='#undef USE_LIBSWSCALE'
+_def_libswscale_a='#undef USE_LIBSWSCALE_A'
+_def_libswscale_so='#undef USE_LIBSWSCALE_SO'
+test "$_libswscale_a" = yes || test "$_libswscale_so" = yes && _libswscale=yes
+test "$_libswscale"    = yes && _def_libswscale='#define USE_LIBSWSCALE 1'
+test "$_libswscale_a"  = yes && _def_libswscale_a='#define USE_LIBSWSCALE_A 1'
+test "$_libswscale_so" = yes && _def_libswscale_so='#define USE_LIBSWSCALE_SO 1'
+# neither static nor shared libswscale is available, but it is mandatory ...
+if test "$_libswscale" = no ; then
+  die "You need static or shared libswscale, MPlayer will not compile without!"
+fi
+echores "$_libswscale"
+
 
 echocheck "md5sum support"
 if test "$_md5sum" = yes; then
@@ -7560,6 +7607,10 @@
 LIBPOSTPROC = $_libpostproc
 LIBPOSTPROC_A = $_libpostproc_a
 LIBPOSTPROC_SO = $_libpostproc_so
+LIBSWSCALE = $_libswscale
+LIBSWSCALE_A = $_libswscale_a
+LIBSWSCALE_SO = $_libswscale_so
+CONFIG_SWSCALER = $_libswscale
 ZORAN = $_zr
 LIBLZO = $_liblzo
 LIBDV = $_libdv
@@ -7967,6 +8018,10 @@
 $_def_libpostproc_a
 $_def_libpostproc_so
 
+$_def_libswscale
+$_def_libswscale_a
+$_def_libswscale_so
+
 /* Win32 DLL support */
 $_def_win32dll
 #define WIN32_PATH "$_win32codecsdir"
diff -urN mplayer-svn-r23664.orig/gui/wm/ws.c mplayer-svn-r23664/gui/wm/ws.c
--- mplayer-svn-r23664.orig/gui/wm/ws.c	2007-06-05 15:35:29 +0300
+++ mplayer-svn-r23664/gui/wm/ws.c	2007-06-26 01:24:03 +0300
@@ -24,8 +24,13 @@
 #include "ws.h"
 #include "wsxdnd.h"
 #include "../cpudetect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "../libswscale/swscale.h"
 #include "../libswscale/rgb2rgb.h"
+#endif
 #include "../libmpcodecs/vf_scale.h"
 #include "../mp_msg.h"
 #include "../help_mp.h"
diff -urN mplayer-svn-r23664.orig/liba52/parse.c mplayer-svn-r23664/liba52/parse.c
--- mplayer-svn-r23664.orig/liba52/parse.c	2006-11-09 16:34:19 +0200
+++ mplayer-svn-r23664/liba52/parse.c	2007-06-26 01:24:03 +0300
@@ -37,7 +37,11 @@
 #include "bitstream.h"
 #include "tables.h"
 #include "mm_accel.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "libavutil/avutil.h"
+#endif
 
 #ifdef HAVE_MEMALIGN
 /* some systems have memalign() but no declaration for it */
diff -urN mplayer-svn-r23664.orig/libaf/af_format.c mplayer-svn-r23664/libaf/af_format.c
--- mplayer-svn-r23664.orig/libaf/af_format.c	2007-06-05 23:28:31 +0300
+++ mplayer-svn-r23664/libaf/af_format.c	2007-06-26 01:24:03 +0300
@@ -21,7 +21,11 @@
 #endif
 
 #include "af.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r23664.orig/libao2/ao_dxr2.c mplayer-svn-r23664/libao2/ao_dxr2.c
--- mplayer-svn-r23664.orig/libao2/ao_dxr2.c	2006-12-08 20:58:11 +0200
+++ mplayer-svn-r23664/libao2/ao_dxr2.c	2007-06-26 01:24:03 +0300
@@ -7,7 +7,11 @@
 #include "config.h"
 #include "mp_msg.h"
 #include "help_mp.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "audio_out.h"
diff -urN mplayer-svn-r23664.orig/libao2/ao_pcm.c mplayer-svn-r23664/libao2/ao_pcm.c
--- mplayer-svn-r23664.orig/libao2/ao_pcm.c	2006-12-08 20:58:11 +0200
+++ mplayer-svn-r23664/libao2/ao_pcm.c	2007-06-26 01:24:03 +0300
@@ -4,7 +4,11 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "subopt-helper.h"
 #include "libaf/af_format.h"
diff -urN mplayer-svn-r23664.orig/libass/mputils.h mplayer-svn-r23664/libass/mputils.h
--- mplayer-svn-r23664.orig/libass/mputils.h	2006-11-20 12:43:26 +0200
+++ mplayer-svn-r23664/libass/mputils.h	2007-06-26 01:24:03 +0300
@@ -6,6 +6,10 @@
 #include "libvo/font_load.h" // for blur()
 #include "subreader.h" // for guess_buffer_cp
 #include "libvo/sub.h" // for utf8_get_char
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #endif
diff -urN mplayer-svn-r23664.orig/libmenu/menu.c mplayer-svn-r23664/libmenu/menu.c
--- mplayer-svn-r23664.orig/libmenu/menu.c	2007-05-20 22:43:27 +0300
+++ mplayer-svn-r23664/libmenu/menu.c	2007-06-26 01:24:03 +0300
@@ -307,7 +307,11 @@
 
 #ifdef USE_FRIBIDI
 #include <fribidi/fribidi.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 char *menu_fribidi_charset = NULL;
 int menu_flip_hebrew = 0;
 int menu_fribidi_flip_commas = 0;
diff -urN mplayer-svn-r23664.orig/libmpcodecs/ad_dk3adpcm.c mplayer-svn-r23664/libmpcodecs/ad_dk3adpcm.c
--- mplayer-svn-r23664.orig/libmpcodecs/ad_dk3adpcm.c	2006-12-08 20:58:10 +0200
+++ mplayer-svn-r23664/libmpcodecs/ad_dk3adpcm.c	2007-06-26 01:24:03 +0300
@@ -16,7 +16,11 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "ad_internal.h"
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/ad_imaadpcm.c mplayer-svn-r23664/libmpcodecs/ad_imaadpcm.c
--- mplayer-svn-r23664.orig/libmpcodecs/ad_imaadpcm.c	2006-12-08 20:58:09 +0200
+++ mplayer-svn-r23664/libmpcodecs/ad_imaadpcm.c	2007-06-26 01:24:03 +0300
@@ -21,7 +21,11 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "ad_internal.h"
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/ad_msadpcm.c mplayer-svn-r23664/libmpcodecs/ad_msadpcm.c
--- mplayer-svn-r23664.orig/libmpcodecs/ad_msadpcm.c	2006-12-08 20:58:10 +0200
+++ mplayer-svn-r23664/libmpcodecs/ad_msadpcm.c	2007-06-26 01:24:03 +0300
@@ -12,7 +12,11 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "ad_internal.h"
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/ad_qtaudio.c mplayer-svn-r23664/libmpcodecs/ad_qtaudio.c
--- mplayer-svn-r23664.orig/libmpcodecs/ad_qtaudio.c	2007-03-19 00:20:09 +0200
+++ mplayer-svn-r23664/libmpcodecs/ad_qtaudio.c	2007-06-26 01:24:03 +0300
@@ -14,7 +14,13 @@
 #include "loader/wine/windef.h"
 
 #include "ad_internal.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/common.h"
+#include "libavutil/bswap.h"
+#endif
 
 #ifdef WIN32_LOADER
 #include "loader/ldt_keeper.h"
diff -urN mplayer-svn-r23664.orig/libmpcodecs/native/nuppelvideo.c mplayer-svn-r23664/libmpcodecs/native/nuppelvideo.c
--- mplayer-svn-r23664.orig/libmpcodecs/native/nuppelvideo.c	2007-06-05 23:28:21 +0300
+++ mplayer-svn-r23664/libmpcodecs/native/nuppelvideo.c	2007-06-26 01:24:03 +0300
@@ -13,7 +13,11 @@
 
 #include "config.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "../libvo/fastmemcpy.h"
diff -urN mplayer-svn-r23664.orig/libmpcodecs/native/RTjpegN.c mplayer-svn-r23664/libmpcodecs/native/RTjpegN.c
--- mplayer-svn-r23664.orig/libmpcodecs/native/RTjpegN.c	2007-01-22 19:01:51 +0200
+++ mplayer-svn-r23664/libmpcodecs/native/RTjpegN.c	2007-06-26 01:24:03 +0300
@@ -30,7 +30,11 @@
 #define MMX
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "RTjpegN.h"
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vd_ffmpeg.c mplayer-svn-r23664/libmpcodecs/vd_ffmpeg.c
--- mplayer-svn-r23664.orig/libmpcodecs/vd_ffmpeg.c	2007-05-28 19:50:48 +0300
+++ mplayer-svn-r23664/libmpcodecs/vd_ffmpeg.c	2007-06-26 01:24:03 +0300
@@ -7,8 +7,15 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/avutil.h>
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/common.h"
+#include "libavutil/avutil.h"
 #include "libavutil/intreadwrite.h"
+#endif
 #include "mpbswap.h"
 
 #include "vd_internal.h"
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vd_ijpg.c mplayer-svn-r23664/libmpcodecs/vd_ijpg.c
--- mplayer-svn-r23664.orig/libmpcodecs/vd_ijpg.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r23664/libmpcodecs/vd_ijpg.c	2007-06-26 01:24:03 +0300
@@ -12,7 +12,11 @@
 
 #include <setjmp.h>
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "vd_internal.h"
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vd_mpng.c mplayer-svn-r23664/libmpcodecs/vd_mpng.c
--- mplayer-svn-r23664.orig/libmpcodecs/vd_mpng.c	2007-06-05 23:28:24 +0300
+++ mplayer-svn-r23664/libmpcodecs/vd_mpng.c	2007-06-26 01:24:03 +0300
@@ -6,7 +6,11 @@
 
 #include <png.h>
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vd_mtga.c mplayer-svn-r23664/libmpcodecs/vd_mtga.c
--- mplayer-svn-r23664.orig/libmpcodecs/vd_mtga.c	2007-06-05 23:28:24 +0300
+++ mplayer-svn-r23664/libmpcodecs/vd_mtga.c	2007-06-26 01:24:03 +0300
@@ -11,7 +11,11 @@
 #include "config.h"
 #include "mp_msg.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vd_qtvideo.c mplayer-svn-r23664/libmpcodecs/vd_qtvideo.c
--- mplayer-svn-r23664.orig/libmpcodecs/vd_qtvideo.c	2007-03-19 00:20:09 +0200
+++ mplayer-svn-r23664/libmpcodecs/vd_qtvideo.c	2007-06-26 01:24:03 +0300
@@ -28,7 +28,7 @@
 
 LIBVD_EXTERN(qtvideo)
 
-#include "bswap.h"
+#include "mpbswap.h"
 
 #ifndef MACOSX
 #include "loader/qtx/qtxsdk/components.h"
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vd_sgi.c mplayer-svn-r23664/libmpcodecs/vd_sgi.c
--- mplayer-svn-r23664.orig/libmpcodecs/vd_sgi.c	2006-12-08 20:58:10 +0200
+++ mplayer-svn-r23664/libmpcodecs/vd_sgi.c	2007-06-26 01:24:03 +0300
@@ -7,7 +7,11 @@
 
 #include "config.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "vd_internal.h"
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vd_xanim.c mplayer-svn-r23664/libmpcodecs/vd_xanim.c
--- mplayer-svn-r23664.orig/libmpcodecs/vd_xanim.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r23664/libmpcodecs/vd_xanim.c	2007-06-26 01:24:03 +0300
@@ -37,7 +37,11 @@
 #include <errno.h> /* strerror, errno */
 
 #include "mp_msg.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 
 #include "osdep/timer.h"
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/ve_qtvideo.c mplayer-svn-r23664/libmpcodecs/ve_qtvideo.c
--- mplayer-svn-r23664.orig/libmpcodecs/ve_qtvideo.c	2007-03-19 00:20:09 +0200
+++ mplayer-svn-r23664/libmpcodecs/ve_qtvideo.c	2007-06-26 01:24:03 +0300
@@ -10,7 +10,11 @@
 
 #include "config.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #ifdef WIN32_LOADER 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_bmovl.c mplayer-svn-r23664/libmpcodecs/vf_bmovl.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_bmovl.c	2007-06-05 23:28:24 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_bmovl.c	2007-06-26 01:24:03 +0300
@@ -69,7 +69,11 @@
 #include "img_format.h"
 
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #include "libvo/fastmemcpy.h"
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_divtc.c mplayer-svn-r23664/libmpcodecs/vf_divtc.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_divtc.c	2006-12-08 20:58:10 +0200
+++ mplayer-svn-r23664/libmpcodecs/vf_divtc.c	2007-06-26 01:24:03 +0300
@@ -7,7 +7,11 @@
 #include "config.h"
 #include "mp_msg.h"
 #include "cpudetect.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "img_format.h"
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_geq.c mplayer-svn-r23664/libmpcodecs/vf_geq.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_geq.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_geq.c	2007-06-26 01:24:03 +0300
@@ -30,10 +30,20 @@
 // Needed to bring in lrintf.
 #define HAVE_AV_CONFIG_H
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#include <ffmpeg/dsputil.h>
+#include <ffmpeg/eval.h>
+#else
 #include "libavcodec/avcodec.h"
 #include "libavcodec/dsputil.h"
 #include "libavcodec/eval.h"
+#endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 /* FIXME: common.h defines printf away when HAVE_AV_CONFIG
  * is defined, but mp_image.h needs printf.
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_halfpack.c mplayer-svn-r23664/libmpcodecs/vf_halfpack.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_halfpack.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_halfpack.c	2007-06-26 01:24:03 +0300
@@ -11,7 +11,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 
 struct vf_priv_s {
 	int field;
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_palette.c mplayer-svn-r23664/libmpcodecs/vf_palette.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_palette.c	2006-07-03 14:16:54 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_palette.c	2007-06-26 01:24:03 +0300
@@ -11,7 +11,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 
 //===========================================================================//
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_qp.c mplayer-svn-r23664/libmpcodecs/vf_qp.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_qp.c	2006-06-13 15:10:35 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_qp.c	2007-06-26 01:24:03 +0300
@@ -37,9 +37,18 @@
 // Needed to bring in lrintf.
 #define HAVE_AV_CONFIG_H
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#include <ffmpeg/dsputil.h>
+#else
 #include "libavcodec/avcodec.h"
 #include "libavcodec/dsputil.h"
+#endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 /* FIXME: common.h defines printf away when HAVE_AV_CONFIG
  * is defined, but mp_image.h needs printf.
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_rectangle.c mplayer-svn-r23664/libmpcodecs/vf_rectangle.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_rectangle.c	2006-12-04 17:59:48 +0200
+++ mplayer-svn-r23664/libmpcodecs/vf_rectangle.c	2007-06-26 01:24:03 +0300
@@ -6,7 +6,11 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 struct vf_priv_s {
     int x, y, w, h;
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_rgb2bgr.c mplayer-svn-r23664/libmpcodecs/vf_rgb2bgr.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_rgb2bgr.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_rgb2bgr.c	2007-06-26 01:24:03 +0300
@@ -10,7 +10,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 
 //===========================================================================//
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_sab.c mplayer-svn-r23664/libmpcodecs/vf_sab.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_sab.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_sab.c	2007-06-26 01:24:03 +0300
@@ -37,7 +37,11 @@
 #include "img_format.h"
 #include "mp_image.h"
 #include "vf.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_scale.c mplayer-svn-r23664/libmpcodecs/vf_scale.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_scale.c	2007-06-08 01:58:17 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_scale.c	2007-06-26 01:24:03 +0300
@@ -11,9 +11,13 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "fmt-conversion.h"
-#include "bswap.h"
-
+#include "mpbswap.h"
+  
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 #include "m_option.h"
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_screenshot.c mplayer-svn-r23664/libmpcodecs/vf_screenshot.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_screenshot.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_screenshot.c	2007-06-26 01:24:03 +0300
@@ -21,7 +21,11 @@
 #include "vf.h"
 #include "vf_scale.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 
 struct vf_priv_s {
     int frameno;
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_smartblur.c mplayer-svn-r23664/libmpcodecs/vf_smartblur.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_smartblur.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_smartblur.c	2007-06-26 01:24:03 +0300
@@ -37,7 +37,11 @@
 #include "img_format.h"
 #include "mp_image.h"
 #include "vf.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_unsharp.c mplayer-svn-r23664/libmpcodecs/vf_unsharp.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_unsharp.c	2007-06-05 23:28:24 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_unsharp.c	2007-06-26 01:24:03 +0300
@@ -35,7 +35,11 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 //===========================================================================//
 
diff -urN mplayer-svn-r23664.orig/libmpcodecs/vf_yuy2.c mplayer-svn-r23664/libmpcodecs/vf_yuy2.c
--- mplayer-svn-r23664.orig/libmpcodecs/vf_yuy2.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r23664/libmpcodecs/vf_yuy2.c	2007-06-26 01:24:03 +0300
@@ -11,7 +11,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r23664.orig/libmpdemux/asf.h mplayer-svn-r23664/libmpdemux/asf.h
--- mplayer-svn-r23664.orig/libmpdemux/asf.h	2007-05-23 00:33:40 +0300
+++ mplayer-svn-r23664/libmpdemux/asf.h	2007-06-26 01:24:03 +0300
@@ -3,7 +3,11 @@
 
 //#include "config.h"	/* for WORDS_BIGENDIAN */
 #include <inttypes.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 ///////////////////////
diff -urN mplayer-svn-r23664.orig/libmpdemux/asfheader.c mplayer-svn-r23664/libmpdemux/asfheader.c
--- mplayer-svn-r23664.orig/libmpdemux/asfheader.c	2007-05-31 00:38:24 +0300
+++ mplayer-svn-r23664/libmpdemux/asfheader.c	2007-06-26 01:24:03 +0300
@@ -6,8 +6,13 @@
 #include <unistd.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/common.h"
 #include "libavutil/intreadwrite.h"
+#endif
 #include "mp_msg.h"
 #include "help_mp.h"
 
diff -urN mplayer-svn-r23664.orig/libmpdemux/aviheader.c mplayer-svn-r23664/libmpdemux/aviheader.c
--- mplayer-svn-r23664.orig/libmpdemux/aviheader.c	2007-04-23 01:35:52 +0300
+++ mplayer-svn-r23664/libmpdemux/aviheader.c	2007-06-26 01:24:03 +0300
@@ -13,7 +13,11 @@
 #include "stheader.h"
 
 #include "aviheader.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 static MainAVIHeader avih;
 
diff -urN mplayer-svn-r23664.orig/libmpdemux/aviheader.h mplayer-svn-r23664/libmpdemux/aviheader.h
--- mplayer-svn-r23664.orig/libmpdemux/aviheader.h	2006-12-11 15:00:14 +0200
+++ mplayer-svn-r23664/libmpdemux/aviheader.h	2007-06-26 01:24:03 +0300
@@ -1,8 +1,12 @@
 #ifndef _aviheader_h
 #define	_aviheader_h
 
-//#include "config.h"	/* get correct definition WORDS_BIGENDIAN */
+#include "config.h"	/* get correct definition WORDS_BIGENDIAN */
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #ifndef mmioFOURCC
diff -urN mplayer-svn-r23664.orig/libmpdemux/demux_audio.c mplayer-svn-r23664/libmpdemux/demux_audio.c
--- mplayer-svn-r23664.orig/libmpdemux/demux_audio.c	2007-06-24 23:25:10 +0300
+++ mplayer-svn-r23664/libmpdemux/demux_audio.c	2007-06-26 01:24:03 +0300
@@ -10,7 +10,11 @@
 #include "stheader.h"
 #include "genres.h"
 #include "mp3_hdr.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 #include <string.h>
 #ifdef MP_DEBUG
diff -urN mplayer-svn-r23664.orig/libmpdemux/demux_lavf.c mplayer-svn-r23664/libmpdemux/demux_lavf.c
--- mplayer-svn-r23664.orig/libmpdemux/demux_lavf.c	2007-05-11 01:00:48 +0300
+++ mplayer-svn-r23664/libmpdemux/demux_lavf.c	2007-06-26 01:24:03 +0300
@@ -40,7 +40,11 @@
 #include "avi.h"
 #include "opt.h"
 #endif
-#include "riff.h"
+#ifdef USE_LIBAVFORMAT_SO
+#include <ffmpeg/riff.h>
+#else
+#include "libavformat/riff.h"
+#endif
 
 #define PROBE_BUF_SIZE 2048
 
diff -urN mplayer-svn-r23664.orig/libmpdemux/demux_mkv.c mplayer-svn-r23664/libmpdemux/demux_mkv.c
--- mplayer-svn-r23664.orig/libmpdemux/demux_mkv.c	2007-06-24 23:25:10 +0300
+++ mplayer-svn-r23664/libmpdemux/demux_mkv.c	2007-06-26 01:24:03 +0300
@@ -28,7 +28,11 @@
 #include "libass/ass.h"
 #include "libass/ass_mp.h"
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #ifdef USE_QTX_CODECS
 #include "loader/qtx/qtxsdk/components.h"
diff -urN mplayer-svn-r23664.orig/libmpdemux/demux_mpc.c mplayer-svn-r23664/libmpdemux/demux_mpc.c
--- mplayer-svn-r23664.orig/libmpdemux/demux_mpc.c	2007-03-19 00:20:28 +0200
+++ mplayer-svn-r23664/libmpdemux/demux_mpc.c	2007-06-26 01:24:03 +0300
@@ -11,7 +11,11 @@
 #include <stdio.h>
 #include <string.h>
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "stream/stream.h"
 #include "demuxer.h"
diff -urN mplayer-svn-r23664.orig/libmpdemux/demux_ogg.c mplayer-svn-r23664/libmpdemux/demux_ogg.c
--- mplayer-svn-r23664.orig/libmpdemux/demux_ogg.c	2007-05-28 19:50:51 +0300
+++ mplayer-svn-r23664/libmpdemux/demux_ogg.c	2007-06-26 01:24:03 +0300
@@ -13,7 +13,11 @@
 #include "stream/stream.h"
 #include "demuxer.h"
 #include "stheader.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 #define FOURCC_VORBIS mmioFOURCC('v', 'r', 'b', 's')
 #define FOURCC_SPEEX  mmioFOURCC('s', 'p', 'x', ' ')
diff -urN mplayer-svn-r23664.orig/libmpdemux/demux_rtp_codec.cpp mplayer-svn-r23664/libmpdemux/demux_rtp_codec.cpp
--- mplayer-svn-r23664.orig/libmpdemux/demux_rtp_codec.cpp	2007-04-10 03:17:58 +0300
+++ mplayer-svn-r23664/libmpdemux/demux_rtp_codec.cpp	2007-06-26 01:24:03 +0300
@@ -6,7 +6,11 @@
 #include <limits.h>
 #include <math.h>
 #include "stheader.h"
-#include "base64.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/base64.h>
+#else
+#include "libavutil/base64.h"
+#endif
 }
 
 #ifdef USE_LIBAVCODEC
diff -urN mplayer-svn-r23664.orig/libmpdemux/demux_vqf.c mplayer-svn-r23664/libmpdemux/demux_vqf.c
--- mplayer-svn-r23664.orig/libmpdemux/demux_vqf.c	2007-03-19 00:20:28 +0200
+++ mplayer-svn-r23664/libmpdemux/demux_vqf.c	2007-06-26 01:24:03 +0300
@@ -2,7 +2,11 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "stream/stream.h"
diff -urN mplayer-svn-r23664.orig/libmpdemux/ebml.c mplayer-svn-r23664/libmpdemux/ebml.c
--- mplayer-svn-r23664.orig/libmpdemux/ebml.c	2007-03-19 00:20:28 +0200
+++ mplayer-svn-r23664/libmpdemux/ebml.c	2007-06-26 01:24:03 +0300
@@ -11,9 +11,17 @@
 
 #include "stream/stream.h"
 #include "ebml.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intfloat_readwrite.h>
+#else
 #include "libavutil/intfloat_readwrite.h"
+#endif
 
 
 #ifndef SIZE_MAX
diff -urN mplayer-svn-r23664.orig/libmpdemux/muxer_lavf.c mplayer-svn-r23664/libmpdemux/muxer_lavf.c
--- mplayer-svn-r23664.orig/libmpdemux/muxer_lavf.c	2007-06-24 16:02:02 +0300
+++ mplayer-svn-r23664/libmpdemux/muxer_lavf.c	2007-06-26 01:25:13 +0300
@@ -22,7 +22,11 @@
 #else
 #include "avformat.h"
 #endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avstring.h>
+#else
 #include "libavutil/avstring.h"
+#endif
 
 extern const struct AVCodecTag *mp_wav_taglists[];
 extern const struct AVCodecTag *mp_bmp_taglists[];
diff -urN mplayer-svn-r23664.orig/libmpdemux/nuppelvideo.h mplayer-svn-r23664/libmpdemux/nuppelvideo.h
--- mplayer-svn-r23664.orig/libmpdemux/nuppelvideo.h	2006-11-29 12:06:38 +0200
+++ mplayer-svn-r23664/libmpdemux/nuppelvideo.h	2007-06-26 01:24:03 +0300
@@ -1,6 +1,10 @@
 /* nuppelvideo.h  rh */
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intfloat_readwrite.h>
+#else
 #include "libavutil/intfloat_readwrite.h"
+#endif
 
 typedef struct  __attribute__((packed)) rtfileheader
 {
diff -urN mplayer-svn-r23664.orig/libswscale/rgb2rgb.c mplayer-svn-r23664/libswscale/rgb2rgb.c
--- mplayer-svn-r23664.orig/libswscale/rgb2rgb.c	2007-05-28 19:50:50 +0300
+++ mplayer-svn-r23664/libswscale/rgb2rgb.c	2007-06-26 01:24:03 +0300
@@ -31,7 +31,11 @@
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "x86_cpu.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 
 #define FAST_BGR2YV12 // use 7 bit coeffs instead of 15bit
 
diff -urN mplayer-svn-r23664.orig/libswscale/swscale.c mplayer-svn-r23664/libswscale/swscale.c
--- mplayer-svn-r23664.orig/libswscale/swscale.c	2007-06-23 00:22:29 +0300
+++ mplayer-svn-r23664/libswscale/swscale.c	2007-06-26 01:24:03 +0300
@@ -70,7 +70,11 @@
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "x86_cpu.h"
-#include "bswap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
+#include "libavutil/bswap.h"
+#endif
 #include "rgb2rgb.h"
 #include "libavcodec/opt.h"
 
diff -urN mplayer-svn-r23664.orig/libswscale/swscale-example.c mplayer-svn-r23664/libswscale/swscale-example.c
--- mplayer-svn-r23664.orig/libswscale/swscale-example.c	2007-05-02 22:43:16 +0300
+++ mplayer-svn-r23664/libswscale/swscale-example.c	2007-06-26 01:24:03 +0300
@@ -25,7 +25,11 @@
 #include <stdarg.h>
 
 #undef HAVE_AV_CONFIG_H
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "rgb2rgb.h"
diff -urN mplayer-svn-r23664.orig/libswscale/swscale_internal.h mplayer-svn-r23664/libswscale/swscale_internal.h
--- mplayer-svn-r23664.orig/libswscale/swscale_internal.h	2007-06-13 23:25:26 +0300
+++ mplayer-svn-r23664/libswscale/swscale_internal.h	2007-06-26 01:24:03 +0300
@@ -25,7 +25,11 @@
 #include <altivec.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 
 #ifdef CONFIG_DARWIN
 #define AVV(x...) (x)
diff -urN mplayer-svn-r23664.orig/libvo/font_load_ft.c mplayer-svn-r23664/libvo/font_load_ft.c
--- mplayer-svn-r23664.orig/libvo/font_load_ft.c	2007-06-24 23:25:09 +0300
+++ mplayer-svn-r23664/libvo/font_load_ft.c	2007-06-26 01:24:03 +0300
@@ -28,7 +28,11 @@
 #include <fontconfig/fontconfig.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "font_load.h"
 #include "mp_msg.h"
diff -urN mplayer-svn-r23664.orig/libvo/mga_common.c mplayer-svn-r23664/libvo/mga_common.c
--- mplayer-svn-r23664.orig/libvo/mga_common.c	2007-06-24 16:02:00 +0300
+++ mplayer-svn-r23664/libvo/mga_common.c	2007-06-26 01:24:03 +0300
@@ -1,8 +1,12 @@
 
 #include "fastmemcpy.h"
 #include "cpudetect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #include "mp_msg.h"
 #include "help_mp.h"
diff -urN mplayer-svn-r23664.orig/libvo/sub.c mplayer-svn-r23664/libvo/sub.c
--- mplayer-svn-r23664.orig/libvo/sub.c	2007-06-21 23:45:23 +0300
+++ mplayer-svn-r23664/libvo/sub.c	2007-06-26 01:24:03 +0300
@@ -21,7 +21,12 @@
 #include "font_load.h"
 #include "sub.h"
 #include "spudec.h"
+
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #define NEW_SPLITTING
 
diff -urN mplayer-svn-r23664.orig/libvo/vo_aa.c mplayer-svn-r23664/libvo/vo_aa.c
--- mplayer-svn-r23664.orig/libvo/vo_aa.c	2007-06-24 16:02:00 +0300
+++ mplayer-svn-r23664/libvo/vo_aa.c	2007-06-26 01:24:03 +0300
@@ -27,7 +27,11 @@
 #include "video_out.h"
 #include "video_out_internal.h"
 #include "aspect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #include "font_load.h"
 #include "sub.h"
diff -urN mplayer-svn-r23664.orig/libvo/vo_vesa.c mplayer-svn-r23664/libvo/vo_vesa.c
--- mplayer-svn-r23664.orig/libvo/vo_vesa.c	2007-06-05 23:28:28 +0300
+++ mplayer-svn-r23664/libvo/vo_vesa.c	2007-06-26 01:24:03 +0300
@@ -39,7 +39,11 @@
 
 #include "fastmemcpy.h"
 #include "sub.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "aspect.h"
 #include "vesa_lvo.h"
@@ -48,7 +52,11 @@
 #endif
 #include "mp_msg.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 
 
diff -urN mplayer-svn-r23664.orig/libvo/vo_x11.c mplayer-svn-r23664/libvo/vo_x11.c
--- mplayer-svn-r23664.orig/libvo/vo_x11.c	2007-05-28 19:50:49 +0300
+++ mplayer-svn-r23664/libvo/vo_x11.c	2007-06-26 01:24:03 +0300
@@ -37,7 +37,11 @@
 
 #include "sub.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #define MODE_RGB  0x1
 #define MODE_BGR  0x2
diff -urN mplayer-svn-r23664.orig/libvo/vo_xv.c mplayer-svn-r23664/libvo/vo_xv.c
--- mplayer-svn-r23664.orig/libvo/vo_xv.c	2007-04-23 23:01:48 +0300
+++ mplayer-svn-r23664/libvo/vo_xv.c	2007-06-26 01:24:03 +0300
@@ -43,7 +43,11 @@
 #include "gui/interface.h"
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 static vo_info_t info = {
     "X11/Xv",
diff -urN mplayer-svn-r23664.orig/libvo/vo_xvmc.c mplayer-svn-r23664/libvo/vo_xvmc.c
--- mplayer-svn-r23664.orig/libvo/vo_xvmc.c	2007-05-28 01:08:55 +0300
+++ mplayer-svn-r23664/libvo/vo_xvmc.c	2007-06-26 01:24:03 +0300
@@ -35,7 +35,11 @@
 #include "gui/interface.h"
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 //no chanse xinerama to be suported in near future
 #undef HAVE_XINERAMA
@@ -46,7 +50,11 @@
 
 #define UNUSED(x) ((void)(x))
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#else
 #include "libavcodec/avcodec.h"
+#endif
 #if LIBAVCODEC_BUILD < ((51<<16)+(40<<8)+2)
 #error You need at least libavcodecs v51.40.2
 #endif
diff -urN mplayer-svn-r23664.orig/libvo/vo_yuv4mpeg.c mplayer-svn-r23664/libvo/vo_yuv4mpeg.c
--- mplayer-svn-r23664.orig/libvo/vo_yuv4mpeg.c	2007-06-05 23:28:28 +0300
+++ mplayer-svn-r23664/libvo/vo_yuv4mpeg.c	2007-06-26 01:24:03 +0300
@@ -40,10 +40,19 @@
 #include "sub.h"
 
 #include "fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/swscale.h"
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/rational.h>
+#else
 #include "libavutil/rational.h"
+#endif
 
 static vo_info_t info = 
 {
diff -urN mplayer-svn-r23664.orig/Makefile mplayer-svn-r23664/Makefile
--- mplayer-svn-r23664.orig/Makefile	2007-05-12 02:32:51 +0300
+++ mplayer-svn-r23664/Makefile	2007-06-26 01:24:03 +0300
@@ -52,13 +52,14 @@
               libaf/libaf.a \
               libmpdemux/libmpdemux.a \
               stream/stream.a \
-              libswscale/libswscale.a \
               libvo/libosd.a \
 
 COMMON_LIBS-$(LIBAVFORMAT_A)      += libavformat/libavformat.a
 COMMON_LIBS-$(LIBAVCODEC_A)       += libavcodec/libavcodec.a
 COMMON_LIBS-$(LIBAVUTIL_A)        += libavutil/libavutil.a
 COMMON_LIBS-$(LIBPOSTPROC_A)      += libpostproc/libpostproc.a
+COMMON_LIBS-$(LIBSWSCALE_A)       += libswscale/libswscale.a
+PARTS-$(CONFIG_LIBSWSCALE)        += libswscale
 COMMON_LIBS-$(WIN32DLL)           += loader/libloader.a
 COMMON_LIBS-$(MP3LIB)             += mp3lib/libmp3.a
 COMMON_LIBS-$(LIBA52)             += liba52/liba52.a
diff -urN mplayer-svn-r23664.orig/mp3lib/dct64_sse.c mplayer-svn-r23664/mp3lib/dct64_sse.c
--- mplayer-svn-r23664.orig/mp3lib/dct64_sse.c	2007-06-07 00:46:28 +0300
+++ mplayer-svn-r23664/mp3lib/dct64_sse.c	2007-06-26 01:24:03 +0300
@@ -5,7 +5,12 @@
  * and mp3lib/dct64_MMX.c
  */
 
-#include <libavutil/mem.h>
+#include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/mem.h>
+#else
+#include "libavutil/mem.h"
+#endif
 
 typedef float real;
 
diff -urN mplayer-svn-r23664.orig/mp3lib/sr1.c mplayer-svn-r23664/mp3lib/sr1.c
--- mplayer-svn-r23664.orig/mp3lib/sr1.c	2007-06-09 04:08:57 +0300
+++ mplayer-svn-r23664/mp3lib/sr1.c	2007-06-26 01:24:03 +0300
@@ -24,7 +24,11 @@
 #include "mpg123.h"
 #include "huffman.h"
 #include "mp3.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "cpudetect.h"
 //#include "liba52/mm_accel.h"
diff -urN mplayer-svn-r23664.orig/mpbswap.h mplayer-svn-r23664/mpbswap.h
--- mplayer-svn-r23664.orig/mpbswap.h	2007-06-24 23:25:10 +0300
+++ mplayer-svn-r23664/mpbswap.h	2007-06-26 01:24:03 +0300
@@ -1,7 +1,12 @@
 #ifndef MPBSWAP_H
 #define MPBSWAP_H
 
+#include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/bswap.h>
+#else
 #include "libavutil/bswap.h"
+#endif
 #ifndef HAVE_SWAB
 void swab(const void *from, void *to, ssize_t n);
 #endif
diff -urN mplayer-svn-r23664.orig/spudec.c mplayer-svn-r23664/spudec.c
--- mplayer-svn-r23664.orig/spudec.c	2007-06-21 23:45:26 +0300
+++ mplayer-svn-r23664/spudec.c	2007-06-26 01:24:03 +0300
@@ -28,7 +28,11 @@
 #else
 #include "avutil.h"
 #endif
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 
 /* Valid values for spu_aamode:
    0: none (fastest, most ugly)
diff -urN mplayer-svn-r23664.orig/stream/realrtsp/real.c mplayer-svn-r23664/stream/realrtsp/real.c
--- mplayer-svn-r23664.orig/stream/realrtsp/real.c	2007-06-24 23:25:08 +0300
+++ mplayer-svn-r23664/stream/realrtsp/real.c	2007-06-26 01:24:03 +0300
@@ -31,17 +31,22 @@
 #include <string.h>
 
 #include "../config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "real.h"
 #include "asmrp.h"
 #include "sdpplin.h"
 #include "xbuffer.h"
 #if USE_LIBAVUTIL_SO
-#include "ffmpeg/md5.h"
+#include <ffmpeg/md5.h>
+#include <ffmpeg/intreadwrite.h>
 #else
 #include "libavutil/md5.h"
-#endif
 #include "libavutil/intreadwrite.h"
+#endif
 #include "../http.h"
 #include "mp_msg.h"
 
diff -urN mplayer-svn-r23664.orig/stream/realrtsp/rmff.c mplayer-svn-r23664/stream/realrtsp/rmff.c
--- mplayer-svn-r23664.orig/stream/realrtsp/rmff.c	2007-03-27 00:16:50 +0300
+++ mplayer-svn-r23664/stream/realrtsp/rmff.c	2007-06-26 01:24:03 +0300
@@ -29,7 +29,11 @@
 #include "rmff.h"
 #include "xbuffer.h"
 #include "mp_msg.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 
 /*
 #define LOG
diff -urN mplayer-svn-r23664.orig/stream/stream_cdda.c mplayer-svn-r23664/stream/stream_cdda.c
--- mplayer-svn-r23664.orig/stream/stream_cdda.c	2006-12-08 20:58:10 +0200
+++ mplayer-svn-r23664/stream/stream_cdda.c	2007-06-26 01:24:03 +0300
@@ -6,7 +6,11 @@
 #include "stream.h"
 #include "m_option.h"
 #include "m_struct.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "libmpdemux/demuxer.h"
 
diff -urN mplayer-svn-r23664.orig/stream/stream_dvdnav.c mplayer-svn-r23664/stream/stream_dvdnav.c
--- mplayer-svn-r23664.orig/stream/stream_dvdnav.c	2007-04-23 01:35:36 +0300
+++ mplayer-svn-r23664/stream/stream_dvdnav.c	2007-06-26 01:24:03 +0300
@@ -11,7 +11,11 @@
 #include "libmpdemux/demuxer.h"
 #include "stream_dvdnav.h"
 #include "libvo/video_out.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "spudec.h"
 #include "m_option.h"
 #include "m_struct.h"
diff -urN mplayer-svn-r23664.orig/stream/stream_netstream.c mplayer-svn-r23664/stream/stream_netstream.c
--- mplayer-svn-r23664.orig/stream/stream_netstream.c	2007-01-22 19:01:51 +0200
+++ mplayer-svn-r23664/stream/stream_netstream.c	2007-06-26 01:24:03 +0300
@@ -58,7 +58,11 @@
 #include "help_mp.h"
 #include "m_option.h"
 #include "m_struct.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 #include "netstream.h"
diff -urN mplayer-svn-r23664.orig/TOOLS/Makefile mplayer-svn-r23664/TOOLS/Makefile
--- mplayer-svn-r23664.orig/TOOLS/Makefile	2007-04-10 03:17:56 +0300
+++ mplayer-svn-r23664/TOOLS/Makefile	2007-06-26 01:24:03 +0300
@@ -15,8 +15,8 @@
 
 subrip$(EXESUF): subrip.c
 	$(CC) $(CFLAGS) -g -o $@ $< ../vobsub.o ../spudec.o ../mp_msg.o \
-          ../unrarlib.o ../libswscale/libswscale.a ../libavutil/libavutil.a \
-          ../libmpcodecs/img_format.o ../osdep/getch2.o -ltermcap -lm
+          ../unrarlib.o ../libmpcodecs/img_format.o ../osdep/getch2.o \
+          -L../libswscale -lswscale -L../libavutil -lavutil -ltinfo -lm
 
 #FIXME: Linking is broken, help welcome.
 vivodump$(EXESUF): vivodump.c
diff -urN mplayer-svn-r23664.orig/vidix/drivers.c mplayer-svn-r23664/vidix/drivers.c
--- mplayer-svn-r23664.orig/vidix/drivers.c	2007-05-11 01:00:46 +0300
+++ mplayer-svn-r23664/vidix/drivers.c	2007-06-26 01:24:03 +0300
@@ -26,7 +26,11 @@
 
 #include "vidixlib.h"
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "config.h"
 
diff -urN mplayer-svn-r23664.orig/vidix/mach64_vid.c mplayer-svn-r23664/vidix/mach64_vid.c
--- mplayer-svn-r23664.orig/vidix/mach64_vid.c	2007-05-02 22:42:49 +0300
+++ mplayer-svn-r23664/vidix/mach64_vid.c	2007-06-26 01:24:03 +0300
@@ -31,7 +31,11 @@
 #include <fcntl.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "vidix.h"
 #include "vidixlib.h"
diff -urN mplayer-svn-r23664.orig/vidix/nvidia_vid.c mplayer-svn-r23664/vidix/nvidia_vid.c
--- mplayer-svn-r23664.orig/vidix/nvidia_vid.c	2007-04-23 01:35:47 +0300
+++ mplayer-svn-r23664/vidix/nvidia_vid.c	2007-06-26 01:24:03 +0300
@@ -38,7 +38,11 @@
 #include "pci_ids.h"
 #include "pci_names.h"
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 
diff -urN mplayer-svn-r23664.orig/vidix/radeon_vid.c mplayer-svn-r23664/vidix/radeon_vid.c
--- mplayer-svn-r23664.orig/vidix/radeon_vid.c	2007-05-02 22:42:49 +0300
+++ mplayer-svn-r23664/vidix/radeon_vid.c	2007-06-26 01:24:03 +0300
@@ -37,7 +37,11 @@
 #include <inttypes.h>
 
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 #include "pci_ids.h"
 #include "pci_names.h"
diff -urN mplayer-svn-r23664.orig/vidix/sysdep/pci_linux.c mplayer-svn-r23664/vidix/sysdep/pci_linux.c
--- mplayer-svn-r23664.orig/vidix/sysdep/pci_linux.c	2007-04-10 03:17:39 +0300
+++ mplayer-svn-r23664/vidix/sysdep/pci_linux.c	2007-06-26 01:24:03 +0300
@@ -132,7 +132,11 @@
 #include <fcntl.h>
 //#include <sys/io.h>
 #include <linux/pci.h>
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 static int pci_get_vendor(
diff -urN mplayer-svn-r23664.orig/vidix/vidixlib.c mplayer-svn-r23664/vidix/vidixlib.c
--- mplayer-svn-r23664.orig/vidix/vidixlib.c	2007-04-23 01:35:47 +0300
+++ mplayer-svn-r23664/vidix/vidixlib.c	2007-06-26 01:24:03 +0300
@@ -40,7 +40,11 @@
 #include "vidixlib.h"
 #include "drivers.h"
 #include "config.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "mpbswap.h"
 
 extern unsigned int  vdlGetVersion( void )
diff -urN mplayer-svn-r23664.orig/vobsub.c mplayer-svn-r23664/vobsub.c
--- mplayer-svn-r23664.orig/vobsub.c	2007-03-12 02:50:41 +0200
+++ mplayer-svn-r23664/vobsub.c	2007-06-26 01:24:03 +0300
@@ -23,7 +23,11 @@
 #ifdef USE_UNRARLIB
 #include "unrarlib.h"
 #endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 extern int vobsub_id;
 
