diff -urN mplayer-svn-r22092.orig/configure mplayer-svn-r22092/configure
--- mplayer-svn-r22092.orig/configure	2007-01-31 13:31:44 +0200
+++ mplayer-svn-r22092/configure	2007-01-31 14:50:37 +0200
@@ -129,7 +129,7 @@
 
 x86_64() {
   case "$host_arch" in
-    x86_64|amd64) return 0 ;;
+    x86-64|x86_64|amd64) return 0 ;;
     *) return 1 ;;
   esac
 }
@@ -1726,7 +1726,7 @@
     _optimizing=''
     ;;
 
-  x86_64|amd64)
+  x86-64|x86_64|amd64)
     _def_arch='#define ARCH_X86_64 1'
     _target_arch='TARGET_ARCH_X86_64 = yes'
     _def_arch_x86="#define ARCH_X86 1"
@@ -2799,6 +2799,7 @@
   _noinputmodules="network $_noinputmodules"
   _def_network='#undef MPLAYER_NETWORK'
   _ftp=no
+  _live=no
 fi
 echores "$_network"
 
@@ -3272,7 +3273,7 @@
 
 
 echocheck "termios"
-if test "$_termios" = auto ; then
+if test "$_termios" != no ; then
   cat > $TMPC <<EOF
 #include <sys/termios.h>
 int main(void) { return 0; }
@@ -4224,7 +4225,9 @@
 if test "$_aa" = yes ; then
   _def_aa='#define HAVE_AA 1'
   if cygwin ; then
-    _libs_mplayer="$_libs_mplayer `aalib-config --libs | cut -d " " -f 2,5,6`"
+    _libs_mplayer="$_libs_mplayer `aalib-config --libs | cut -d ' ' -f 2,5,6`"
+  else
+    _libs_mplayer="$_libs_mplayer `aalib-config --libs | cut -d ' ' -f 1,2,4-6`"
   fi
   _vosrc="$_vosrc vo_aa.c"
   _vomodules="aa $_vomodules"
@@ -4252,7 +4255,7 @@
 if test "$_caca" = yes ; then
   _def_caca='#define HAVE_CACA 1'
   _inc_extra="$_inc_extra `caca-config --cflags`"
-  _libs_mplayer="$_libs_mplayer `caca-config --libs`"
+  _libs_mplayer="$_libs_mplayer `caca-config --libs | cut -d ' ' -f 1-4`"
   _vosrc="$_vosrc vo_caca.c"
   _vomodules="caca $_vomodules"
 else
@@ -4917,7 +4920,7 @@
 
 
 echocheck "JACK"
-if test "$_jack" = auto ; then
+if test "$_jack" != no ; then
   _jack=yes
 
 cat > $TMPC << EOF
@@ -4944,7 +4947,7 @@
 echores "$_jack"
 
 echocheck "OpenAL"
-if test "$_openal" = auto ; then
+if test "$_openal" != no ; then
   _openal=no
 cat > $TMPC << EOF
 #ifdef OPENAL_AL_H
@@ -5402,7 +5405,7 @@
     _res_comment="freetype support needed"
 fi
 echocheck "fontconfig"
-if test "$_fontconfig" = auto ; then
+if test "$_fontconfig" != no ; then
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <fontconfig/fontconfig.h>
@@ -5720,7 +5723,7 @@
 echores "$_speex"
 
 echocheck "OggTheora support"
-if test "$_theora" = auto ; then
+if test "$_theora" != no ; then
   _theora=no
     cat > $TMPC << EOF
 #include <theora/theora.h>
@@ -5800,7 +5803,7 @@
 echores "$_liba52"
 
 echocheck "libdts support"
-if test "$_libdts" = auto ; then
+if test "$_libdts" != no ; then
   _libdts=no
   cat > $TMPC << EOF
 #include <inttypes.h>
@@ -5855,7 +5858,7 @@
 
 
 echocheck "FAAC (AAC encoder) support"
-if test "$_faac" = auto ; then
+if test "$_faac" != no ; then
   cat > $TMPC <<EOF
 #include <inttypes.h> 
 #include <faac.h>
@@ -5975,7 +5978,7 @@
 
 
 echocheck "Win32 codecs"
-if test "$_win32" = auto ; then
+if test "$_win32" != no ; then
   _win32=no
   if x86_32 && not qnx; then
     _win32=yes
@@ -6046,56 +6049,6 @@
 echores "$_real"
 
 
-echocheck "LIVE555 Streaming Media libraries"
-if test "$_live" = auto  && test "$_network" = yes ; then
-  cat > $TMPCPP << EOF
-#include <liveMedia.hh>
-#if (LIVEMEDIA_LIBRARY_VERSION_INT < 1090195200)
-#error Please upgrade to version 2004.07.19 or later of the "LIVE555 Streaming Media" libraries - available from <www.live555.com/liveMedia/>
-#endif
-int main(void) {}
-EOF
-
-  _live=no
-  for I in $_inc_extra "-I$_libdir/live" "-I/usr/lib/live" "-I/usr/local/live" "-I/usr/local/lib/live" ; do
-    cxx_check $I/liveMedia/include $I/UsageEnvironment/include \
-      $I/groupsock/include && _livelibdir=`echo $I| sed s/-I//` && \
-      _live=yes && break
-  done
-  if test "$_live" != yes ; then
-      if cxx_check -I/usr/include/liveMedia -I/usr/include/UsageEnvironment -I/usr/include/groupsock; then
-	  _live_dist=yes
-      fi
-  fi
-fi
-if test "$_live" = yes && test "$_network" = yes ; then
-  _res_comment="using $_livelibdir"
-  _def_live='#define STREAMING_LIVE555 1'
-  _ld_extra="$_ld_extra $_livelibdir/liveMedia/libliveMedia.a \
-           $_livelibdir/groupsock/libgroupsock.a \
-           $_livelibdir/UsageEnvironment/libUsageEnvironment.a \
-           $_livelibdir/BasicUsageEnvironment/libBasicUsageEnvironment.a \
-           -lstdc++"
-  _inc_extra="$_inc_extra -I$_livelibdir/liveMedia/include \
-             -I$_livelibdir/UsageEnvironment/include \
-             -I$_livelibdir/BasicUsageEnvironment/include \
-             -I$_livelibdir/groupsock/include"
- _inputmodules="live555 $_inputmodules"
-elif test "$_live_dist" = yes && test "$_network" = yes ; then
-  _res_comment="using distribution version"
-  _live="yes"
-  _def_live='#define STREAMING_LIVE555 1'
-  _ld_extra="$_ld_extra -lliveMedia -lgroupsock -lUsageEnvironment -lBasicUsageEnvironment -lstdc++"
-  _inc_extra="$_inc_extra -I/usr/include/liveMedia -I/usr/include/UsageEnvironment \
-             -I/usr/include/BasicUsageEnvironment -I/usr/include/groupsock"
- _inputmodules="live555 $_inputmodules"
-else
-  _def_live='#undef STREAMING_LIVE555'
-  _noinputmodules="live555 $_noinputmodules"
-fi
-echores "$_live"
-
-
 echocheck "FFmpeg libavutil (static)"
 if test "$_libavutil" = auto ; then
   if test -d libavutil ; then
@@ -6160,7 +6113,7 @@
 
 if test "$_libavutil" != yes ; then
 echocheck "FFmpeg libavutil (dynamic)"
-if test "$_libavutil_so" = auto ; then
+if test "$_libavutil_so" != no ; then
   _libavutil_so=no
   cat > $TMPC << EOF
 #include <ffmpeg/common.h>
@@ -6186,7 +6139,7 @@
 
 if test "$_libavcodec" != yes ; then
 echocheck "FFmpeg libavcodec (dynamic)"
-if test "$_libavcodec_so" = auto ; then
+if test "$_libavcodec_so" != no ; then
   _libavcodec_so=no
   _res_comment="libavcodec.so is broken/obsolete"
 # FIXME : check for avcodec_find_encoder_by_name() for mencoder
@@ -6213,7 +6166,7 @@
 
 if test "$_libavformat" != yes ; then
 echocheck "FFmpeg libavformat (dynamic)"
-if test "$_libavformat_so" = auto ; then
+if test "$_libavformat_so" != no ; then
   _libavformat_so=no
   cat > $TMPC <<EOF
   #include <ffmpeg/avformat.h>
@@ -6236,7 +6189,7 @@
 
 if test "$_libpostproc" != yes ; then
 echocheck "FFmpeg libpostproc (dynamic)"
-if test "$_libpostproc_so" = auto ; then
+if test "$_libpostproc_so" != no ; then
   _libpostproc_so=no
   cat > $TMPC << EOF
   #define USE_LIBPOSTPROC 1
@@ -6449,7 +6402,7 @@
 
 
 echocheck "XviD"
-if test "$_xvid" = auto ; then
+if test "$_xvid" != no ; then
   _xvid=no
   cat > $TMPC << EOF
 #include <xvid.h>
@@ -6487,7 +6440,7 @@
 
 
 echocheck "x264"
-if test "$_x264" = auto ; then
+if test "$_x264" != no ; then
   cat > $TMPC << EOF
 #include <inttypes.h>
 #include <x264.h>
@@ -6525,7 +6478,7 @@
 
 
 echocheck "nut"
-if test "$_nut" = auto ; then
+if test "$_nut" != no ; then
   cat > $TMPC << EOF
 #include <stdio.h>
 #include <stdlib.h>
@@ -6836,6 +6789,53 @@
 fi
 echores "$_ftp"
 
+echocheck "LIVE555 Streaming Media libraries"
+if test "$_live" != no  && test "$_network" = yes ; then
+  cat > $TMPCPP << EOF
+#include <liveMedia.hh>
+#if (LIVEMEDIA_LIBRARY_VERSION_INT < 1090195200)
+#error Please upgrade to version 2004.07.19 or later of the "LIVE555 Streaming Media" libraries - available from <www.live555.com/liveMedia/>
+#endif
+int main(void) {}
+EOF
+
+  _live=no
+  for I in "$_livelibdir" "$_libdir/live" "/usr/lib/live" "/usr/local/live" "/usr/local/lib/live" ; do
+    cxx_check -I$I/liveMedia/include -I$I/UsageEnvironment/include -I$I/groupsock/include && _livelibdir=$I && _live=yes && break
+  done
+  if test "$_live" != yes ; then
+      if cxx_check -I/usr/include/liveMedia -I/usr/include/UsageEnvironment -I/usr/include/groupsock; then
+	  _live_dist=yes
+      fi
+  fi
+fi
+if test "$_live" = yes && test "$_network" = yes ; then
+  _res_comment="using $_livelibdir"
+  _def_live='#define STREAMING_LIVE555 1'
+  _ld_extra="$_ld_extra $_livelibdir/liveMedia/libliveMedia.a \
+           $_livelibdir/groupsock/libgroupsock.a \
+           $_livelibdir/UsageEnvironment/libUsageEnvironment.a \
+           $_livelibdir/BasicUsageEnvironment/libBasicUsageEnvironment.a \
+           -lstdc++"
+  _inc_extra="$_inc_extra -I$_livelibdir/liveMedia/include \
+             -I$_livelibdir/UsageEnvironment/include \
+             -I$_livelibdir/BasicUsageEnvironment/include \
+             -I$_livelibdir/groupsock/include"
+ _inputmodules="live555 $_inputmodules"
+elif test "$_live_dist" = yes && test "$_network" = yes ; then
+  _res_comment="using distribution version"
+  _live="yes"
+  _def_live='#define STREAMING_LIVE555 1'
+  _ld_extra="$_ld_extra -lliveMedia -lgroupsock -lUsageEnvironment -lBasicUsageEnvironment -lstdc++"
+  _inc_extra="$_inc_extra -I/usr/include/liveMedia -I/usr/include/UsageEnvironment \
+             -I/usr/include/BasicUsageEnvironment -I/usr/include/groupsock"
+ _inputmodules="live555 $_inputmodules"
+else
+  _def_live='#undef STREAMING_LIVE555'
+  _noinputmodules="live555 $_noinputmodules"
+fi
+echores "$_live"
+
 echocheck "vstream client"
 if test "$_vstream" = auto ; then
   _vstream=no
@@ -6931,7 +6931,7 @@
   if darwin ; then
      _ld_extra="$_ld_extra ${_xmmslibdir}/libxmms.dylib"
   else
-     _ld_extra="$_ld_extra ${_xmmslibdir}/libxmms.so.1 -export-dynamic"
+     _ld_extra="$_ld_extra `xmms-config --libs`"
   fi
 else
   _def_xmms='#undef HAVE_XMMS'
@@ -7254,6 +7254,9 @@
 if test "$_vidix_internal" = yes || test "$_vidix_external" = yes ; then
   _vidix=yes
   _def_vidix='#define CONFIG_VIDIX 1'
+  if test "$_vidix_external" = yes ; then
+    _def_vidix_ext='#define VIDIX_EXT 1'
+  fi
 else
   _vidix=no
   _def_vidix='#undef CONFIG_VIDIX'
@@ -7263,7 +7266,7 @@
   _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/mplayer/vidix/\" "
 elif test "$_vidix_external" = yes ; then
   _libs_mplayer="$_libs_mplayer -lvidix"
-  _def_vidix_pfx='#define MP_VIDIX_PFX "" '
+  _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/vidix/\" "
 fi
 
 if test "$_vidix" = yes; then
@@ -8239,6 +8242,7 @@
 /* enables / disables VIDIX usage */
 $_def_vidix
 $_def_vidix_pfx
+$_def_vidix_ext
 
 /* enables / disables new input joystick support */
 $_def_joystick
