diff -urN mplayer-svn-r22092.orig/etc/codecs.conf mplayer-svn-r22092/etc/codecs.conf
--- mplayer-svn-r22092.orig/etc/codecs.conf	2007-01-09 13:19:53 +0200
+++ mplayer-svn-r22092/etc/codecs.conf	2007-02-05 01:04:42 +0200
@@ -425,6 +425,14 @@
   dll snow
   out YV12
 
+videocodec ffdirac
+  info "FFDIRAC (Dirac wavelet codec)"
+  status working
+  fourcc drac
+  driver ffmpeg
+  dll dirac
+  out YV12
+
 videocodec ffasv1
   info "FFmpeg ASUS V1"
   status working
diff -urN mplayer-svn-r22092.orig/libmpcodecs/ve_lavc.c mplayer-svn-r22092/libmpcodecs/ve_lavc.c
--- mplayer-svn-r22092.orig/libmpcodecs/ve_lavc.c	2006-11-08 14:49:37 +0200
+++ mplayer-svn-r22092/libmpcodecs/ve_lavc.c	2007-02-05 01:03:26 +0200
@@ -986,6 +986,8 @@
 	mux_v->bih->biCompression = mmioFOURCC('F', 'F', 'V', '1');
     else if (!strcasecmp(lavc_param_vcodec, "snow"))
 	mux_v->bih->biCompression = mmioFOURCC('S', 'N', 'O', 'W');
+    else if (!strcasecmp(lavc_param_vcodec, "dirac"))
+	mux_v->bih->biCompression = mmioFOURCC('d', 'r', 'a', 'c');
     else if (!strcasecmp(lavc_param_vcodec, "flv"))
 	mux_v->bih->biCompression = mmioFOURCC('F', 'L', 'V', '1');
     else
