diff -urN MPlayer-1.0pre4.orig/configure MPlayer-1.0pre4/configure
--- MPlayer-1.0pre4.orig/configure	2004-05-04 21:14:51 +0700
+++ MPlayer-1.0pre4/configure	2004-05-04 21:17:13 +0700
@@ -3991,7 +3991,7 @@
   _def_arts='#define USE_ARTS 1'
   _aosrc="$_aosrc ao_arts.c"
   _aomodules="arts $_aomodules"
-  _ld_arts=`artsc-config --libs`
+  _ld_arts="-lartsc"
   _inc_arts=`artsc-config --cflags`
 else
   _noaomodules="arts $_noaomodules"
