diff -urN mplayer-svn-r19389.orig/libmpcodecs/vf_qp.c mplayer-svn-r19389/libmpcodecs/vf_qp.c
--- mplayer-svn-r19389.orig/libmpcodecs/vf_qp.c	2006-08-13 21:41:44 +0300
+++ mplayer-svn-r19389/libmpcodecs/vf_qp.c	2006-08-17 13:45:33 +0300
@@ -37,9 +37,15 @@
 // Needed to bring in lrintf.
 #define HAVE_AV_CONFIG_H
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#include <ffmpeg/dsputil.h>
+#include <ffmpeg/common.h>
+#else
 #include "libavcodec/avcodec.h"
 #include "libavcodec/dsputil.h"
 #include "libavutil/common.h"
+#endif
 
 /* FIXME: common.h defines printf away when HAVE_AV_CONFIG
  * is defined, but mp_image.h needs printf.
diff -urN mplayer-svn-r19389.orig/libvo/sub.c mplayer-svn-r19389/libvo/sub.c
--- mplayer-svn-r19389.orig/libvo/sub.c	2006-08-13 21:59:25 +0300
+++ mplayer-svn-r19389/libvo/sub.c	2006-08-17 13:46:24 +0300
@@ -14,7 +14,12 @@
 #include "font_load.h"
 #include "sub.h"
 #include "spudec.h"
+
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #define NEW_SPLITTING
 
