diff -urN mplayer-svn-r21352.orig/bswap.h mplayer-svn-r21352/bswap.h
--- mplayer-svn-r21352.orig/bswap.h	2006-11-27 14:05:42 +0200
+++ mplayer-svn-r21352/bswap.h	2006-11-28 18:59:38 +0200
@@ -1,8 +1,121 @@
 #ifndef __MP_BSWAP_H__
 #define __MP_BSWAP_H__
 
-#include "libavutil/common.h"
-#include "libavutil/bswap.h"
+#ifdef HAVE_BYTESWAP_H
+#include <byteswap.h>
+#else
+
+#include <inttypes.h>
+
+#ifdef ARCH_X86_64
+#  define LEGACY_REGS "=Q"
+#else
+#  define LEGACY_REGS "=q"
+#endif
+
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
+static inline uint16_t ByteSwap16(uint16_t x)
+{
+  __asm("xchgb %b0,%h0"	:
+        LEGACY_REGS (x)	:
+        "0" (x));
+    return x;
+}
+#define bswap_16(x) ByteSwap16(x)
+
+static inline uint32_t ByteSwap32(uint32_t x)
+{
+#if __CPU__ != 386
+ __asm("bswap	%0":
+      "=r" (x)     :
+#else
+ __asm("xchgb	%b0,%h0\n"
+      "	rorl	$16,%0\n"
+      "	xchgb	%b0,%h0":
+      LEGACY_REGS (x)		:
+#endif
+      "0" (x));
+  return x;
+}
+#define bswap_32(x) ByteSwap32(x)
+
+static inline uint64_t ByteSwap64(uint64_t x)
+{
+#ifdef ARCH_X86_64
+  __asm("bswap	%0":
+        "=r" (x)     :
+        "0" (x));
+  return x;
+#else
+  register union { __extension__ uint64_t __ll;
+          uint32_t __l[2]; } __x;
+  asm("xchgl	%0,%1":
+      "=r"(__x.__l[0]),"=r"(__x.__l[1]):
+      "0"(bswap_32((unsigned long)x)),"1"(bswap_32((unsigned long)(x>>32))));
+  return __x.__ll;
+#endif
+}
+#define bswap_64(x) ByteSwap64(x)
+
+#elif defined(ARCH_SH4)
+
+static inline uint16_t ByteSwap16(uint16_t x) {
+	__asm__("swap.b %0,%0":"=r"(x):"0"(x));
+	return x;
+}
+
+static inline uint32_t ByteSwap32(uint32_t x) {
+	__asm__(
+	"swap.b %0,%0\n"
+	"swap.w %0,%0\n"
+	"swap.b %0,%0\n"
+	:"=r"(x):"0"(x));
+	return x;
+}
+
+#define bswap_16(x) ByteSwap16(x)
+#define bswap_32(x) ByteSwap32(x)
+
+static inline uint64_t ByteSwap64(uint64_t x)
+{
+    union { 
+        uint64_t ll;
+        struct {
+           uint32_t l,h;
+        } l;
+    } r;
+    r.l.l = bswap_32 (x);
+    r.l.h = bswap_32 (x>>32);
+    return r.ll;
+}
+#define bswap_64(x) ByteSwap64(x)
+
+#else
+
+#define bswap_16(x) (((x) & 0x00ff) << 8 | ((x) & 0xff00) >> 8)
+			
+
+// code from bits/byteswap.h (C) 1997, 1998 Free Software Foundation, Inc.
+#define bswap_32(x) \
+     ((((x) & 0xff000000) >> 24) | (((x) & 0x00ff0000) >>  8) | \
+      (((x) & 0x0000ff00) <<  8) | (((x) & 0x000000ff) << 24))
+
+static inline uint64_t ByteSwap64(uint64_t x)
+{
+    union { 
+        uint64_t ll;
+        uint32_t l[2]; 
+    } w, r;
+    w.ll = x;
+    r.l[0] = bswap_32 (w.l[1]);
+    r.l[1] = bswap_32 (w.l[0]);
+    return r.ll;
+}
+#define bswap_64(x) ByteSwap64(x)
+
+#endif	/* !ARCH_X86 */
+
+#endif	/* !HAVE_BYTESWAP_H */
 
 static inline float bswap_flt(float x) {
   union {uint32_t i; float f;} u;
@@ -39,6 +152,12 @@
 // le2me ... LittleEndian to MachineEndian
 
 #ifdef WORDS_BIGENDIAN
+#define be2me_16(x) (x)
+#define be2me_32(x) (x)
+#define be2me_64(x) (x)
+#define le2me_16(x) bswap_16(x)
+#define le2me_32(x) bswap_32(x)
+#define le2me_64(x) bswap_64(x)
 #define be2me_flt(x) (x)
 #define be2me_dbl(x) (x)
 #define be2me_ldbl(x) (x)
@@ -46,6 +165,12 @@
 #define le2me_dbl(x) bswap_dbl(x)
 #define le2me_ldbl(x) bswap_ldbl(x)
 #else
+#define be2me_16(x) bswap_16(x)
+#define be2me_32(x) bswap_32(x)
+#define be2me_64(x) bswap_64(x)
+#define le2me_16(x) (x)
+#define le2me_32(x) (x)
+#define le2me_64(x) (x)
 #define be2me_flt(x) bswap_flt(x)
 #define be2me_dbl(x) bswap_dbl(x)
 #define be2me_ldbl(x) bswap_ldbl(x)
diff -urN mplayer-svn-r21352.orig/configure mplayer-svn-r21352/configure
--- mplayer-svn-r21352.orig/configure	2006-11-28 12:50:10 +0200
+++ mplayer-svn-r21352/configure	2006-11-28 19:03:06 +0200
@@ -283,10 +283,12 @@
   --disable-libavcodec      disable libavcodec [autodetect]
   --disable-libavformat     disable libavformat [autodetect]
   --disable-libpostproc     disable libpostproc [autodetect]
+  --disable-libswscale      disable libswscale [autodetect]
   --disable-libavutil_so    disable shared libavutil [autodetect]
   --disable-libavcodec_so   disable shared libavcodec [autodetect]
   --disable-libavformat_so  disable shared libavformat [autodetect]
   --disable-libpostproc_so  disable shared libpostproc [autodetect]
+  --disable-libswscale_so disable shared libswscale [autodetect]
   --disable-libavcodec_mpegaudio_hp disable high precision audio decoding
                                     in libavcodec [enabled]
   --disable-tremor-internal disable internal Tremor [enabled]
@@ -1577,6 +1579,8 @@
 _libavformat_so=auto
 _libpostproc=auto
 _libpostproc_so=auto
+_libswscale=auto
+_libswscale_so=auto
 _libavcodec_mpegaudio_hp=yes
 _mencoder=yes
 _mplayer=yes
@@ -1978,6 +1982,10 @@
   --disable-libpostproc)	_libpostproc=no		;;
   --enable-libpostproc_so)	_libpostproc_so=yes	;;
   --disable-libpostproc_so)	_libpostproc_so=no	;;
+  --enable-libswscale)		_libswscale=yes		;;
+  --disable-libswscale)		_libswscale=no		;;
+  --enable-libswscale_so)	_libswscale_so=yes	;;
+  --disable-libswscale_so)	_libswscale_so=no	;;
   --enable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=yes	;;
   --disable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=no	;;
 
@@ -6097,6 +6105,15 @@
 fi
 echores "$_libpostproc"
 
+echocheck "FFmpeg libswscale (static)"
+if test "$_libswscale" = auto ; then
+_libswscale=no
+if test -d libswscale && test -f libswscale/swscale.h ; then
+   _libswscale='yes'
+fi
+fi
+echores "$_libswscale"
+
 
 if test "$_libavutil" != yes ; then
 echocheck "FFmpeg libavutil (dynamic)"
@@ -6195,6 +6212,27 @@
 echores "$_libpostproc_so"
 fi #if test "$_libpostproc" != yes ; then
 
+if test "$_libswscale" != yes ; then
+echocheck "FFmpeg libswscale (dynamic)"
+if test "$_libswscale_so" != no ; then
+  _libswscale_so=no
+  cat > $TMPC << EOF
+  #define USE_LIBPOSTPROC 1
+  #include <inttypes.h>
+  #include <ffmpeg/swscale.h>
+  int main(void) {
+      sws_getIdentityVec();
+      return 0;}
+EOF
+  if cc_check -lswscale $_ld_lm ; then
+    _ld_extra="$_ld_extra -lswscale"
+    _libswscale_so=yes
+    _res_comment="using libswscale.so, but static libswscale is recommended"
+  fi
+fi
+echores "$_libswscale_so"
+fi
+
 _def_libavutil='#undef USE_LIBAVUTIL'
 _def_libavutil_so='#undef USE_LIBAVUTIL_SO'
 if test "$_libavutil" = yes ; then
@@ -6248,6 +6286,19 @@
   fi
 fi
 
+_def_libswscale='#undef USE_LIBSWSCALE'
+_def_libswscale_so='#undef USE_LIBSWSCALE_SO'
+if test "$_libswscale" = yes ; then
+  _def_libswscale='#define USE_LIBSWSCALE 1'
+  _ld_libswscale='libswscale/libswscale.a'
+  _dep_libswscale='libswscale/libswscale.a'
+else
+  if test "$_libswscale_so" = yes ; then
+    _def_libswscale_so='#define USE_LIBSWSCALE_SO 1'
+    _ld_libswscale='-lswscale'
+  fi
+fi
+
 echocheck "md5sum support"
 if test "$_md5sum" = yes; then
   _def_md5sum="#define HAVE_MD5SUM"
@@ -7424,6 +7475,10 @@
 CONFIG_LIBAVFORMAT_SO = $_libavformat_so
 CONFIG_LIBPOSTPROC = $_libpostproc
 CONFIG_LIBPOSTPROC_SO = $_libpostproc_so
+CONFIG_LIBSWSCALE = $_libswscale
+CONFIG_SWSCALER = $_libswscale
+CONFIG_LIBSWSCALE_SO = $_libswscale_so
+SWSCALE_LIB = $_ld_libswscale
 ZORAN = $_zr
 LIBDV = $_libdv
 XVID4 = $_xvid
@@ -7781,6 +7836,9 @@
 $_def_libpostproc
 $_def_libpostproc_so
 
+$_def_libswscale
+$_def_libswscale_so
+
 /* Win32 DLL support */
 $_def_win32
 #define WIN32_PATH "$_win32codecsdir"
diff -urN mplayer-svn-r21352.orig/fmt-conversion.h mplayer-svn-r21352/fmt-conversion.h
--- mplayer-svn-r21352.orig/fmt-conversion.h	2006-09-24 19:47:48 +0300
+++ mplayer-svn-r21352/fmt-conversion.h	2006-11-28 18:59:38 +0200
@@ -1,4 +1,8 @@
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 #include "img_format.h"
 
 enum PixelFormat imgfmt2pixfmt(int fmt)
diff -urN mplayer-svn-r21352.orig/Gui/wm/ws.c mplayer-svn-r21352/Gui/wm/ws.c
--- mplayer-svn-r21352.orig/Gui/wm/ws.c	2006-07-12 16:44:27 +0300
+++ mplayer-svn-r21352/Gui/wm/ws.c	2006-11-28 18:59:38 +0200
@@ -24,8 +24,13 @@
 #include "ws.h"
 #include "wsxdnd.h"
 #include "../cpudetect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "../libswscale/swscale.h"
 #include "../libswscale/rgb2rgb.h"
+#endif
 #include "../libmpcodecs/vf_scale.h"
 #include "../mp_msg.h"
 #include "../help_mp.h"
diff -urN mplayer-svn-r21352.orig/liba52/parse.c mplayer-svn-r21352/liba52/parse.c
--- mplayer-svn-r21352.orig/liba52/parse.c	2006-11-09 16:34:19 +0200
+++ mplayer-svn-r21352/liba52/parse.c	2006-11-28 18:59:38 +0200
@@ -37,7 +37,11 @@
 #include "bitstream.h"
 #include "tables.h"
 #include "mm_accel.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "libavutil/avutil.h"
+#endif
 
 #ifdef HAVE_MEMALIGN
 /* some systems have memalign() but no declaration for it */
diff -urN mplayer-svn-r21352.orig/libass/mputils.h mplayer-svn-r21352/libass/mputils.h
--- mplayer-svn-r21352.orig/libass/mputils.h	2006-11-20 12:43:26 +0200
+++ mplayer-svn-r21352/libass/mputils.h	2006-11-28 18:59:38 +0200
@@ -6,6 +6,10 @@
 #include "libvo/font_load.h" // for blur()
 #include "subreader.h" // for guess_buffer_cp
 #include "libvo/sub.h" // for utf8_get_char
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #endif
diff -urN mplayer-svn-r21352.orig/libmpcodecs/vf_halfpack.c mplayer-svn-r21352/libmpcodecs/vf_halfpack.c
--- mplayer-svn-r21352.orig/libmpcodecs/vf_halfpack.c	2006-08-17 13:36:26 +0300
+++ mplayer-svn-r21352/libmpcodecs/vf_halfpack.c	2006-11-28 18:59:38 +0200
@@ -12,7 +12,12 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
+
 
 struct vf_priv_s {
 	int field;
diff -urN mplayer-svn-r21352.orig/libmpcodecs/vf_palette.c mplayer-svn-r21352/libmpcodecs/vf_palette.c
--- mplayer-svn-r21352.orig/libmpcodecs/vf_palette.c	2006-07-03 14:16:54 +0300
+++ mplayer-svn-r21352/libmpcodecs/vf_palette.c	2006-11-28 18:59:38 +0200
@@ -11,7 +11,11 @@
 #include "mp_image.h"
 #include "vf.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 
 //===========================================================================//
 
diff -urN mplayer-svn-r21352.orig/libmpcodecs/vf_qp.c mplayer-svn-r21352/libmpcodecs/vf_qp.c
--- mplayer-svn-r21352.orig/libmpcodecs/vf_qp.c	2006-06-13 15:10:35 +0300
+++ mplayer-svn-r21352/libmpcodecs/vf_qp.c	2006-11-28 18:59:38 +0200
@@ -37,9 +37,15 @@
 // Needed to bring in lrintf.
 #define HAVE_AV_CONFIG_H
 
+#ifdef USE_LIBAVCODEC_SO
+#include <ffmpeg/avcodec.h>
+#include <ffmpeg/dsputil.h>
+#include <ffmpeg/common.h>
+#else
 #include "libavcodec/avcodec.h"
 #include "libavcodec/dsputil.h"
 #include "libavutil/common.h"
+#endif
 
 /* FIXME: common.h defines printf away when HAVE_AV_CONFIG
  * is defined, but mp_image.h needs printf.
diff -urN mplayer-svn-r21352.orig/libmpcodecs/vf_rgb2bgr.c mplayer-svn-r21352/libmpcodecs/vf_rgb2bgr.c
--- mplayer-svn-r21352.orig/libmpcodecs/vf_rgb2bgr.c	2006-07-03 14:16:54 +0300
+++ mplayer-svn-r21352/libmpcodecs/vf_rgb2bgr.c	2006-11-28 18:59:38 +0200
@@ -11,7 +11,12 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
+
 
 //===========================================================================//
 
diff -urN mplayer-svn-r21352.orig/libmpcodecs/vf_sab.c mplayer-svn-r21352/libmpcodecs/vf_sab.c
--- mplayer-svn-r21352.orig/libmpcodecs/vf_sab.c	2006-09-18 11:03:58 +0300
+++ mplayer-svn-r21352/libmpcodecs/vf_sab.c	2006-11-28 18:59:38 +0200
@@ -29,12 +29,20 @@
 #include <malloc.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 #include "img_format.h"
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 
diff -urN mplayer-svn-r21352.orig/libmpcodecs/vf_scale.c mplayer-svn-r21352/libmpcodecs/vf_scale.c
--- mplayer-svn-r21352.orig/libmpcodecs/vf_scale.c	2006-09-18 11:03:58 +0300
+++ mplayer-svn-r21352/libmpcodecs/vf_scale.c	2006-11-28 18:59:38 +0200
@@ -13,7 +13,11 @@
 #include "fmt-conversion.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 #include "m_option.h"
diff -urN mplayer-svn-r21352.orig/libmpcodecs/vf_screenshot.c mplayer-svn-r21352/libmpcodecs/vf_screenshot.c
--- mplayer-svn-r21352.orig/libmpcodecs/vf_screenshot.c	2006-11-17 12:30:00 +0200
+++ mplayer-svn-r21352/libmpcodecs/vf_screenshot.c	2006-11-28 18:59:38 +0200
@@ -22,7 +22,11 @@
 #include "vf_scale.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 
 struct vf_priv_s {
     int frameno;
diff -urN mplayer-svn-r21352.orig/libmpcodecs/vf_smartblur.c mplayer-svn-r21352/libmpcodecs/vf_smartblur.c
--- mplayer-svn-r21352.orig/libmpcodecs/vf_smartblur.c	2006-09-18 11:03:58 +0300
+++ mplayer-svn-r21352/libmpcodecs/vf_smartblur.c	2006-11-28 18:59:38 +0200
@@ -29,12 +29,21 @@
 #include <malloc.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
+
 #include "img_format.h"
 #include "mp_image.h"
 #include "vf.h"
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r21352.orig/libmpcodecs/vf_yuy2.c mplayer-svn-r21352/libmpcodecs/vf_yuy2.c
--- mplayer-svn-r21352.orig/libmpcodecs/vf_yuy2.c	2006-07-03 14:16:54 +0300
+++ mplayer-svn-r21352/libmpcodecs/vf_yuy2.c	2006-11-28 18:59:38 +0200
@@ -12,7 +12,11 @@
 #include "vf.h"
 
 #include "libvo/fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "vf_scale.h"
 
 //===========================================================================//
diff -urN mplayer-svn-r21352.orig/libswscale/swscale-example.c mplayer-svn-r21352/libswscale/swscale-example.c
--- mplayer-svn-r21352.orig/libswscale/swscale-example.c	2006-11-02 12:29:44 +0200
+++ mplayer-svn-r21352/libswscale/swscale-example.c	2006-11-28 18:59:38 +0200
@@ -25,7 +25,11 @@
 #include <stdarg.h>
 
 #undef HAVE_AV_CONFIG_H
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "rgb2rgb.h"
diff -urN mplayer-svn-r21352.orig/libswscale/swscale_internal.h mplayer-svn-r21352/libswscale/swscale_internal.h
--- mplayer-svn-r21352.orig/libswscale/swscale_internal.h	2006-11-20 12:43:26 +0200
+++ mplayer-svn-r21352/libswscale/swscale_internal.h	2006-11-28 18:59:38 +0200
@@ -25,7 +25,11 @@
 #include <altivec.h>
 #endif
 
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
 
 #ifdef CONFIG_DARWIN
 #define AVV(x...) (x)
diff -urN mplayer-svn-r21352.orig/libvo/mga_common.c mplayer-svn-r21352/libvo/mga_common.c
--- mplayer-svn-r21352.orig/libvo/mga_common.c	2006-07-03 14:16:55 +0300
+++ mplayer-svn-r21352/libvo/mga_common.c	2006-11-28 18:59:38 +0200
@@ -1,7 +1,11 @@
 
 #include "fastmemcpy.h"
 #include "cpudetect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #include "mp_msg.h"
 #include "help_mp.h"
diff -urN mplayer-svn-r21352.orig/libvo/sub.c mplayer-svn-r21352/libvo/sub.c
--- mplayer-svn-r21352.orig/libvo/sub.c	2006-11-27 14:05:38 +0200
+++ mplayer-svn-r21352/libvo/sub.c	2006-11-28 18:59:38 +0200
@@ -20,7 +20,12 @@
 #include "font_load.h"
 #include "sub.h"
 #include "spudec.h"
+
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 
 #define NEW_SPLITTING
 
diff -urN mplayer-svn-r21352.orig/libvo/vo_aa.c mplayer-svn-r21352/libvo/vo_aa.c
--- mplayer-svn-r21352.orig/libvo/vo_aa.c	2006-11-27 14:05:38 +0200
+++ mplayer-svn-r21352/libvo/vo_aa.c	2006-11-28 18:59:38 +0200
@@ -27,7 +27,11 @@
 #include "video_out.h"
 #include "video_out_internal.h"
 #include "aspect.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #include "font_load.h"
 #include "sub.h"
diff -urN mplayer-svn-r21352.orig/libvo/vo_vesa.c mplayer-svn-r21352/libvo/vo_vesa.c
--- mplayer-svn-r21352.orig/libvo/vo_vesa.c	2006-08-28 14:40:31 +0300
+++ mplayer-svn-r21352/libvo/vo_vesa.c	2006-11-28 18:59:38 +0200
@@ -47,7 +47,11 @@
 #endif
 #include "mp_msg.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 
 
diff -urN mplayer-svn-r21352.orig/libvo/vo_x11.c mplayer-svn-r21352/libvo/vo_x11.c
--- mplayer-svn-r21352.orig/libvo/vo_x11.c	2006-10-25 19:49:02 +0300
+++ mplayer-svn-r21352/libvo/vo_x11.c	2006-11-28 18:59:38 +0200
@@ -38,7 +38,11 @@
 #include "fastmemcpy.h"
 #include "sub.h"
 
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
 #define MODE_RGB  0x1
 #define MODE_BGR  0x2
diff -urN mplayer-svn-r21352.orig/libvo/vo_yuv4mpeg.c mplayer-svn-r21352/libvo/vo_yuv4mpeg.c
--- mplayer-svn-r21352.orig/libvo/vo_yuv4mpeg.c	2006-10-26 12:45:00 +0300
+++ mplayer-svn-r21352/libvo/vo_yuv4mpeg.c	2006-11-28 18:59:38 +0200
@@ -39,10 +39,19 @@
 #include "sub.h"
 
 #include "fastmemcpy.h"
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#include <ffmpeg/rgb2rgb.h>
+#else
 #include "libswscale/swscale.h"
 #include "libswscale/rgb2rgb.h"
+#endif
 #include "libmpcodecs/vf_scale.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/rational.h>
+#else
 #include "libavutil/rational.h"
+#endif
 
 static vo_info_t info = 
 {
diff -urN mplayer-svn-r21352.orig/Makefile mplayer-svn-r21352/Makefile
--- mplayer-svn-r21352.orig/Makefile	2006-11-28 12:50:10 +0200
+++ mplayer-svn-r21352/Makefile	2006-11-28 18:59:38 +0200
@@ -55,7 +55,7 @@
               libaf/libaf.a \
               libmpdemux/libmpdemux.a \
               stream/stream.a \
-              libswscale/libswscale.a \
+              $(SWSCALE_LIB) \
               libvo/libosd.a \
 
 LIBS_MPLAYER = libvo/libvo.a \
@@ -93,6 +93,8 @@
 PARTS-$(CONFIG_LIBAVUTIL)         += libavutil
 COMMON_LIBS-$(CONFIG_LIBPOSTPROC) += libpostproc/libpostproc.a
 PARTS-$(CONFIG_LIBPOSTPROC)       += libpostproc
+COMMON_LIBS-$(CONFIG_LIBSWSCALE)  += libswscale/libswscale.a
+PARTS-$(CONFIG_LIBSWSCALE)        += libswscale
 COMMON_LIBS-$(WIN32DLL)           += loader/libloader.a
 PARTS-$(WIN32DLL)                 += loader
 COMMON_LIBS-$(MP3LIB)             += mp3lib/libMP3.a
diff -urN mplayer-svn-r21352.orig/spudec.c mplayer-svn-r21352/spudec.c
--- mplayer-svn-r21352.orig/spudec.c	2006-09-18 11:04:02 +0300
+++ mplayer-svn-r21352/spudec.c	2006-11-28 18:59:38 +0200
@@ -23,8 +23,17 @@
 #include <math.h>
 #include "libvo/video_out.h"
 #include "spudec.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avutil.h>
+#else
 #include "avutil.h"
+#endif
+#ifdef USE_LIBSWSCALE_SO
+#include <ffmpeg/swscale.h>
+#else
 #include "libswscale/swscale.h"
+#endif
+
 
 #define MIN(a, b)	((a)<(b)?(a):(b))
 
diff -urN mplayer-svn-r21352.orig/stream/stream_dvdnav.c mplayer-svn-r21352/stream/stream_dvdnav.c
--- mplayer-svn-r21352.orig/stream/stream_dvdnav.c	2006-11-28 12:50:09 +0200
+++ mplayer-svn-r21352/stream/stream_dvdnav.c	2006-11-28 18:59:38 +0200
@@ -11,7 +11,11 @@
 #include "libmpdemux/demuxer.h"
 #include "stream_dvdnav.h"
 #include "libvo/video_out.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/common.h>
+#else
 #include "libavutil/common.h"
+#endif
 #include "spudec.h"
 #include "m_option.h"
 #include "m_struct.h"
diff -urN mplayer-svn-r21352.orig/TOOLS/Makefile mplayer-svn-r21352/TOOLS/Makefile
--- mplayer-svn-r21352.orig/TOOLS/Makefile	2006-11-27 14:05:41 +0200
+++ mplayer-svn-r21352/TOOLS/Makefile	2006-11-28 18:59:38 +0200
@@ -27,8 +27,7 @@
 
 subrip: subrip.c
 	$(CC) $(CFLAGS) -g -o $@ $< ../vobsub.o ../spudec.o ../mp_msg.o \
-          ../unrarlib.o ../libswscale/swscale.o ../libswscale/rgb2rgb.o \
-          ../libswscale/yuv2rgb.o ../libmpcodecs/img_format.o -lm
+          ../unrarlib.o ../libmpcodecs/img_format.o -L../libswscale -lswscale -lm
 
 vivodump: vivodump.c
 	$(CC) $(CFLAGS) -o $@ $< ../mp_msg.o ../libmpdemux/libmpdemux.a ../stream/stream.a
