diff -urN MPlayer-svn-20060711.orig/configure MPlayer-svn-20060711/configure
--- MPlayer-svn-20060711.orig/configure	2006-07-10 14:48:37 +0300
+++ MPlayer-svn-20060711/configure	2006-07-12 16:02:20 +0300
@@ -4488,7 +4488,7 @@
 echocheck "VESA support"
 if test "$_vesa" = auto ; then
   cat > $TMPC << EOF
-#include <vbe.h>
+#include <vbe/vbe.h>
 int main(void) { vbeVersion(); return 0; }
 EOF
   _vesa=no
diff -urN MPlayer-svn-20060711.orig/libvo/gtf.h MPlayer-svn-20060711/libvo/gtf.h
--- MPlayer-svn-20060711.orig/libvo/gtf.h	2006-06-13 15:11:09 +0300
+++ MPlayer-svn-20060711/libvo/gtf.h	2006-07-12 16:02:37 +0300
@@ -1,7 +1,7 @@
 #ifndef __GTF_H
 #define __GTF_H
 
-#include <vbe.h>
+#include <vbe/vbe.h>
 
 #define GTF_VF 0 
 #define GTF_HF 1 
diff -urN MPlayer-svn-20060711.orig/libvo/vo_vesa.c MPlayer-svn-20060711/libvo/vo_vesa.c
--- MPlayer-svn-20060711.orig/libvo/vo_vesa.c	2006-07-03 14:16:55 +0300
+++ MPlayer-svn-20060711/libvo/vo_vesa.c	2006-07-12 16:02:30 +0300
@@ -32,7 +32,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
-#include <vbe.h>
+#include <vbe/vbe.h>
 
 #include "video_out.h"
 #include "video_out_internal.h"
