diff -Nurp MPlayer-1.0pre7.orig/configure MPlayer-1.0pre7/configure
--- MPlayer-1.0pre7.orig/configure	2005-04-13 14:46:35 +0300
+++ MPlayer-1.0pre7/configure	2005-04-29 22:17:37 +0300
@@ -3789,6 +3789,8 @@ if test "$_aa" = yes ; then
   _def_aa='#define HAVE_AA 1'
   if cygwin ; then
     _ld_aa=`aalib-config --libs | cut -d " " -f 2,5,6`
+  else
+    _ld_aa='-laa'
   fi
   _vosrc="$_vosrc vo_aa.c"
   _vomodules="aa $_vomodules"
