diff -urN mplayer-svn-r20777.orig/TOOLS/bmovl-test/Makefile mplayer-svn-r20777/TOOLS/bmovl-test/Makefile
--- mplayer-svn-r20777.orig/TOOLS/bmovl-test/Makefile	2006-11-08 14:49:41 +0200
+++ mplayer-svn-r20777/TOOLS/bmovl-test/Makefile	2006-11-09 01:45:29 +0200
@@ -1,7 +1,7 @@
 include ../../config.mak
 
 bmovl-test$(EXESUF): bmovl-test.c
-	$(CC) -O3 $(SDL_INC) -lSDL_image -o $@ bmovl-test.c
+	$(CC) -O3 $(EXTRA_INC) -o $@ bmovl-test.c -lSDL_image
 
 clean:
 	rm -f bmovl-test
