diff -urN mplayer-svn-r19558.orig/libvo/osd_template.c mplayer-svn-r19558/libvo/osd_template.c
--- mplayer-svn-r19558.orig/libvo/osd_template.c	2006-06-13 15:11:09 +0300
+++ mplayer-svn-r19558/libvo/osd_template.c	2006-08-29 12:13:29 +0300
@@ -204,7 +204,7 @@
     for(y=0;y<h;y++){
         register unsigned char *dst = dstbase;
         register int x;
-#if defined(ARCH_X86) || defined(ARCH_X86_64)
+#ifdef ARCH_X86
 #ifdef HAVE_MMX
     asm volatile(
 	PREFETCHW" %0\n\t"
@@ -324,7 +324,7 @@
 #endif /* HAVE_MMX */
     for(y=0;y<h;y++){
         register int x;
-#if defined(ARCH_X86) || defined(ARCH_X86_64)
+#ifdef ARCH_X86
 #ifdef HAVE_MMX
 #ifdef HAVE_3DNOW
     asm volatile(
