diff -urN mplayer-svn-r22518.orig/DOCS/xml/configure mplayer-svn-r22518/DOCS/xml/configure
--- mplayer-svn-r22518.orig/DOCS/xml/configure	2007-03-12 02:50:30 +0200
+++ mplayer-svn-r22518/DOCS/xml/configure	2007-03-12 03:42:50 +0200
@@ -8,6 +8,7 @@
 
 echo "Searching for XML catalogs..."
 for _try_catalog in \
+  `pwd`/catalog \
   /etc/sgml/catalog \
   /usr/share/xml/docbook/*/catalog.xml \
   /opt/local/share/xml/docbook-xml/*/catalog.xml \
@@ -40,6 +41,7 @@
 echo "Searching for stylesheets..."
 echo "Searching for html/chunk.xsl..."
 for _try_chunk_xsl in \
+  /usr/share/xml/docbook/xsl-stylesheets/html/chunk.xsl \
   /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/html/chunk.xsl \
   /usr/share/sgml/docbook/yelp/docbook/html/chunk.xsl \
   /usr/local/share/sgml/docbook/stylesheet/xsl/nwalsh/html/chunk.xsl \
@@ -68,6 +70,7 @@
 
 echo "Searching for html/docbook.xsl..."
 for _try_docbook_xsl in \
+  /usr/share/xml/docbook/xsl-stylesheets/html/docbook.xsl \
   /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/html/docbook.xsl \
   /usr/share/sgml/docbook/yelp/docbook/html/docbook.xsl \
   /usr/local/share/sgml/docbook/stylesheet/xsl/nwalsh/html/docbook.xsl \
@@ -126,6 +129,7 @@
 echo "Searching for DTD..."
 #FIXME: This should prefer higher version numbers, not the other way around ..
 for _try_dtd in \
+  /usr/share/xml/docbook/dtd/*/docbookx.dtd \
   /usr/share/xml/docbook/*/dtd/4*/docbookx.dtd \
   /usr/share/xml/docbook/*/docbookx.dtd \
   /usr/share/sgml/docbook/*/docbookx.dtd \
@@ -281,6 +285,7 @@
     echo "I will try to use OpenJade or Jade (using '$_jade')."
     echo "They aren't (currently) fully supported, however."
     for _try_docbook_dsl in \
+      /usr/share/sgml/docbook/dsssl-stylesheets/html/docbook.dsl \
       /usr/share/sgml/docbook/stylesheet/dsssl/modular/html/docbook.dsl \
       /usr/local/share/sgml/docbook/stylesheet/dsssl/modular/html/docbook.dsl
     do
@@ -291,6 +296,7 @@
       fi
     done
     for _try_xml_dcl in \
+      /usr/share/sgml/xml.dcl \
       /usr/share/sgml/declaration/xml.dcl \
       /usr/local/share/sgml/declaration/xml.dcl
     do
