diff -urN mplayer-svn-r19416.orig/configure mplayer-svn-r19416/configure
--- mplayer-svn-r19416.orig/configure	2006-08-17 13:36:39 +0300
+++ mplayer-svn-r19416/configure	2006-08-17 18:45:51 +0300
@@ -4128,6 +4128,8 @@
   _def_aa='#define HAVE_AA 1'
   if cygwin ; then
     _ld_aa=`aalib-config --libs | cut -d " " -f 2,5,6`
+  else
+    _ld_aa=`aalib-config --libs | cut -d " " -f 1,2,4-6`
   fi
   _vosrc="$_vosrc vo_aa.c"
   _vomodules="aa $_vomodules"
@@ -4152,7 +4154,7 @@
 if test "$_caca" = yes ; then
   _def_caca='#define HAVE_CACA 1'
   _inc_extra="$_inc_extra `caca-config --cflags`"
-  _ld_caca=`caca-config --libs`
+  _ld_caca=`caca-config --libs | cut -d " " -f 1-4`
   _vosrc="$_vosrc vo_caca.c"
   _vomodules="caca $_vomodules"
 else
@@ -4842,7 +4844,7 @@
 
 
 echocheck "JACK"
-if test "$_jack" = auto ; then
+if test "$_jack" != no ; then
   _jack=yes
 
 cat > $TMPC << EOF
@@ -4869,7 +4871,7 @@
 echores "$_jack"
 
 echocheck "OpenAL"
-if test "$_openal" = auto ; then
+if test "$_openal" != no ; then
   _openal=no
 cat > $TMPC << EOF
 #include <AL/al.h>
@@ -5313,7 +5315,7 @@
     _res_comment="freetype support needed"
 fi
 echocheck "fontconfig"
-if test "$_fontconfig" = auto ; then
+if test "$_fontconfig" != no ; then
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <fontconfig/fontconfig.h>
