--- MPlayer-cvs-20060220.orig/configure	2006-02-20 23:02:44 +0200
+++ MPlayer-cvs-20060220/configure	2006-02-21 02:08:44 +0200
@@ -5271,7 +5271,7 @@
     _res_comment="freetype support needed"
 fi
 echocheck "fontconfig"
-if test "$_fontconfig" = auto ; then
+if test "$_fontconfig" = auto -o "$_fontconfig" = yes; then
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <fontconfig/fontconfig.h>
