diff -urN mplayer-svn-r26454.orig/configure mplayer-svn-r26454/configure
--- mplayer-svn-r26454.orig/configure	2008-04-14 00:02:39 +0300
+++ mplayer-svn-r26454/configure	2008-04-16 00:23:38 +0300
@@ -292,10 +292,12 @@
   --disable-libavcodec_a    disable static libavcodec [autodetect]
   --disable-libavformat_a   disable static libavformat [autodetect]
   --disable-libpostproc_a   disable static libpostproc [autodetect]
+  --disable-libswscale_a    disable static libswscale [autodetect]
   --disable-libavutil_so    disable shared libavutil [autodetect]
   --disable-libavcodec_so   disable shared libavcodec [autodetect]
   --disable-libavformat_so  disable shared libavformat [autodetect]
   --disable-libpostproc_so  disable shared libpostproc [autodetect]
+  --disable-libswscale_so   disable shared libswscale [autodetect]
   --disable-libavcodec_mpegaudio_hp disable high precision audio decoding
                                     in libavcodec [enabled]
   --disable-tremor-internal disable internal Tremor [enabled]
@@ -511,6 +513,8 @@
 _libavformat_so=auto
 _libpostproc_a=auto
 _libpostproc_so=auto
+_libswscale_a=auto
+_libswscale_so=auto
 _libavcodec_mpegaudio_hp=yes
 _mencoder=yes
 _mplayer=yes
@@ -1045,6 +1049,10 @@
   --disable-libpostproc_a)	_libpostproc_a=no	;;
   --enable-libpostproc_so)	_libpostproc_so=yes	;;
   --disable-libpostproc_so)	_libpostproc_so=no	;;
+  --enable-libswscale_a)	_libswscale_a=yes	;;
+  --disable-libswscale_a)	_libswscale_a=no	;;
+  --enable-libswscale_so)	_libswscale_so=yes	;;
+  --disable-libswscale_so)	_libswscale_so=no	;;
   --enable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=yes	;;
   --disable-libavcodec_mpegaudio_hp)	_libavcodec_mpegaudio_hp=no	;;
 
@@ -6290,6 +6298,45 @@
 fi
 echores "$_real"
 
+echocheck "FFmpeg libswscale"
+if test "$_libswscale_a" = auto ; then
+  if test -d libswascale ; then
+    _libswscale_a=yes
+    _res_comment="static"
+  else
+    die "MPlayer will not compile without libswscale in the source tree."
+  fi
+elif test "$_libswscale_so" != no ; then
+  _libswscale_so=no
+  cat > $TMPC << EOF
+#include <ffmpeg/swscale.h>
+int main(void) { sws_getIdentityVec(); return 0; }
+EOF
+  if $_pkg_config --exists libswscale ; then
+    _inc_libswscale=`$_pkg_config --cflags libswscale`
+    _ld_tmp=`$_pkg_config --libs libswscale`
+    cc_check $_inc_libswscale $_ld_tmp && _ld_extra="$_ld_extra $_ld_tmp" \
+      && _libswscale_so=yes
+  elif cc_check -lswscale $_ld_lm ; then
+    _ld_extra="$_ld_extra -lswscale"
+    _libswscale_so=yes
+    _res_comment="using libswscale.so, but static libswscale is recommended"
+  fi
+fi
+_libswscale=no
+_def_libswscale='#undef USE_LIBSWSCALE'
+_def_libswscale_a='#undef USE_LIBSWSCALE_A'
+_def_libswscale_so='#undef USE_LIBSWSCALE_SO'
+test "$_libswscale_a" = yes || test "$_libswscale_so" = yes && _libswscale=yes
+test "$_libswscale"    = yes && _def_libswscale='#define USE_LIBSWSCALE 1'
+test "$_libswscale_a"  = yes && _def_libswscale_a='#define USE_LIBSWSCALE_A 1'
+test "$_libswscale_so" = yes && _def_libswscale_so='#define USE_LIBSWSCALE_SO 1'
+# neither static nor shared libswscale is available, but it is mandatory ...
+if test "$_libswscale" = no ; then
+  die "You need static or shared libswscale, MPlayer will not compile without!"
+fi
+echores "$_libswscale"
+
 
 echocheck "QuickTime codecs"
 _qtx_emulation=no
@@ -7853,6 +7900,10 @@
 LIBPOSTPROC = $_libpostproc
 LIBPOSTPROC_A = $_libpostproc_a
 LIBPOSTPROC_SO = $_libpostproc_so
+LIBSWSCALE = $_libswscale
+LIBSWSCALE_A = $_libswscale_a
+LIBSWSCALE_SO = $_libswscale_so
+CONFIG_SWSCALE = $_libswscale
 
 BUILD_STATIC=yes
 SRC_PATH=..
@@ -8190,6 +8241,10 @@
 $_def_libpostproc_a
 $_def_libpostproc_so
 
+$_def_libswscale
+$_def_libswscale_a
+$_def_libswscale_so
+
 /* Win32 DLL support */
 $_def_win32dll
 #define WIN32_PATH "$_win32codecsdir"
diff -urN mplayer-svn-r26454.orig/dvdread/bswap.h mplayer-svn-r26454/dvdread/bswap.h
--- mplayer-svn-r26454.orig/dvdread/bswap.h	2007-08-19 23:12:48 +0300
+++ mplayer-svn-r26454/dvdread/bswap.h	2008-04-16 00:27:44 +0300
@@ -1,7 +1,7 @@
 #ifndef DVDREAD_BSWAP_H
 #define DVDREAD_BSWAP_H
 
-#include "../libavutil/bswap.h"
+#include "libavutil/bswap.h"
 
 #ifdef WORDS_BIGENDIAN
 #define B2N_16(x)
diff -urN mplayer-svn-r26454.orig/input/input.c mplayer-svn-r26454/input/input.c
--- mplayer-svn-r26454.orig/input/input.c	2008-03-17 00:11:15 +0200
+++ mplayer-svn-r26454/input/input.c	2008-04-16 00:25:02 +0300
@@ -20,7 +20,7 @@
 #include "osdep/getch2.h"
 #include "osdep/keycodes.h"
 #include "osdep/timer.h"
-#include "avstring.h"
+#include "libavutil/avstring.h"
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "m_config.h"
diff -urN mplayer-svn-r26454.orig/libmpcodecs/ad_qtaudio.c mplayer-svn-r26454/libmpcodecs/ad_qtaudio.c
--- mplayer-svn-r26454.orig/libmpcodecs/ad_qtaudio.c	2008-03-17 00:11:15 +0200
+++ mplayer-svn-r26454/libmpcodecs/ad_qtaudio.c	2008-04-16 00:35:17 +0300
@@ -14,7 +14,8 @@
 #include "loader/wine/windef.h"
 
 #include "ad_internal.h"
-#include "bswap.h"
+#include "libavutil/common.h"
+#include "mpbswap.h"
 
 #ifdef WIN32_LOADER
 #include "loader/ldt_keeper.h"
diff -urN mplayer-svn-r26454.orig/libmpcodecs/vd_ffmpeg.c mplayer-svn-r26454/libmpcodecs/vd_ffmpeg.c
--- mplayer-svn-r26454.orig/libmpcodecs/vd_ffmpeg.c	2008-04-14 00:02:35 +0300
+++ mplayer-svn-r26454/libmpcodecs/vd_ffmpeg.c	2008-04-16 00:31:40 +0300
@@ -8,6 +8,7 @@
 #include "help_mp.h"
 
 #include "libavutil/common.h"
+#include "libavutil/avutil.h"
 #include "libavutil/intreadwrite.h"
 #include "mpbswap.h"
 
diff -urN mplayer-svn-r26454.orig/libmpcodecs/vd_qtvideo.c mplayer-svn-r26454/libmpcodecs/vd_qtvideo.c
--- mplayer-svn-r26454.orig/libmpcodecs/vd_qtvideo.c	2008-03-17 00:11:15 +0200
+++ mplayer-svn-r26454/libmpcodecs/vd_qtvideo.c	2008-04-16 00:32:52 +0300
@@ -28,7 +28,7 @@
 
 LIBVD_EXTERN(qtvideo)
 
-#include "bswap.h"
+#include "mpbswap.h"
 
 #ifndef HAVE_QUICKTIME
 #include "loader/qtx/qtxsdk/components.h"
diff -urN mplayer-svn-r26454.orig/libmpcodecs/vd_xanim.c mplayer-svn-r26454/libmpcodecs/vd_xanim.c
--- mplayer-svn-r26454.orig/libmpcodecs/vd_xanim.c	2007-05-28 01:08:51 +0300
+++ mplayer-svn-r26454/libmpcodecs/vd_xanim.c	2008-04-16 00:33:55 +0300
@@ -37,7 +37,7 @@
 #include <errno.h> /* strerror, errno */
 
 #include "mp_msg.h"
-#include "bswap.h"
+#include "mpbswap.h"
 
 #include "osdep/timer.h"
 
diff -urN mplayer-svn-r26454.orig/libmpcodecs/vf_scale.c mplayer-svn-r26454/libmpcodecs/vf_scale.c
--- mplayer-svn-r26454.orig/libmpcodecs/vf_scale.c	2007-12-02 17:50:48 +0200
+++ mplayer-svn-r26454/libmpcodecs/vf_scale.c	2008-04-16 00:34:24 +0300
@@ -11,7 +11,7 @@
 #include "mp_image.h"
 #include "vf.h"
 #include "fmt-conversion.h"
-#include "bswap.h"
+#include "mpbswap.h"
 
 #include "libswscale/swscale.h"
 #include "vf_scale.h"
diff -urN mplayer-svn-r26454.orig/libmpdemux/demux_lavf.c mplayer-svn-r26454/libmpdemux/demux_lavf.c
--- mplayer-svn-r26454.orig/libmpdemux/demux_lavf.c	2008-04-14 00:02:38 +0300
+++ mplayer-svn-r26454/libmpdemux/demux_lavf.c	2008-04-16 00:26:00 +0300
@@ -33,7 +33,6 @@
 
 #include "libavformat/avformat.h"
 #include "libavutil/avutil.h"
-#include "libavformat/avi.h"
 #include "libavcodec/opt.h"
 #include "libavformat/riff.h"
 
diff -urN mplayer-svn-r26454.orig/libmpdemux/demux_rtp_codec.cpp mplayer-svn-r26454/libmpdemux/demux_rtp_codec.cpp
--- mplayer-svn-r26454.orig/libmpdemux/demux_rtp_codec.cpp	2008-02-15 02:21:55 +0200
+++ mplayer-svn-r26454/libmpdemux/demux_rtp_codec.cpp	2008-04-16 00:26:52 +0300
@@ -6,7 +6,7 @@
 #include <limits.h>
 #include <math.h>
 #include "stheader.h"
-#include "base64.h"
+#include "libavutil/base64.h"
 }
 
 #ifdef USE_LIBAVCODEC
diff -urN mplayer-svn-r26454.orig/libswscale/rgb2rgb.c mplayer-svn-r26454/libswscale/rgb2rgb.c
--- mplayer-svn-r26454.orig/libswscale/rgb2rgb.c	2008-02-06 00:33:19 +0200
+++ mplayer-svn-r26454/libswscale/rgb2rgb.c	2008-04-16 00:35:07 +0300
@@ -31,7 +31,7 @@
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "x86_cpu.h"
-#include "bswap.h"
+#include "libavutil/bswap.h"
 
 #define FAST_BGR2YV12 // use 7 bit coeffs instead of 15bit
 
diff -urN mplayer-svn-r26454.orig/libswscale/swscale.c mplayer-svn-r26454/libswscale/swscale.c
--- mplayer-svn-r26454.orig/libswscale/swscale.c	2008-04-01 01:43:03 +0300
+++ mplayer-svn-r26454/libswscale/swscale.c	2008-04-16 00:29:38 +0300
@@ -70,7 +70,7 @@
 #include "swscale.h"
 #include "swscale_internal.h"
 #include "x86_cpu.h"
-#include "bswap.h"
+#include "libavutil/bswap.h"
 #include "rgb2rgb.h"
 #include "libavcodec/opt.h"
 
diff -urN mplayer-svn-r26454.orig/libswscale/swscale-example.c mplayer-svn-r26454/libswscale/swscale-example.c
--- mplayer-svn-r26454.orig/libswscale/swscale-example.c	2008-04-14 00:02:37 +0300
+++ mplayer-svn-r26454/libswscale/swscale-example.c	2008-04-16 00:28:49 +0300
@@ -25,8 +25,8 @@
 #include <stdarg.h>
 
 #undef HAVE_AV_CONFIG_H
-#include "avutil.h"
-#include "swscale.h"
+#include "libavutil/avutil.h"
+#include "libswscale/swscale.h"
 #include "swscale_internal.h"
 #include "rgb2rgb.h"
 
diff -urN mplayer-svn-r26454.orig/libswscale/swscale_internal.h mplayer-svn-r26454/libswscale/swscale_internal.h
--- mplayer-svn-r26454.orig/libswscale/swscale_internal.h	2008-03-11 03:52:00 +0200
+++ mplayer-svn-r26454/libswscale/swscale_internal.h	2008-04-16 00:29:55 +0300
@@ -27,7 +27,7 @@
 #include <altivec.h>
 #endif
 
-#include "avutil.h"
+#include "libavutil/avutil.h"
 
 #define MAX_FILTER_SIZE 256
 
diff -urN mplayer-svn-r26454.orig/Makefile mplayer-svn-r26454/Makefile
--- mplayer-svn-r26454.orig/Makefile	2008-04-14 22:57:07 +0300
+++ mplayer-svn-r26454/Makefile	2008-04-16 00:23:32 +0300
@@ -160,13 +160,14 @@
 COMMON_LIBS = libmpcodecs/libmpcodecs.a \
               libmpdemux/libmpdemux.a \
               stream/stream.a \
-              libswscale/libswscale.a \
               libvo/libosd.a \
 
 COMMON_LIBS-$(LIBAVFORMAT_A)      += libavformat/libavformat.a
 COMMON_LIBS-$(LIBAVCODEC_A)       += libavcodec/libavcodec.a
 COMMON_LIBS-$(LIBAVUTIL_A)        += libavutil/libavutil.a
 COMMON_LIBS-$(LIBPOSTPROC_A)      += libpostproc/libpostproc.a
+COMMON_LIBS-$(LIBSWSCALE_A)       += libswscale/libswscale.a
+PARTS-$(CONFIG_LIBSWSCALE)        += libswscale
 COMMON_LIBS-$(WIN32DLL)           += loader/libloader.a
 COMMON_LIBS-$(MP3LIB)             += mp3lib/libmp3.a
 COMMON_LIBS-$(LIBA52)             += liba52/liba52.a
diff -urN mplayer-svn-r26454.orig/TOOLS/Makefile mplayer-svn-r26454/TOOLS/Makefile
--- mplayer-svn-r26454.orig/TOOLS/Makefile	2008-04-14 00:02:38 +0300
+++ mplayer-svn-r26454/TOOLS/Makefile	2008-04-16 00:23:59 +0300
@@ -14,14 +14,14 @@
 #	vivodump$(EXESUF) \
 
 COMMON_LIBS = ../mp_msg.o ../mp_fifo.o ../osdep/$(TIMER) ../osdep/$(GETCH) \
-              -ltermcap -lm
+              -ltinfo -lm
 
 all: $(OBJS)
 
 bmovl-test$(EXESUF): bmovl-test.c -lSDL_image
 
-subrip$(EXESUF): subrip.c ../vobsub.o ../spudec.o ../unrar_exec.o \
-  ../libswscale/libswscale.a ../libavutil/libavutil.a $(COMMON_LIBS)
+subrip$(EXESUF): subrip.c ../vobsub.o ../spudec.o ../unrarlib.o \
+  $(COMMON_LIBS) -L../libswscale -lswscale -L../libavutil -lavutil
 
 vfw2menc$(EXESUF): vfw2menc.c -lwinmm -lole32
 
