diff -urN mplayer-svn-r21402.orig/Makefile mplayer-svn-r21402/Makefile
--- mplayer-svn-r21402.orig/Makefile	2006-11-28 12:50:10 +0200
+++ mplayer-svn-r21402/Makefile	2006-12-01 00:21:50 +0200
@@ -255,11 +255,13 @@
 	-install -m 755 $(INSTALLSTRIP) -p vidix/drivers/*.so $(LIBDIR)/mplayer/vidix
 endif
 	$(INSTALL) -d $(BINDIR)
+ifeq ($(MPLAYER),yes)
 	$(INSTALL) -m 755 $(INSTALLSTRIP) mplayer$(EXESUF) \
 		$(BINDIR)/mplayer$(EXESUF)
 ifeq ($(GUI),yes)
 	-ln -sf mplayer$(EXESUF) $(BINDIR)/gmplayer$(EXESUF)
 endif
+endif
 	$(INSTALL) -d $(MANDIR)/man1
 	for i in $(MAN_LANG); do \
 		if test "$$i" = en ; then \
