diff -urN mplayer-svn-r20022.orig/configure mplayer-svn-r20022/configure
--- mplayer-svn-r20022.orig/configure	2006-10-02 13:09:55 +0300
+++ mplayer-svn-r20022/configure	2006-10-04 20:21:17 +0300
@@ -3247,7 +3247,7 @@
 
 
 echocheck "termios"
-if test "$_termios" = auto ; then
+if test "$_termios" != no ; then
   cat > $TMPC <<EOF
 #include <sys/termios.h>
 int main(void) { return 0; }
@@ -4156,6 +4156,8 @@
   _def_aa='#define HAVE_AA 1'
   if cygwin ; then
     _ld_aa=`aalib-config --libs | cut -d " " -f 2,5,6`
+  else
+    _ld_aa=`aalib-config --libs | cut -d " " -f 1,2,4-6`
   fi
   _vosrc="$_vosrc vo_aa.c"
   _vomodules="aa $_vomodules"
@@ -4183,7 +4185,7 @@
 if test "$_caca" = yes ; then
   _def_caca='#define HAVE_CACA 1'
   _inc_extra="$_inc_extra `caca-config --cflags`"
-  _ld_caca=`caca-config --libs`
+  _ld_caca=`caca-config --libs | cut -d " " -f 1-4`
   _vosrc="$_vosrc vo_caca.c"
   _vomodules="caca $_vomodules"
 else
@@ -4873,7 +4875,7 @@
 
 
 echocheck "JACK"
-if test "$_jack" = auto ; then
+if test "$_jack" != no ; then
   _jack=yes
 
 cat > $TMPC << EOF
@@ -4900,7 +4902,7 @@
 echores "$_jack"
 
 echocheck "OpenAL"
-if test "$_openal" = auto ; then
+if test "$_openal" != no ; then
   _openal=no
 cat > $TMPC << EOF
 #include <AL/al.h>
@@ -5361,7 +5363,7 @@
     _res_comment="freetype support needed"
 fi
 echocheck "fontconfig"
-if test "$_fontconfig" = auto ; then
+if test "$_fontconfig" != no ; then
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <fontconfig/fontconfig.h>
@@ -5906,7 +5908,7 @@
 
 if x86 && not qnx; then
 
-if test "$_win32" = auto ; then
+if test "$_win32" != no ; then
     if test -z "$_win32libdir" ; then
       for I in "$_libdir/codecs" "$_libdir/win32" /usr/local/lib/codecs /usr/lib/codecs /usr/local/lib/win32 /usr/lib/win32 ; do
         if test -d "$I" ; then
@@ -7286,6 +7288,9 @@
 if test "$_vidix_internal" = yes || test "$_vidix_external" = yes ; then
   _vidix=yes
   _def_vidix='#define CONFIG_VIDIX 1'
+  if test "$_vidix_external" = yes ; then
+    _def_vidix_ext='#define VIDIX_EXT 1'
+  fi
 else
   _vidix=no
   _def_vidix='#undef CONFIG_VIDIX'
@@ -7295,7 +7300,7 @@
   _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/mplayer/vidix/\" "
 elif test "$_vidix_external" = yes ; then
   _ld_vidix_external="-lvidix"
-  _def_vidix_pfx='#define MP_VIDIX_PFX "" '
+  _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/vidix/\" "
 fi
 
 if test "$_vidix" = yes; then
@@ -8221,6 +8226,7 @@
 /* enables / disables VIDIX usage */
 $_def_vidix
 $_def_vidix_pfx
+$_def_vidix_ext
 
 /* enables / disables new input joystick support */
 $_def_joystick
