diff -urN mplayer-svn-r19447.orig/configure mplayer-svn-r19447/configure
--- mplayer-svn-r19447.orig/configure	2006-08-19 13:05:32 +0300
+++ mplayer-svn-r19447/configure	2006-08-19 17:44:37 +0300
@@ -3814,7 +3814,7 @@
 
 
 echocheck "Xv"
-if test "$_x11" = yes && test "$_xv" = auto ; then
+if test "$_x11" = yes && test "$_xv" != no ; then
   cat > $TMPC <<EOF
 #include <X11/Xlib.h>
 #include <X11/extensions/Xvlib.h>
@@ -3870,7 +3870,7 @@
 
 
 echocheck "Xinerama"
-if test "$_x11" = yes && test "$_xinerama" = auto ; then
+if test "$_x11" = yes && test "$_xinerama" != no ; then
   cat > $TMPC <<EOF
 #include <X11/Xlib.h>
 #include <X11/extensions/Xinerama.h>
@@ -4130,6 +4130,8 @@
   _def_aa='#define HAVE_AA 1'
   if cygwin ; then
     _ld_aa=`aalib-config --libs | cut -d " " -f 2,5,6`
+  else
+    _ld_aa=`aalib-config --libs | cut -d " " -f 1,2,4-6`
   fi
   _vosrc="$_vosrc vo_aa.c"
   _vomodules="aa $_vomodules"
@@ -4154,7 +4156,7 @@
 if test "$_caca" = yes ; then
   _def_caca='#define HAVE_CACA 1'
   _inc_extra="$_inc_extra `caca-config --cflags`"
-  _ld_caca=`caca-config --libs`
+  _ld_caca=`caca-config --libs | cut -d " " -f 1-4`
   _vosrc="$_vosrc vo_caca.c"
   _vomodules="caca $_vomodules"
 else
@@ -4844,7 +4846,7 @@
 
 
 echocheck "JACK"
-if test "$_jack" = auto ; then
+if test "$_jack" != no ; then
   _jack=yes
 
 cat > $TMPC << EOF
@@ -4871,7 +4873,7 @@
 echores "$_jack"
 
 echocheck "OpenAL"
-if test "$_openal" = auto ; then
+if test "$_openal" != no ; then
   _openal=no
 cat > $TMPC << EOF
 #include <AL/al.h>
@@ -5315,7 +5317,7 @@
     _res_comment="freetype support needed"
 fi
 echocheck "fontconfig"
-if test "$_fontconfig" = auto ; then
+if test "$_fontconfig" != no ; then
         cat > $TMPC << EOF
 #include <stdio.h>
 #include <fontconfig/fontconfig.h>
@@ -5868,7 +5870,7 @@
 
 if x86 && not qnx; then
 
-if test "$_win32" = auto ; then
+if test "$_win32" != no ; then
     if test -z "$_win32libdir" ; then
       for I in "$_libdir/codecs" "$_libdir/win32" /usr/local/lib/codecs /usr/lib/codecs /usr/local/lib/win32 /usr/lib/win32 ; do
         if test -d "$I" ; then
@@ -7169,7 +7171,7 @@
   _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/mplayer/vidix/\" "
 elif test "$_vidix_external" = yes ; then
   _ld_vidix_external="-lvidix"
-  _def_vidix_pfx='#define MP_VIDIX_PFX "" '
+  _def_vidix_pfx="#define MP_VIDIX_PFX \"$_libdir\" \"/vidix/\" "
 fi
 
 if test "$_vidix" = yes; then
