diff -urN mplayer-svn-r26450.orig/gui/bitmap.c mplayer-svn-r26450/gui/bitmap.c
--- mplayer-svn-r26450.orig/gui/bitmap.c	2008-02-27 01:48:45 +0200
+++ mplayer-svn-r26450/gui/bitmap.c	2008-04-14 23:07:29 +0300
@@ -5,8 +5,16 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "bitmap.h"
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/avcodec.h>
+#else
 #include "libavcodec/avcodec.h"
+#endif
+#ifdef USE_LIBAVUTIL_SO
+#include <ffmpeg/intreadwrite.h>
+#else
 #include "libavutil/intreadwrite.h"
+#endif
 #include "libvo/fastmemcpy.h"
 
 static int pngRead( unsigned char * fname,txSample * bf )
