--- MPlayer-0.90pre7/cfgparser.c.orig	2002-08-31 18:04:06 +0300
+++ MPlayer-0.90pre7/cfgparser.c	2002-09-17 11:21:47 +0300
@@ -555,12 +555,12 @@
 			if ((*endptr == ':') || (*endptr == '/'))
 				tmp_float /= strtod(endptr+1, &endptr);
 
-			if (*endptr) {
-				mp_msg(MSGT_CFGPARSER, MSGL_ERR, "parameter must be a floating point number"
-				       " or a ratio (numerator[:/]denominator): %s\n", param);
-				ret = ERR_MISSING_PARAM;
-				goto out;
-			}
+//			if (*endptr) {
+//				mp_msg(MSGT_CFGPARSER, MSGL_ERR, "parameter must be a floating point number"
+//				       " or a ratio (numerator[:/]denominator): %s\n", param);
+//				ret = ERR_MISSING_PARAM;
+//				goto out;
+//			}
 
 			if (conf[i].flags & CONF_MIN)
 				if (tmp_float < conf[i].min) {
