diff -urN mplayer-svn-r19427.orig/libdha/Makefile mplayer-svn-r19427/libdha/Makefile
--- mplayer-svn-r19427.orig/libdha/Makefile	2006-08-18 20:55:36 +0300
+++ mplayer-svn-r19427/libdha/Makefile	2006-08-19 14:13:41 +0300
@@ -15,12 +15,8 @@
 #SHORTNAME = libdha.so
 VSHORTNAME = libdha.so
 endif
-ifeq ($(TARGET_WIN32),yes)
 LIBNAME = libdha.a
 SHORTNAME = libdha.a
-else
-LIBNAME = libdha.so.$(VERSION)
-endif
 
 SRCS=libdha.c \
      mtrr.c \
@@ -52,14 +48,7 @@
 	$(CC) -c $(CFLAGS) -o $@ $<
 
 $(LIBNAME):     $(OBJS)
-ifeq ($(TARGET_WIN32),yes)
 	$(AR) r $(LIBNAME) $(OBJS)
-else
-	#$(CC) -shared $(SONAME_FLAGS) -o $(LIBNAME) $(OBJS) $(LIBS)
-	$(CC) -shared -Wl,-soname -Wl,$(LIBNAME) -o $(LIBNAME) $(OBJS) $(LIBS)
-	ln -sf $(LIBNAME) $(SHORTNAME)
-	ln -sf $(LIBNAME) $(VSHORTNAME)
-endif
 
 all:    $(LIBNAME) $(SHORTNAME)
 
@@ -85,12 +74,15 @@
 	-mkdir -p $(LIBDIR)
 	install -m 755 $(INSTALLSTRIP) -p $(LIBNAME) $(LIBDIR)/$(LIBNAME)
 	rm -f $(LIBDIR)/libdha.so
+ifneq ($(LIBNAME),$(SHORTNAME))
 	ln -sf $(LIBNAME) $(LIBDIR)/$(SHORTNAME)
+else
 ifeq ($(TARGET_OS),OpenBSD)
 	-ldconfig -R
 else
 	-ldconfig
 endif
+endif
 
 uninstall:
 	rm -f $(LIBDIR)/libdha.so $(LIBDIR)/$(SHORTNAME) $(LIBDIR)/$(LIBNAME)
diff -urN mplayer-svn-r19427.orig/Makefile mplayer-svn-r19427/Makefile
--- mplayer-svn-r19427.orig/Makefile	2006-08-17 13:36:39 +0300
+++ mplayer-svn-r19427/Makefile	2006-08-19 14:15:13 +0300
@@ -226,7 +226,7 @@
 COMMON_LIBS += tremor/libvorbisidec.a
 endif
 ifeq ($(VIDIX),yes)
-COMMON_DEPS += libdha/libdha.so vidix/libvidix.a
+COMMON_DEPS += libdha/libdha.a vidix/libvidix.a
 endif
 ifeq ($(FAME),yes)
 COMMON_DEPS += libfame/libfame.a
@@ -325,10 +325,10 @@
 tremor/libvorbisidec.a:
 	$(MAKE) -C tremor
 
-libdha/libdha.so:
+libdha/libdha.a:
 	$(MAKE) -C libdha
 
-vidix/libvidix.a: libdha/libdha.so
+vidix/libvidix.a: libdha/libdha.a
 	$(MAKE) -C vidix
 
 Gui/libgui.a:
@@ -417,7 +417,6 @@
 
 install: $(ALL_PRG)
 ifeq ($(VIDIX),yes)
-	$(MAKE) -C libdha install
 	$(MAKE) -C vidix install
 endif
 	$(INSTALL) -d $(BINDIR)
@@ -581,7 +580,7 @@
 loader/dmo/libDMO_Filter.a: .norecurse $(wildcard loader/dmo/*.[ch])
 loader/dshow/libDS_Filter.a: .norecurse $(wildcard loader/dshow/*.[ch])
 
-libdha/libdha.so: .norecurse $(wildcard libdha/*.[ch])
+libdha/libdha.a: .norecurse $(wildcard libdha/*.[ch])
 vidix/libvidix.a: .norecurse $(wildcard vidix/*.[ch])
 Gui/libgui.a: .norecurse $(wildcard Gui/*.[ch] Gui/*/*.[ch] Gui/*/*/*.[ch])
 
