--- MPlayer-20020901/libmpdvdkit2/libdvdcss.c.warn	Sat Aug 17 00:50:22 2002
+++ MPlayer-20020901/libmpdvdkit2/libdvdcss.c	Sun Sep  1 17:40:22 2002
@@ -133,7 +133,7 @@
  * dvdcss_open() returns a handle to be used for all subsequent \e libdvdcss
  * calls. If an error occured, NULL is returned.
  */
-extern dvdcss_t dvdcss_open ( char *psz_target )
+extern dvdcss_t dvdcss_open ( const char *psz_target )
 {
     int i_ret;
 
--- MPlayer-20020901/libmpdvdkit2/dvdcss.h.warn	Sat Aug 17 00:50:21 2002
+++ MPlayer-20020901/libmpdvdkit2/dvdcss.h	Sun Sep  1 17:40:22 2002
@@ -69,7 +69,7 @@
 /*
  * Exported prototypes.
  */
-extern dvdcss_t dvdcss_open  ( char *psz_target );
+extern dvdcss_t dvdcss_open  ( const char *psz_target );
 extern int      dvdcss_close ( dvdcss_t );
 extern int      dvdcss_title ( dvdcss_t,
                                int i_block );
--- MPlayer-20020901/libmpdvdkit2/dvd_input.c.warn	Sat Aug 17 00:50:21 2002
+++ MPlayer-20020901/libmpdvdkit2/dvd_input.c	Sun Sep  1 17:40:22 2002
@@ -27,7 +27,7 @@
 
 #include "dvdcss.h"
 
-dvdcss_handle (*DVDcss_open)  (const char *);
+dvdcss_t      (*DVDcss_open)  (const char *);
 int           (*DVDcss_close) (dvdcss_handle);
 int           (*DVDcss_seek)  (dvdcss_handle, int, int);
 int           (*DVDcss_title) (dvdcss_handle, int); 
