diff -urN mplayer-svn-r21352.orig/libdha/Makefile mplayer-svn-r21352/libdha/Makefile
--- mplayer-svn-r21352.orig/libdha/Makefile	2006-11-27 14:05:38 +0200
+++ mplayer-svn-r21352/libdha/Makefile	2006-11-28 17:00:38 +0200
@@ -15,12 +15,8 @@
 #SHORTNAME = libdha.so
 VSHORTNAME = libdha.so
 endif
-ifeq ($(TARGET_WIN32),yes)
 LIBNAME = libdha.a
 SHORTNAME = libdha.a
-else
-LIBNAME = libdha.so.$(VERSION)
-endif
 
 SRCS=libdha.c \
      mtrr.c \
@@ -52,14 +48,7 @@
 	$(CC) -c $(CFLAGS) -o $@ $<
 
 $(LIBNAME):     $(OBJS)
-ifeq ($(TARGET_WIN32),yes)
 	$(AR) r $(LIBNAME) $(OBJS)
-else
-	#$(CC) -shared $(SONAME_FLAGS) -o $(LIBNAME) $(OBJS) $(LIBS)
-	$(CC) -shared -Wl,-soname -Wl,$(LIBNAME) -o $(LIBNAME) $(OBJS) $(LIBS)
-	ln -sf $(LIBNAME) $(SHORTNAME)
-	ln -sf $(LIBNAME) $(VSHORTNAME)
-endif
 
 all:    $(LIBNAME) $(SHORTNAME)
 
@@ -83,12 +72,15 @@
 	-mkdir -p $(LIBDIR)
 	install -m 755 $(INSTALLSTRIP) -p $(LIBNAME) $(LIBDIR)/$(LIBNAME)
 	rm -f $(LIBDIR)/libdha.so
+ifneq ($(LIBNAME),$(SHORTNAME))
 	ln -sf $(LIBNAME) $(LIBDIR)/$(SHORTNAME)
+else
 ifeq ($(TARGET_OS),OpenBSD)
 	-ldconfig -R
 else
 	-ldconfig
 endif
+endif
 
 uninstall:
 	rm -f $(LIBDIR)/libdha.so $(LIBDIR)/$(SHORTNAME) $(LIBDIR)/$(LIBNAME)
diff -urN mplayer-svn-r21352.orig/Makefile mplayer-svn-r21352/Makefile
--- mplayer-svn-r21352.orig/Makefile	2006-11-28 12:50:10 +0200
+++ mplayer-svn-r21352/Makefile	2006-11-28 17:05:44 +0200
@@ -105,7 +105,7 @@
 PARTS-$(FAAD_INTERNAL)            += libfaad2
 COMMON_LIBS-$(TREMOR_INTERNAL)    += tremor/libvorbisidec.a
 PARTS-$(TREMOR_INTERNAL)          += tremor
-LIBS_MPLAYER-$(VIDIX)             += vidix/libvidix.a
+LIBS_MPLAYER-$(VIDIX)             += vidix/libvidix.a libdha/libdha.a
 PARTS-$(VIDIX)                    += libdha vidix
 COMMON_LIBS-$(DVDREAD_INTERNAL)   += dvdread/libdvdread.a
 PARTS-$(DVDREAD_INTERNAL)         += dvdread
@@ -213,10 +213,10 @@
 tremor/libvorbisidec.a:
 	$(MAKE) -C tremor
 
-libdha/libdha.so:
+libdha/libdha.a:
 	$(MAKE) -C libdha
 
-vidix/libvidix.a: libdha/libdha.so
+vidix/libvidix.a: libdha/libdha.a
 	$(MAKE) -C vidix
 
 Gui/libgui.a:
@@ -250,7 +250,6 @@
 
 install: $(ALL_PRG)
 ifeq ($(VIDIX),yes)
-	$(MAKE) -C libdha install
 	mkdir -p $(LIBDIR)/mplayer/vidix
 	-install -m 755 $(INSTALLSTRIP) -p vidix/drivers/*.so $(LIBDIR)/mplayer/vidix
 endif
@@ -412,7 +411,7 @@
 libfaad2/libfaad2.a: .norecurse $(wildcard libfaad2/*.[ch] libfaad2/*/*.[ch])
 
 loader/libloader.a: .norecurse $(wildcard loader/*.[chSs])
-libdha/libdha.so: .norecurse $(wildcard libdha/*.[ch])
+libdha/libdha.a: .norecurse $(wildcard libdha/*.[ch])
 vidix/libvidix.a: .norecurse $(wildcard vidix/*.[ch])
 Gui/libgui.a: .norecurse $(wildcard Gui/*.[ch] Gui/*/*.[ch] Gui/*/*/*.[ch])
 
