diff -urN MPlayer-1.0pre4.orig/configure MPlayer-1.0pre4/configure
--- MPlayer-1.0pre4.orig/configure	2004-04-26 16:44:06 +0700
+++ MPlayer-1.0pre4/configure	2004-05-02 23:26:54 +0700
@@ -3890,8 +3890,8 @@
 
 _def_fame='#undef USE_LIBFAME'
 if test "$_fame" = yes ; then
-  _def_fame='#define USE_LIBFAME 1'	
-  _ld_fame='libfame/libfame.a'
+  _def_fame='#define USE_LIBFAME 1'
+  _ld_fame='-lfame'
 fi
 
 
diff -urN MPlayer-1.0pre4.orig/libmpcodecs/vf_fame.c MPlayer-1.0pre4/libmpcodecs/vf_fame.c
diff -urN MPlayer-1.0pre4.orig/Makefile MPlayer-1.0pre4/Makefile
--- MPlayer-1.0pre4.orig/Makefile	2004-04-13 18:22:37 +0700
+++ MPlayer-1.0pre4/Makefile	2004-05-02 23:22:41 +0700
@@ -47,9 +47,6 @@
 ifeq ($(VIDIX),yes)
 PARTS += libdha vidix
 endif
-ifeq ($(FAME),yes)
-PARTS += libfame
-endif
 ifeq ($(DVDKIT2),yes)
 PARTS += libmpdvdkit2
 else
@@ -80,9 +77,6 @@
 ifeq ($(VIDIX),yes)
 COMMON_DEPS += libdha/libdha.so vidix/libvidix.a
 endif
-ifeq ($(FAME),yes)
-COMMON_DEPS += libfame/libfame.a
-endif
 ifeq ($(DVDKIT2),yes)
 ifeq ($(DVDKIT_SHARED),yes)
 COMMON_DEPS += libmpdvdkit2/libmpdvdkit.so
@@ -122,9 +116,6 @@
 loader/libloader.a:
 	$(MAKE) -C loader
 
-libfame/libfame.a:
-	$(MAKE) -C libfame
-
 libmpdemux/libmpdemux.a:
 	$(MAKE) -C libmpdemux
 
