--- mplayer-svn-r19416/libvo/osd_template.c~	2006-06-13 16:11:09 +0400
+++ mplayer-svn-r19416/libvo/osd_template.c	2006-08-18 14:59:48 +0400
@@ -204,7 +204,7 @@ static inline void RENAME(vo_draw_alpha_
     for(y=0;y<h;y++){
         register unsigned char *dst = dstbase;
         register int x;
-#if defined(ARCH_X86) || defined(ARCH_X86_64)
+#if defined(ARCH_X86)
 #ifdef HAVE_MMX
     asm volatile(
 	PREFETCHW" %0\n\t"
@@ -324,7 +324,7 @@ static inline void RENAME(vo_draw_alpha_
 #endif /* HAVE_MMX */
     for(y=0;y<h;y++){
         register int x;
-#if defined(ARCH_X86) || defined(ARCH_X86_64)
+#if defined(ARCH_X86)
 #ifdef HAVE_MMX
 #ifdef HAVE_3DNOW
     asm volatile(
