diff -urN MPlayer-svn-20060823.orig/configure MPlayer-svn-20060823/configure
--- MPlayer-svn-20060823.orig/configure	2006-08-23 14:46:32 +0300
+++ MPlayer-svn-20060823/configure	2006-08-23 14:56:39 +0300
@@ -4695,7 +4695,7 @@
 _def_libfame='#undef USE_LIBFAME'
 if test "$_libfame" = yes ; then
   _def_libfame='#define USE_LIBFAME 1'
-  _ld_libfame='libfame/libfame.a'
+  _ld_libfame='-lfame'
 fi
 
 
diff -urN MPlayer-svn-20060823.orig/Makefile MPlayer-svn-20060823/Makefile
--- MPlayer-svn-20060823.orig/Makefile	2006-08-23 14:46:31 +0300
+++ MPlayer-svn-20060823/Makefile	2006-08-23 14:55:59 +0300
@@ -178,9 +178,6 @@
 ifeq ($(VIDIX),yes)
 PARTS += libdha vidix
 endif
-ifeq ($(FAME),yes)
-PARTS += libfame
-endif
 ifeq ($(DVDKIT2),yes)
 PARTS += libmpdvdkit2
 endif
@@ -236,9 +233,6 @@
 ifeq ($(VIDIX),yes)
 COMMON_DEPS += libdha/libdha.so vidix/libvidix.a
 endif
-ifeq ($(FAME),yes)
-COMMON_DEPS += libfame/libfame.a
-endif
 ifeq ($(DVDKIT2),yes)
 COMMON_DEPS += libmpdvdkit2/libmpdvdkit.a
 endif
@@ -276,9 +270,6 @@
 loader/libloader.a:
 	$(MAKE) -C loader
 
-libfame/libfame.a:
-	$(MAKE) -C libfame
-
 libass/libass.a:
 	$(MAKE) -C libass
 
